

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Identifying OTUs with References &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: Mapping reads to references" href="mapping_reads_to_refs.html" />
    <link rel="prev" title="2. Classification with curated databases" href="id_with_curated_db.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="database.html">Database Classification</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Identifying OTUs with References</a><ul>
<li class="toctree-l3"><a class="reference internal" href="mapping_reads_to_refs.html">Extension: Mapping reads to references</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="phylogeny_based.html">Phylogenetic classification</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../otuid.html">E: Identifying OTU sequences</a> &raquo;</li>
        
      <li>Identifying OTUs with References</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otuid/id_local.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="id-local">
<span id="identifying-otus-with-references"></span><h1>Identifying OTUs with References<a class="headerlink" href="#id-local" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Global and curated databases are very useful for assigning broad taxonomy, but don’t do great identifying our OTUs to lower taxonomic levels for these sorts of samples. Depending on your study, it’s often very useful to generate your own set of high-accuracy reference sequences based on a reference set of specimens. Obviously the feasibility of this depends on the richness of your study community and how cryptic your study taxa is.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data to this tutorial is a FASTA file of sequences that you want to classify, and a FASTA file of sequences corresponding to known reference specimens. If you’ve been following along step-by-step, you can classify the ASVs or OTUs you produced in previous sections. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">otus_greedy_0.97.fasta</span></code> within the <a class="reference internal" href="../data.html#sectionedata"><span class="std std-ref">sectionE archive</span></a> can be used as example data. For reference data, the files <code class="docutils literal notranslate"><span class="pre">references_CCCPbarcodes.fasta</span></code> and <code class="docutils literal notranslate"><span class="pre">taxonomy_CCCPbarcodes.csv</span></code> from the same archive can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#blast"><span class="std std-ref">BLAST</span></a> software.</p>
</div>
</div>
<div class="section" id="using-blast">
<h2>Using BLAST<a class="headerlink" href="#using-blast" title="Permalink to this headline">¶</a></h2>
<p>We will use BLAST to search our OTUs against the FASTA of references. <span class="guilabel">Run the following BLAST command</span>, remembering to replace the <code class="docutils literal notranslate"><span class="pre">otus.fasta</span></code> and <code class="docutils literal notranslate"><span class="pre">references.fasta</span></code> with paths to the respective files, and <code class="docutils literal notranslate"><span class="pre">output.txt</span></code> with a sensible name.</p>
<pre class="literal-block">blastn -query <span class="var">otus.fasta</span> -subject <span class="var">references.fasta</span> -outfmt 6 -out <span class="var">output.txt</span> -num_threads 1 -evalue 0.001 -perc_identity 97</pre>
<p>This is similar to the command we used in the tutorial on <a class="reference internal" href="id_with_megan.html#id-using-megan"><span class="std std-ref">classification with BLAST</span></a>. However, we know that both our OTUs and our reference set are likely to all be closely related, so we’re setting a strict <code class="docutils literal notranslate"><span class="pre">-evalue</span></code> and a threshold percentage identity so that we don’t simply get every OTU matching against every reference.</p>
<p>Use <code class="docutils literal notranslate"><span class="pre">cat</span></code> to view the output file. We’re using the standard blast tabulated output (<code class="docutils literal notranslate"><span class="pre">-​</span> <span class="pre">outfmt</span> <span class="pre">6</span></code>); you can find out what the columns refer to <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK279684/">here</a>. Rather obviously, the first refers to the query sequence, the second the subject, and the third the percent identity. You can see we’ve got some clear hits for some of our OTUs!</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Do any of our OTUs hit multiple different references? Why might this be?</p></li>
<li><p>Do the OTUs matching Coleoptera references correspond to those assigned to Coleoptera using MEGAN?</p></li>
<li><p>Does this give us more information about any of our OTUs compared with the global database search?</p></li>
</ul>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>If you have a comprehensive reference set of barcodes for all of the species you expect to find in your metabarcoding, or at least all of the ones you’re interested in, you could actually skip a considerable proportion of the metabarcoding pipeline. This is because there’s no need to necessarily filter and delimit OTUs from scratch - you already have your “OTU” sequences and once you’ve finished processing your reads, you can simply search them against your references. You can read more about this and try it out in the extension: <a class="reference internal" href="mapping_reads_to_refs.html#map-to-ref"><span class="std std-ref">mapping reads to references</span></a>.</p>
<p>A final method for generating taxonomic information will be covered in the next tutorial on <a class="reference internal" href="phylogeny_based.html#phylogenetic-classification"><span class="std std-ref">phylogenetic classification</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="mapping_reads_to_refs.html" class="btn btn-neutral float-right" title="Extension: Mapping reads to references" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="id_with_curated_db.html" class="btn btn-neutral float-left" title="2. Classification with curated databases" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>