

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Identification using GenBank and MEGAN &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: MEGAN settings" href="extensions/megan_settings.html" />
    <link rel="prev" title="Extension: Mapping reads to CROP and swarm OTUs" href="extensions/map_to_other_otus.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">The Practical Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="mapping_reads.html">Mapping reads to OTUs</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Identification using GenBank and MEGAN</a><ul>
<li class="toctree-l3"><a class="reference internal" href="extensions/megan_settings.html">Extension: MEGAN settings</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="id_with_curated_db.html">Identifiation using a curated database</a></li>
<li class="toctree-l2"><a class="reference internal" href="id_local.html">Identification using local database</a></li>
<li class="toctree-l2"><a class="reference internal" href="mapping_reads_to_refs.html">Mapping reads to references</a></li>
<li class="toctree-l2"><a class="reference internal" href="id_contigs.html">Identifying contigs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../otuid.html">Identifying OTUs and Generating Outputs</a> &raquo;</li>
        
      <li>Identification using GenBank and MEGAN</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otuid/id_with_megan.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="id-using-megan">
<span id="identification-using-genbank-and-megan"></span><h1>Identification using GenBank and MEGAN<a class="headerlink" href="#id-using-megan" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<p>The most straightforward automated way of assigning taxonomy to OTUs is to BLAST the OTUs against the GenBank nucleotide database. The server has a subset of this database, and the blast executables.</p>
<p>Select one of your OTU files and blast it against nt using the following command:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ blastn -db /AMM/blastdb/nt -query ​in.fasta​ -outfmt <span class="m">5</span> -out ​out.xml​ -evalue <span class="m">0</span>.001
</pre></div>
</div>
<p>This command generates a very large XML file containing the full record of all the alignments BLAST has found for our OTUs. We need to transfer this to our computer for the next step, but to save bandwidth, let’s first compress the xml using zip:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ zip ​out.xml.zip in.xml
</pre></div>
</div>
<p>Using your FTP client, or whatever file transfer method you like, transfer the zipped XML file to your computer and extract it.</p>
<p>We will use the software MEGAN to parse the BLAST results and assign taxonomy to our OTUs. This is one of many algorithms out there for doing this - we introduce it here because it has a handy GUI interface and you’re probably bored of looking at the terminal.</p>
<p>Open up MEGAN. It usually takes a little while to open because it has to load the entire NCBI taxonomy into memory and display it. I bet you miss the efficiency of the terminal already.</p>
<p>MEGAN works through each OTU and find the location on the NCBI taxonomy tree for each GenBank sequence the OTU had a hit against. It then uses a Lowest Common Ancestor algorithm to estimate the most appropriate location on the tree for an OTU, and assigns the OTU the appropriate taxonomy.</p>
<p>Once MEGAN has opened and loaded the tree, you should see a very high-level cladogram of living organisms. This is the entire NCBI taxonomy. To map the OTU BLAST data onto this, you need to load the XML. Go to ​File &gt; Import From BLAST​. In the Files tab of the window that appears, use the button to the right of the first box to browse to and select your XML file. MEGAN will automatically fill the third box. It should look something like this:</p>
<img alt="../_images/megan_screenshot.png" class="align-center" src="../_images/megan_screenshot.png" />
<p>Go to the ​LCA Params tab at the top. Here you will see the parameters that MEGAN uses when assigning taxonomy using its lowest common ancestor algorithm. For now we’ll leave these as default and just press Apply​.</p>
<p>Once MEGAN has finished you should see a reduced version of the taxonomy tree. It may not be very detailed: at the top bar, select Rank and choose Species. Have a look at the tree.</p>
<ul class="simple">
<li><p>Are all the OTUs Coleoptera?</p></li>
</ul>
<p>Each circle on the tree is one or more OTUs that have been assigned to a node. The larger the circle, the more OTUs have been assigned to that node. If you click on a node, you’ll see two values. ​Assigned is the number of OTUs assigned to that node, ​Summed is the number of OTUs assigned to that node and all child nodes. If you ​right click on a node and click ​Inspect​, you can see more details about that node and the OTU(s) assigned to it, as well as all the BLAST information. The greyed out BLAST hits are those that aren’t taken into account in the LCA analysis.</p>
<p>You’ll notice that many OTUs have been assigned to internal nodes. Inspect some of these.</p>
<ul class="simple">
<li><p>Why do you think the algorithm has assigned them to internal nodes?</p></li>
<li><p>Do you think algorithm is always correct?</p></li>
</ul>
<p>To output the taxonomic assignment for all of the OTUs for use in analysis, we need to select all of the nodes. You can do this by going to ​Select &gt; All Nodes​. Then go to ​File &gt; Export &gt; Text (CSV) Format​. For the Choose data to export: field, select ​readName_to_taxonPath​, click OK and select your output location. This generates a comma-separated table with the OTU name and full NCBI taxon path of the assigned node.</p>
<p>Use your FTP client to send this file to your directory on the server. We’ll come back to it later.</p>
<p>Here we have gone with the default settings on MEGAN. To learn how we can edit these, look at this <a class="reference internal" href="extensions/megan_settings.html#megan-settings"><span class="std std-ref">extension task</span></a>
Alternatively, head over to the next section on <a class="reference internal" href="id_with_curated_db.html#id-curated"><span class="std std-ref">identifiction using curated databases</span></a>.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="extensions/megan_settings.html" class="btn btn-neutral float-right" title="Extension: MEGAN settings" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extensions/map_to_other_otus.html" class="btn btn-neutral float-left" title="Extension: Mapping reads to CROP and swarm OTUs" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>