

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>1) Mapping reads to OTUs &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/customjs/custom.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: Mapping reads to CROP and swarm OTUs" href="extensions/map_to_other_otus.html" />
    <link rel="prev" title="Identifying OTUs and Generating Outputs" href="../otuid.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">1) Mapping reads to OTUs</a><ul>
<li class="toctree-l3"><a class="reference internal" href="extensions/map_to_other_otus.html">Extension: Mapping reads to CROP and swarm OTUs</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="id_with_megan.html">2) Identification using GenBank and MEGAN</a></li>
<li class="toctree-l2"><a class="reference internal" href="id_with_curated_db.html">3) Identifiation using a curated database</a></li>
<li class="toctree-l2"><a class="reference internal" href="id_local.html">4) Identification using local database</a></li>
<li class="toctree-l2"><a class="reference internal" href="mapping_reads_to_refs.html">Extension: Mapping reads to references</a></li>
<li class="toctree-l2"><a class="reference internal" href="id_contigs.html">Identifying contigs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../otuid.html">Identifying OTUs and Generating Outputs</a> &raquo;</li>
        
      <li>1) Mapping reads to OTUs</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otuid/mapping_reads.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="mapping-reads-to-otus">
<h1>1) Mapping reads to OTUs<a class="headerlink" href="#mapping-reads-to-otus" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<p>This step takes our trimmed, merged and quality filtered reads and tries to find which OTU each read belongs to. We take our reads prior to the more recent read filtering because this filtering may have removed many true but rare variants of our OTUs, or slight PCR or sequencing-induced error variants. These reads get matched to the closest OTU, but we set a similarity cut-off so that any reads that are major errors will not match to any OTU and be ignored.</p>
<p>We are going to use our 3% clustering OTUs for this at first, using the <code class="docutils literal notranslate"><span class="pre">​VSEARCH</span> <span class="pre">--usearch_global</span></code> command to search the reads against the OTUs, and output an OTU table.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --usearch_global mbc_concat.fasta -db ​otus.fasta ​-id <span class="m">0</span>.97 -uc ​mapresults.uc
</pre></div>
</div>
<p>We use the <code class="docutils literal notranslate"><span class="pre">-id</span> <span class="pre">0.97</span></code> parameter to set a 3% similarity cutoff, which is obviously appropriate for the 3% OTUs. The algorithm finds the best match in the database for each query sequence and output this information in a <code class="docutils literal notranslate"><span class="pre">.uc</span></code> file, which is a tabulated output. Have a look at this file using head or cat. Each line shows the hit statistics for each input read against its best match, if it has one. You should be able to see that each read has a <code class="docutils literal notranslate"><span class="pre">sample=</span></code> part to the name that we added earlier to identify its source. So all we need to do now is count up the number of hits for each OTU for each sample.</p>
<p>Thankfully, vsearch can do this for us. Run the command above again, but this time swap out the <code class="docutils literal notranslate"><span class="pre">-uc</span> <span class="pre">out.uc</span></code> o​ption for <code class="docutils literal notranslate"><span class="pre">-otutabout</span> <span class="pre">​out.tsv.</span></code> Have a look at the table output using head or cat. Perfect!</p>
<p>If your interested in how you’d map reads to CROP and swarm OTUs, have a look at the next <a class="reference internal" href="extensions/map_to_other_otus.html#map-to-other-otus"><span class="std std-ref">extension</span></a>.
Alternatively, move on to the next <a class="reference internal" href="id_with_megan.html#id-using-megan"><span class="std std-ref">section</span></a></p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="extensions/map_to_other_otus.html" class="btn btn-neutral float-right" title="Extension: Mapping reads to CROP and swarm OTUs" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../otuid.html" class="btn btn-neutral float-left" title="Identifying OTUs and Generating Outputs" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>