

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Phylogenetic Classification &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Contact us" href="../contactus.html" />
    <link rel="prev" title="Extension: Mapping reads to references" href="mapping_reads_to_refs.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="database.html">Database Classification</a></li>
<li class="toctree-l2"><a class="reference internal" href="id_local.html">Identifying OTUs with References</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Phylogenetic Classification</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../otuid.html">E: Identifying OTU sequences</a> &raquo;</li>
        
      <li>Phylogenetic Classification</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otuid/phylogeny_based.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="phylogenetic-classification">
<span id="id1"></span><h1>Phylogenetic Classification<a class="headerlink" href="#phylogenetic-classification" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>As we saw in the previous section (<a class="reference internal" href="../building_otu_phylogeny.html#phylogeny"><span class="std std-ref">Building OTU Phylogeny</span></a>), we can place OTUs onto phylogenetic trees constructed from reference sequences. If we then annotate the tree with the taxonomy of these reference sequences, we often observe OTUs placed within clades that correspond to a taxonomic group. Thus it follows that we should be able to use this tree to assign taxonomy to OTUs based on their placement.</p>
<p>The upside of using phylogeny for taxonomic classification is that the process of phylogenetic placement, if performed accurately, is much more thorough than similarity or local alignment used in the database-searching approaches. Since phylogenetic reconstruction takes account of all sequences in the dataset at the same time, the placement of an OTU in a phylogeny is much more accurate in showing the degree of relation between an OTU and references than a database search.</p>
<p>The downside of using phylogeny is that the size of the reference set is much more limited, by necessity of the increased computational requirements. Rather than using all possible sequences available on public databases, we are forced to select a limited taxonomic range and generally then only a subset of sequences within that range. This means that the application of phylogenetic placement to taxonomic classification is limited to a specific taxon, such as the Coleoptera, and the depth of classifications achieved is likely to be less than that of database classification.</p>
<p>The optimal classification may then be best achieved through a combination of broad scale database classification on the entire OTU set, then detailed phylogenetic placement for taxa of interest. Phylogenetic placement is best applied when you want certainty about the membership (or not) of a set of OTUs within a particular taxon, and accurate group-level placement within that taxon. For example, here we will place our OTUs onto a Coleoptera tree, which will provide us with clear evidence of which OTUs are Coleoptera or not. Then we will see which families the OTUs are placed within, giving us largely-accurate family-level assignment. However, genus or species assignments largely won’t be possible, as we simply don’t have the depth of coverage for this.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a phylogenetic tree with query sequences (i.e. OTUs) placed among a set of references with known taxonomy. You will also need a table recording taxonomy for these references. If you have been following along step-by-step, you can use the phylogenetic tree you produced in the <cite>placing OTUs in trees&lt;phylogenetic_placement&gt;</cite> tutorial, and the corresponding taxonomy table. Alternatively, the files <code class="docutils literal notranslate"><span class="pre">phylogeny_GB_CCCP_otus_greedy_0.97.tre</span></code> and <code class="docutils literal notranslate"><span class="pre">taxonomy_GB_CCCP.csv</span></code> within the <a class="reference internal" href="../data.html#sectionedata"><span class="std std-ref">sectionE archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#phylostuff"><span class="std std-ref">phylabel.R</span></a> and <a class="reference internal" href="../gettingstarted/setup/installing_software.html#phylostuff"><span class="std std-ref">treedentify.R</span></a> scripts.</p>
</div>
</div>
<div class="section" id="approaches">
<h2>Approaches<a class="headerlink" href="#approaches" title="Permalink to this headline">¶</a></h2>
<p>This task has two main steps. First we must place OTUs onto a phylogenetic tree. We cover one way of doing this in the <a class="reference internal" href="../otu_phylogeny/placing_otus_in_trees.html#phylogenetic-placement"><span class="std std-ref">phylogenetic placement</span></a> tutorial, we strongly suggest you head over and do that tutorial before this one.</p>
<p>The second step involves parsing the taxonomy of the references and inferring the taxonomy of the OTUs. We do this with some custom R scripts, because we find this option is the most flexible. Here we will use these R scripts as standalone tools, but if you’re familiar with R we suggest you open up the scripts and see what they’re doing. The key function within these scripts is the <code class="docutils literal notranslate"><span class="pre">nodelabel.phylo</span></code> function from geiger <a class="reference external" href="https://www.rdocumentation.org/packages/geiger/versions/2.0.7/topics/nodelabel.phylo">https://www.rdocumentation.org/packages/geiger/versions/2.0.7/topics/nodelabel.phylo</a>, which does the work of assigning taxonomy to the internal nodes of the tree based on the tips.</p>
</div>
<div class="section" id="taxonomising-the-tree">
<h2>Taxonomising the tree<a class="headerlink" href="#taxonomising-the-tree" title="Permalink to this headline">¶</a></h2>
<p>First, we must assign taxonomy to the tree. We do this with the <code class="docutils literal notranslate"><span class="pre">phylabel.R</span></code> script. This script takes as input the tree and the taxonomy, and outputs a tree with taxonomic names assigned to some of the internal nodes. We’ve seen this script before for relabelling terminals, but it also works on nodes.</p>
<pre class="literal-block">phylabel.R -p <span class="var">input.tre</span> -x -t <span class="var">taxonomy.csv</span> -o <span class="var">output.tre</span></pre>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Download this tree and open it in your tree viewer. Have a look at the internal nodes, looking particularly for those to which the script has assigned taxonomy.</p></li>
<li><p>Do the nodes correspond with the tips?</p></li>
<li><p>Are the taxonomic groups always monophyletic?</p></li>
<li><p>Do most of the OTUs look like they’ll have taxonomy assigned?</p></li>
</ul>
</div>
<p>You’ll see that a lot of the node names have quotation marks (<code class="docutils literal notranslate"><span class="pre">&quot;</span></code>). This is added when most, but not all, of the descendents of a node match this taxon. We can be more strict by adding the strict <code class="docutils literal notranslate"><span class="pre">-s</span></code> option to the <code class="docutils literal notranslate"><span class="pre">phylabel.R</span></code> command.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Run the script again with strict on.</p></li>
<li><p>Look at your new version of the tree. Have fewer nodes been assigned taxonomy? How will this affect the OTUs?</p></li>
<li><p>Which is more appropriate?</p></li>
</ul>
</div>
</div>
<div class="section" id="inferring-otu-classification">
<h2>Inferring OTU classification<a class="headerlink" href="#inferring-otu-classification" title="Permalink to this headline">¶</a></h2>
<p>Now that we have our taxonomised tree, we can infer the OTU classification. This is performed with another R script, <strong>treedentify</strong>, and the logic here is very straightforward: the script looks at the parent nodes of each OTU, and assigns the OTU the taxonomy of the most recent parent node with any taxonomy.</p>
<p>Run this as follows. We suggest you use as <code class="docutils literal notranslate"><span class="pre">input.tre</span></code> the first taxonomised tree you generated, without strict taxonomisation.</p>
<pre class="literal-block">treedentify.R -p <span class="var">input.tre</span> -e ^otu -u -o <cite>output.csv</cite></pre>
<p>The <code class="docutils literal notranslate"><span class="pre">-e</span></code> argument tells <strong>treedentify</strong> that we want to classify any tips beginning with <code class="docutils literal notranslate"><span class="pre">otu</span></code>. You might want to swap this for <code class="docutils literal notranslate"><span class="pre">uniq</span></code> or something else if your OTU or ASV sequences have different names. The <code class="docutils literal notranslate"><span class="pre">-u</span></code> argument tells <strong>treedentify</strong> to check NCBI to get the complete taxonomy of any matches.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Download and open the taxonomy classification table. Check a few of the OTUs against the tree. Do they fit?</p></li>
<li><p>Have all the OTUs recieved the same depth of classification? Why might some have fewer or more taxonomic levels?</p></li>
<li><p>Compare the classification of some of the OTUs against the classification achieved by the database approaches. Can you see any discrepancies? Which do you think is more accurate?</p></li>
</ul>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../contactus.html" class="btn btn-neutral float-right" title="Contact us" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="mapping_reads_to_refs.html" class="btn btn-neutral float-left" title="Extension: Mapping reads to references" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>