.. Analysis of Metagenetic data for Macroecology documentation master file, created by
   sphinx-quickstart on Tue Nov  3 10:33:09 2020.
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.

====================================================
Bioinformatic Methods for Biodiversity Metabarcoding
====================================================

.. toctree::
	:hidden:
	:maxdepth: 2
	:titlesonly:

	gettingstarted
	readprocessing
	filtering
	otu_asv_read
	building_otu_phylogeny
	otuid

Introduction
============

These resources aim to introduce a beginner to the bioinformatics required for metabarcoding through a course of self-guided exercises. These exercises take a set of example data from the stage of raw sequence reads through to the generation of real biodiversity data for use in a wide range of applications. Each exercise is generally centred around a particular step in metabarcoding, introducing the important concepts of this step, a software tool to perform this step, and key parameters to consider when using this softwarce. The main body of the course consists of a linear sequence of core exercises, with each core exercise building on the results of the previous core exercise and generating data for the next. These core exercises are supplemented by extensions, which explore more advanced topics around a bioinformatic step for those looking for an added challenge. There is also a substantial reference section covering basic topics in bioinformatics to assist those who are just starting out.

While we have made every effort to cover a wide range of metabarcoding bioinfirmatic topics, metabarcoding is a continually evolving field and it's impossible to be comprehensive. This course explicitly focuses on metabarcoding for bulk community samples of arthropods using a protein-coding markers, as distinct from many other forms of metabarcoding such as environmental DNA sequencing, dietary-derived metabarcoding, or microbiome sequencing using ribosomal RNA markers. As such, we do not tackle some areas, such as adaptation of the bioinformatic approaches to degraded DNA, and some areas we do cover may not be relevant, such as translation-based ASV filtering. Nonetheless, the majority of the core concepts are identical across all metabarcoding so we hope this resource will be useful to any metabarcoder, no matter their background.


TODO page links.






