

<!DOCTYPE html>
<html class="writer-html4" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Cheatsheet of useful commands &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/css/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../_static/language_data.js"></script>
        <script type="text/javascript" src="../../_static/js/custom.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Keeping your data organised" href="keeping_organised.html" />
    <link rel="prev" title="Loops in bash" href="loops.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../../gettingstarted.html">Getting started</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../setup.html">Setting up a bioinformatics machine</a></li>
<li class="toctree-l2"><a class="reference internal" href="../remote_access.html">Accessing linux servers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../how_to_use.html">How to use these resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="../toolbox_pipelines.html">The bioinformatics toolbox</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../cli_bioinformatics.html">Command line bioinformatics</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="UNIX_cmd_line.html">UNIX command line</a></li>
<li class="toctree-l3"><a class="reference internal" href="basic_UNIX.html">Basic UNIX commands</a></li>
<li class="toctree-l3"><a class="reference internal" href="nano.html">Using nano to edit documents</a></li>
<li class="toctree-l3"><a class="reference internal" href="grep.html">The grep command</a></li>
<li class="toctree-l3"><a class="reference internal" href="sed.html">The sed command</a></li>
<li class="toctree-l3"><a class="reference internal" href="loops.html">Loops in bash</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Cheatsheet of useful commands</a></li>
<li class="toctree-l3"><a class="reference internal" href="keeping_organised.html">Keeping your data organised</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../gettingstarted.html">Getting started</a> &raquo;</li>
        
          <li><a href="../cli_bioinformatics.html">Command line bioinformatics</a> &raquo;</li>
        
      <li>Cheatsheet of useful commands</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../_sources/gettingstarted/cli_bioinformatics/cheatsheet.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="cheatsheet-of-useful-commands">
<span id="cheatsheet"></span><h1>Cheatsheet of useful commands<a class="headerlink" href="#cheatsheet-of-useful-commands" title="Permalink to this headline">¶</a></h1>
<p>Here are some useful bash commands for doing command-line bioinformatics. These draw upon some of the basics described in other pages in this section.</p>
<div class="section" id="unwrapping-fasta-files">
<h2>Unwrapping FASTA files<a class="headerlink" href="#unwrapping-fasta-files" title="Permalink to this headline">¶</a></h2>
<p>A lot of software outputs wrapped FASTA files, where sequence lines are truncated at a certain number of characters and continued on the following line, like this:</p>
<pre class="literal-block">
&gt;sequence1
ATACGACGATAAGCAGACTAGACATATT
CGCCCGATTGACCG
&gt;sequence2
ACGCGCTGAAAAGCAGACTTTTTACACG
CGCGTAAACAACGC
</pre>
<p>This can be really annoying for quick and easy command line bash bioinformatics, because the number of lines for each sequence is unpredictable. The following perl one-liner, from <a class="reference external" href="https://www.biostars.org/p/9262/#118460">this biostars answer</a>, easily unwraps a FASTA file:</p>
<pre class="codebg literal-block">
perl -pe '$. &gt; 1 and /^&gt;/ ? print &quot;\n&quot; : chomp' <span class="var">input.fasta</span> &gt; <span class="var">output.fasta</span>
</pre>
<p>This command is great because it can easily be used within a pipe.</p>
</div>
<div class="section" id="replacing-bases">
<h2>Replacing bases<a class="headerlink" href="#replacing-bases" title="Permalink to this headline">¶</a></h2>
<p>You may sometimes have sequence data that isn’t quite appropriate for a piece of software, for example you have an alignment but the software won’t accept it, or you have ambiguity codes but the software only accepts <code class="docutils literal notranslate"><span class="pre">A</span></code>, <code class="docutils literal notranslate"><span class="pre">T</span></code>, <code class="docutils literal notranslate"><span class="pre">C</span></code>, <code class="docutils literal notranslate"><span class="pre">G</span></code> or <code class="docutils literal notranslate"><span class="pre">N</span></code>. You can use <code class="docutils literal notranslate"><span class="pre">sed</span></code> to remove or replace all offending bases, but you must be careful only to do so on non-header lines - you don’t want to edit your sequence headers by accident! We do this by adding a line specifier, <code class="docutils literal notranslate"><span class="pre">/^[^&gt;]/,</span></code> before the substitution command. This specifies that we want to only operate on lines that do not start with <code class="docutils literal notranslate"><span class="pre">&gt;</span></code>.</p>
<p>Degap an alignment (i.e. de-align):</p>
<pre class="codebg literal-block">
sed -e &quot;/^[^&gt;]/s/-//g&quot; <span class="var">input.fasta</span> &gt; <span class="var">output.fasta</span>
</pre>
<p>Replace any ambiguities with <code class="docutils literal notranslate"><span class="pre">N</span></code>:</p>
<pre class="codebg literal-block">
sed -e &quot;/^[^&gt;]/s/[YRWSKMDVHBX]/N/g&quot; <span class="var">input.fasta</span> &gt; <span class="var">output.fasta</span>
</pre>
</div>
<div class="section" id="getting-sequence-lengths">
<h2>Getting sequence lengths<a class="headerlink" href="#getting-sequence-lengths" title="Permalink to this headline">¶</a></h2>
<p>Here we delete header lines, delete gaps then count how many characters each line has.</p>
<pre class="codebg literal-block">
sed -e &quot;/^&gt;/d&quot; -e &quot;s/-//g&quot; <span class="var">input.fasta</span> | awk '{ print length }' &gt; <span class="var">lengths.txt</span>
</pre>
<p>The FASTA file must be unwrapped - if it’s not, you could add the perl one-liner above to the beginning of the pipe.</p>
<pre class="codebg literal-block">
perl -pe '$. &gt; 1 and /^&gt;/ ? print &quot;\n&quot; : chomp' <span class="var">input.fasta</span> |
sed -e &quot;/^&gt;/d&quot; -e &quot;s/-//g&quot; | awk '{ print length }' &gt; <span class="var">lengths.txt</span>
</pre>
<p>What to with these lengths? Well, we can get a rough idea of the distribution by counting the number of each unique length:</p>
<pre class="codebg literal-block">
sed -e &quot;/^&gt;/d&quot; -e &quot;s/-//g&quot; <span class="var">input.fasta</span> |
awk '{ print length }' |
sort | uniq -c | sort -hr
</pre>
<p>Or we could use an R one-liner to count the mean and standard deviation. Note that we’ve split this over lines to</p>
<pre class="codebg literal-block">
sed -e &quot;/^&gt;/d&quot; -e &quot;s/-//g&quot; <span class="var">input.fasta</span> |
awk '{ print length }' |
Rscript -e 'd&lt;-scan(&quot;stdin&quot;,quiet=TRUE);
            print(c(round(mean(d),0),
                    round(100 * mean(c(mean(d)-min(d),
                                        max(d)-mean(d)))/mean(d),
                          1)));'
</pre>
</div>
<div class="section" id="generating-a-random-fasta">
<h2>Generating a random FASTA<a class="headerlink" href="#generating-a-random-fasta" title="Permalink to this headline">¶</a></h2>
<p>Sometimes we want some dummy data to test on. This command makes 30 sequences, each 300 bases long:</p>
<pre class="codebg literal-block">
paste -d '\n'     &lt;(for i in {01..30}; do echo &quot;&gt;seq$i&quot;; done)     &lt;( cat /dev/urandom | tr -dc 'ATCG' | fold -w 300 | head -n 30 )     &gt; <span class="var">output.fasta</span>
</pre>
</div>
<div class="section" id="get-the-headers-of-a-fasta">
<h2>Get the headers of a FASTA<a class="headerlink" href="#get-the-headers-of-a-fasta" title="Permalink to this headline">¶</a></h2>
<p>There are two ways to do this:</p>
<pre class="codebg literal-block">
grep &quot;^&gt;&quot; <span class="var">input.fasta</span> | sed -e &quot;s/&gt;//&quot; &gt; <span class="var">headers.txt</span>
grep -oP &quot;(?&lt;=^&gt;).*$&quot; <span class="var">input.fasta</span> &gt; <span class="var">headers.txt</span>
</pre>
<p>The second one uses a fancy Regular Expression term called a “positive lookbehind”.</p>
</div>
<div class="section" id="extract-information-from-tables">
<h2>Extract information from tables<a class="headerlink" href="#extract-information-from-tables" title="Permalink to this headline">¶</a></h2>
<p>Say I have a table, <code class="docutils literal notranslate"><span class="pre">data.csv</span></code>, that looks like this:</p>
<pre class="literal-block">
seq1,Coleoptera,425
seq2,Diptera,256
seq3,Hemiptera,786
</pre>
<p>If I wanted to find the sequence names that are Coleoptera, I could do this:</p>
<pre class="codebg literal-block">
grep &quot;Coleoptera&quot; data.csv | cut -d, -f1 &gt; <span class="var">headers.txt</span>
</pre>
<p>Or the sequences where column three is greater than 500:</p>
<pre class="codebg literal-block">
awk -F',' ' $3 &gt; 500 ' data.csv | cut -d, -f1 &gt; <span class="var">headers.txt</span>
</pre>
<p>The functions <code class="docutils literal notranslate"><span class="pre">awk</span></code> or <code class="docutils literal notranslate"><span class="pre">grep</span></code> do the searching, while <code class="docutils literal notranslate"><span class="pre">cut</span></code> extracts the first column from the result. If your table is tab-delimited, you can skip the <code class="docutils literal notranslate"><span class="pre">-F</span></code> argument in <code class="docutils literal notranslate"><span class="pre">awk</span></code> and the <code class="docutils literal notranslate"><span class="pre">-d</span></code> argument in <code class="docutils literal notranslate"><span class="pre">cut</span></code>, tab-delimited is the default for both.</p>
</div>
<div class="section" id="extract-sequence-by-header">
<h2>Extract sequence by header<a class="headerlink" href="#extract-sequence-by-header" title="Permalink to this headline">¶</a></h2>
<p>Say we have a list of sequence headers and we want only those sequences from a larger (unwrapped!) FASTA.</p>
<pre class="codebg literal-block">
grep --no-group-separator -A1 -F -f <span class="var">headers.txt</span> <span class="var">sequences.fasta</span> &gt; <span class="var">output.fasta</span>
</pre>
<p>By default grep prints only lines that match, the <code class="docutils literal notranslate"><span class="pre">-A1</span></code> adds one line after each match as well (this is why it must be unwrapped!). The <code class="docutils literal notranslate"><span class="pre">-F</span></code> means the search looks for fixed text strings, not RegEx, and the <code class="docutils literal notranslate"><span class="pre">-f</span></code> looks for a list of search terms supplied in a file.</p>
<p>We can of course pipe this with some of our previous examples. Note that the standard input to <code class="docutils literal notranslate"><span class="pre">grep</span></code> is the file to be searched, so we need to redirect standard input elsewhere:</p>
<pre class="codebg literal-block">
grep &quot;Coleoptera&quot; data.csv | cut -d, -f1 | grep --no-group-separator -A1 -f -F /dev/stdin <span class="var">sequences.fasta</span> &gt; <span class="var">output.fasta</span>
</pre>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="keeping_organised.html" class="btn btn-neutral float-right" title="Keeping your data organised" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="loops.html" class="btn btn-neutral float-left" title="Loops in bash" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020, The Vogler Lab.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>