

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>3. Aligning protein coding genes &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/language_data.js"></script>
        <script src="../../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="4. Concatenating alignments" href="concatenating_alignments.html" />
    <link rel="prev" title="2. Extracting Protein Coding Genes" href="extracting_protein_coding_genes.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../otu_trees.html">Making a Barcode Tree</a></li>
<li class="toctree-l2"><a class="reference internal" href="../placing_otus_in_trees.html">Phylogenetic Placement</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../trees.html">Building a Mitogenome Reference Tree</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="preparing_mitogenome_data.html">1. Preparing mitogenome data</a></li>
<li class="toctree-l3"><a class="reference internal" href="extracting_protein_coding_genes.html">2. Extracting Protein Coding Genes</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">3. Aligning protein coding genes</a></li>
<li class="toctree-l3"><a class="reference internal" href="concatenating_alignments.html">4. Concatenating alignments</a></li>
<li class="toctree-l3"><a class="reference internal" href="tree_building.html">5. Tree building</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a> &raquo;</li>
        
          <li><a href="../trees.html">Building a Mitogenome Reference Tree</a> &raquo;</li>
        
      <li>3. Aligning protein coding genes</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../_sources/otu_phylogeny/trees/aligning_protein_coding_genes.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="aligning-protein-coding-genes">
<span id="aligning"></span><h1>3. Aligning protein coding genes<a class="headerlink" href="#aligning-protein-coding-genes" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>The sequences for each gene need to be aligned so that comparable regions are lined up. In phylogenetic terms, this means we need to ensure that the positions used for tree inference are homologous.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a directory of gene sequences in FASTA format, as produced in <a class="reference internal" href="extracting_protein_coding_genes.html#extract-genes"><span class="std std-ref">the previous tutorial</span></a>.</p>
<p>This tutorial uses the <a class="reference internal" href="../../gettingstarted/setup/installing_software.html#mafft"><span class="std std-ref">MAFFT</span></a> software. You should also have an alignment viewer, such as <strong>Aliview</strong>, installed on your personal computer.</p>
</div>
</div>
<div class="section" id="running-alignment">
<h2>Running alignment<a class="headerlink" href="#running-alignment" title="Permalink to this headline">¶</a></h2>
<p>We will do a quick alignment using <strong>MAFFT</strong> for each gene, using the FFT-NS-2 method. This is a fast, but less-accurate method.
First, make a directory to store your aligned genes.</p>
<p>Run the following command, making sure to replace <code class="docutils literal notranslate"><span class="pre">raw</span></code> with the name of the directory you extracted your genes into in the last step, and ​``new​`` with the name of your new directory:</p>
<pre class="literal-block">for f in <span class="var">raw</span>/*;
do
        mafft --thread 1 --retree 2 --maxiterate 0 $f &gt; <span class="var">new</span>/​${f#*/};
done</pre>
<p>Notice how we’ve looped on the extracted gene files, running MAFFT on each file as input and outputting the alignment as a file with the same name, but in new directory.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>While this is running, have a look at the <a class="reference external" href="https://mafft.cbrc.jp/alignment/software/manual/manual.html">manual for MAFFT</a>, in particular the different methods for alignment. Think about what might be best for these genes.</p>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<p>FFT-NS-2 is a fast but inaccurate method; if using your own data, you might want to instead use the <strong>MAFFT</strong> G-INS-i method. We generally would use this method, optimising gap opening and extension penalties for each gene to reduce gap formation. Sometimes the L-INS-I method works better on poorly-aligning genes. But there are no hard and fast rules about which is the most appropriate method in any case, generally the best strategy is to try several methods and use whatever works the best</p>
</div>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Once alignment is complete, pick one of your alignments and download it to your computer. Use an alignment viewer, such as AliView, to open the alignment. If you’re new to viewing alignments, see <a class="reference internal" href="../otu_trees/alignment_viewing.html#alignment-viewing"><span class="std std-ref">here</span></a> for an introduction.</p>
<ul class="simple">
<li><p>How does it look? Is the reading frame maintained?</p></li>
<li><p>Are all of the start and stop codons present and complete for all sequences, are they at the beginning or end of the sequences, and are they aligned properly? What might it mean if they aren’t?</p></li>
<li><p>Are there any sequences that look very out of place. If there were, what could this mean?</p></li>
</ul>
</div>
<p>Sequence alignment is a big topic, beyond the scope of this course. There is a wide range of software out there: two other commonly-used alignment tools are <strong>MUSCLE</strong> and <strong>CLUSTAL</strong>. We find <strong>MAFFT</strong> to be very versatile and fast for all sorts of alignment tasks. If doing this with your own data, you would generally try to optimise the gap opening parameters for each gene in order to generate an optimum alignment, and you might manually edit your alignment. Manual editing is used to remove sequencing errors, correct obvious misalignments and most importantly to check start/stop codons and ensure that codon positions are correctly maintained across the alignment.</p>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>Now that we have an alignment of each of our genes, we can bring this data together by <a class="reference internal" href="concatenating_alignments.html#concat-alignments"><span class="std std-ref">4. Concatenating Alignments</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="concatenating_alignments.html" class="btn btn-neutral float-right" title="4. Concatenating alignments" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extracting_protein_coding_genes.html" class="btn btn-neutral float-left" title="2. Extracting Protein Coding Genes" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>