

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Extension: Phylogenetic delimitation &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/language_data.js"></script>
        <script src="../../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Comparing OTU delimitation methods" href="comparing_otus.html" />
    <link rel="prev" title="Bayesian clustering" href="bayesian.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="greedy.html">Greedy Clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="linkage.html">Linkage-based clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="bayesian.html">Bayesian clustering</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Extension: Phylogenetic delimitation</a></li>
<li class="toctree-l3"><a class="reference internal" href="comparing_otus.html">Comparing OTU delimitation methods</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../mapping_reads.html">Mapping reads</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a> &raquo;</li>
        
          <li><a href="../otudelim.html">OTU Delimitation</a> &raquo;</li>
        
      <li>Extension: Phylogenetic delimitation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../_sources/asv_otu_readmap/otudelim/bptp.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="extension-phylogenetic-delimitation">
<span id="bptp-delim"></span><h1>Extension: Phylogenetic delimitation<a class="headerlink" href="#extension-phylogenetic-delimitation" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a FASTA file comprising unique sequences (ASVs). If you’re following along step-by-step, this was produced at the end of <a class="reference internal" href="../../filtering/chimera_filtering.html#chimera"><span class="std std-ref">the previous section</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">9_mbc_final.fasta</span></code> within the <a class="reference internal" href="../../data.html#sectioncdata"><span class="std std-ref">sectionC archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../../gettingstarted/setup/installing_software.html#fasttree"><span class="std std-ref">**FastTree**</span></a> and <span class="xref std std-ref">**bPTP**</span> software.</p>
</div>
</div>
<div class="section" id="real-datasets">
<h2>Real datasets<a class="headerlink" href="#real-datasets" title="Permalink to this headline">¶</a></h2>
<p>Software for species delimitation of sequences is much more resource-intensive than the methods for OTU delimitation we’ve already seen. Our example dataset is small enough that we can run this on the whole dataset, but many real life datasets might be too large. The best way to utilise phylogenetic delimitation in your pipeline may be to use it as validation for choosing another OTU delimitation method and threshold. In this case, we would suggest the following process:</p>
<ol class="arabic simple">
<li><p>Run OTU delimitation</p></li>
<li><p>Build a phylogenetic tree of your OTUs (see <a class="reference internal" href="../../building_otu_phylogeny.html#phylogeny"><span class="std std-ref">section D: OTU phylogeny</span></a>)</p></li>
<li><p>Identify one or more clades on the tree with around 20-50 OTUs</p></li>
<li><p>For each clade, find the ASVs that form these OTUs.</p></li>
<li><p>For each set of ASVs, build a phylogenetic tree and run phylogenetic delimitation</p></li>
<li><p><a class="reference internal" href="comparing_otus.html#comparing-otus"><span class="std std-ref">Compare the OTU delimitations between the methods</span></a></p></li>
</ol>
<p>Here we will just do step 5.</p>
</div>
<div class="section" id="building-a-tree">
<h2>Building a tree<a class="headerlink" href="#building-a-tree" title="Permalink to this headline">¶</a></h2>
<p>The first thing to do is build a phylogenetic tree of our ASVs. We do this using the <strong>FastTree</strong>. This is a quick and dirty approach, we’ll revisit building OTU phylogeneny in more detail in the <a class="reference internal" href="../../building_otu_phylogeny.html#phylogeny"><span class="std std-ref">Building OTU phylogeny</span></a> section.</p>
<p>First we must strip out our <code class="docutils literal notranslate"><span class="pre">;size=</span></code> annotations, as these often cause issues in newick-format trees. <span class="guilabel">Run the following command, replacing ``input.fasta`</span> with the name of your ASV file`.</p>
<pre class="literal-block">sed -e &quot;s/;size=.*$//&quot; <span class="var">input.fasta</span> &gt; <span class="var">output.fasta</span></pre>
<p><span class="guilabel">Now run FastTree; your input file should be the output from the ``sed`</span> command`.</p>
<pre class="literal-block">FastTree -gtr -nt &lt; <span class="var">input.fasta</span> &gt; <span class="var">output.tre</span></pre>
</div>
<div class="section" id="running-bptp">
<h2>Running bPTP<a class="headerlink" href="#running-bptp" title="Permalink to this headline">¶</a></h2>
<p><strong>bPTP</strong> creates a lot of output files so it’s useful to make a directory for the outputs to keep things tidy.</p>
<p>The <code class="docutils literal notranslate"><span class="pre">-s</span></code> argument sets a seed (a starting number for random number generation in the software). Setting a seed means that performing the same command twice will have the same outcome.</p>
<pre class="literal-block">mkdir bPTP_outputs
bPTP -t <span class="var">input.tre</span> -r -s 1234 -o bPTP_outputs/<span class="var">output</span></pre>
</div>
<div class="section" id="bptp-outputs">
<h2>bPTP outputs<a class="headerlink" href="#bptp-outputs" title="Permalink to this headline">¶</a></h2>
<p>There are a lot of outputs, if you want to know about all of them have a look at the <a class="reference external" href="https://github.com/zhangjiajie/PTP/blob/master/README.md">bPTP documentation</a>. We’ll look at two.</p>
<p>Download the file ending “PTPhSupportPartition.txt.png” to your personal computer and open it. You’ll see a phylogenetic tree of your ASVs, with some branches in red. ASVs linked by red branches have been grouped together into OTUs. Any ASVs with blue branches have formed OTUs by themselves.</p>
<p>Use <code class="docutils literal notranslate"><span class="pre">more</span></code> to look at the file ending <code class="docutils literal notranslate"><span class="pre">PTPhSupportPartition.txt</span></code>. This lists the asvs forming each “Species”, with a support value of each Species group. These values are probabilities, so values closer to 1 are high probabilities.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>How well supported are the OTU clusters? Does it vary?</p></li>
</ul>
</div>
</div>
<div class="section" id="picking-otus">
<h2>Picking OTUs<a class="headerlink" href="#picking-otus" title="Permalink to this headline">¶</a></h2>
<p>Like other methods, bPTP has grouped together ASV sequences into groups we can call OTUs. However, unlike other methods it hasn’t automatically selected a representative sequence for each OTU. Depending on your application, this might not matter. However, many downstream applications such as taxonomic classification require OTU sequences, so we need to select representative sequences after the fact. We will do this based on abundance - for each OTU group we will use the most common ASV as the representative sequence.</p>
<p>First let’s reformat the <code class="docutils literal notranslate"><span class="pre">PTPhSupportPartition.txt</span></code> file a little bit. We want to just extract out the lists of ASVs, dropping all other lines. We then change the lists to space-delimited to make it easier to handle. <span class="guilabel">Run the following command:</span></p>
<pre class="literal-block"># find only lines starting with space, the lists
grep &quot;^ &quot; <span class="var">input.txt</span> |

# remove the spaces and then replace commas with spaces
sed -e &quot;s/ //g&quot; -e &quot;s/,/ /g&quot; &gt; <span class="var">output.txt</span></pre>
<p>Then, for each line we’re going to find the ASV with the largest size. This is a little bit of effort. First we must sort our ASVs into size order. The input to this command must not be the file you stripped <code class="docutils literal notranslate"><span class="pre">;size=</span></code> tags from earlier!</p>
<pre class="literal-block">vsearch --sortbysize <span class="var">input.fasta</span> --output <span class="var">output.fasta</span></pre>
<p>The following command loops through the lines of the reformatted partition summary text file that we just created. For each line, it converts the space-separated list of ASVs into a multi-line string, then searches for those ASVs in the sorted ASVs FASTA we just created. Because this is sorted by size, the most frequent ASV will be the first hit, so we take the first line, remove the <code class="docutils literal notranslate"><span class="pre">&gt;</span></code> at the beginning of the line and output this to a file. <span class="guilabel">Run this command</span>.</p>
<pre class="literal-block">while read l;
do
        grep -Ff &lt;(echo -e &quot;${l// /;\n}&quot;) <span class="var">sortedASVs.fasta</span> |
        head -1 | sed -e &quot;s/&gt;//&quot; ;
done &lt; <span class="var">reformattedpartitionsummary.txt</span> &gt; <span class="var">output.txt</span></pre>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Check the contents of the output file with <code class="docutils literal notranslate"><span class="pre">head</span></code>.</p></li>
<li><p>Use <code class="docutils literal notranslate"><span class="pre">wc</span> <span class="pre">-l</span></code> to make sure the number of lines matches that of the partition summary</p></li>
</ul>
</div>
<p><span class="guilabel">Paste this output together with the reformatted partition table to get our final list of ASVs and OTUs</span></p>
<pre class="literal-block">paste -d$' '<span class="var">representativeASVs.txt</span> <span class="var">reformattedpartitionsummary.txt</span> &gt; <span class="var">output.txt</span></pre>
<p>This output file is the record of ASVs grouped into each OTU, just like we generated in the OTU delimitation methods.</p>
<p>Finally, let’s filter our ASVs to get only the OTU representative sequences. See the bash command <a class="reference internal" href="../../gettingstarted/cli_bioinformatics/cheatsheet.html#cheatsheet"><span class="std std-ref">cheatsheet</span></a> for how this command works. Run this command, making sure to use the version of the ASVs <em>with</em> <code class="docutils literal notranslate"><span class="pre">;size=</span></code> tags.</p>
<pre class="literal-block">perl -pe '$. &gt; 1 and /^&gt;/ ? print &quot;n&quot; : chomp' <span class="var">asvs.fasta</span> |
grep --no-group-separator -A1 -Ff <span class="var">representativeASVs.txt</span> &lt;() &gt; <span class="var">output.fasta</span></pre>
<p>The output file here is analogous to the OTU files generated in the other OTU delimitation methods</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>How does the number of OTUs generated compare with the other methods?</p></li>
<li><p>What might improve the accuracy of this phylogenetic method?</p></li>
</ul>
</div>
<div class="admonition-solution admonition">
<p class="admonition-title">Solution</p>
<p>We have performed a species delimitation method on what may only be a subsample of the true variation within the real species of our dataset. Thus there may not be enough data of both within-species and between-species variation for the <strong>bPTP</strong> algorithm to efficiently distinguish species groups. One way we could improve on this is to add more sequences in that we have <em>a priori</em> knowledge about. We could go to GenBank, search for some well-studied species that are closely related to our metabarcoded community and that have many haplotypes available, download them and add them to our ASVs.</p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We’ve generated a set of OTUs using one method. If you haven’t already, try out other types of OTU clustering: <a class="reference internal" href="greedy.html#greedy-clustering"><span class="std std-ref">greedy clustering</span></a>, <a class="reference internal" href="linkage.html#linkage-delimitation"><span class="std std-ref">linkage delimitation</span></a> or <a class="reference internal" href="bayesian.html#bayesian-clustering"><span class="std std-ref">bayesian clustering</span></a>.</p>
<p>If you’ve now tried all of the methods, the <a class="reference internal" href="comparing_otus.html#comparing-otus"><span class="std std-ref">comparing OTU delimitation methods</span></a> tutorial outlines some ways to compare these methods and some hints for selecting a method and parameters.</p>
<p>Once you have a final set of OTUs and an associated record of ASVs within each OTU, you can proceed with these two files to the <a class="reference internal" href="../mapping_reads.html#mapping-reads"><span class="std std-ref">mapping reads</span></a> subsection to learn about mapping your reads to your OTUs. Later on, we’ll also use the OTU sequences to generate a phylogeny of our OTUs in the <a class="reference internal" href="../../building_otu_phylogeny.html#phylogeny"><span class="std std-ref">Building OTU phylogeny</span></a> section, and to retrieve taxonomic identifications and/or classifications for the OTU sequences in the <a class="reference internal" href="../../otuid.html#otuid"><span class="std std-ref">Identifying OTU sequences</span></a> section.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="comparing_otus.html" class="btn btn-neutral float-right" title="Comparing OTU delimitation methods" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="bayesian.html" class="btn btn-neutral float-left" title="Bayesian clustering" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>