

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Greedy Clustering &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/language_data.js"></script>
        <script src="../../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Linkage-based clustering" href="linkage.html" />
    <link rel="prev" title="OTU Delimitation" href="../otudelim.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a><ul class="current">
<li class="toctree-l3 current"><a class="current reference internal" href="#">Greedy Clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="linkage.html">Linkage-based clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="bayesian.html">Bayesian clustering</a></li>
<li class="toctree-l3"><a class="reference internal" href="bptp.html">Extension: Phylogenetic delimitation</a></li>
<li class="toctree-l3"><a class="reference internal" href="comparing_otus.html">Comparing OTU delimitation methods</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../mapping_reads.html">Mapping reads</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a> &raquo;</li>
        
          <li><a href="../otudelim.html">OTU Delimitation</a> &raquo;</li>
        
      <li>Greedy Clustering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../../_sources/asv_otu_readmap/otudelim/greedy.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="greedy-clustering">
<span id="id1"></span><h1>Greedy Clustering<a class="headerlink" href="#greedy-clustering" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Greedy clustering is the conceptually most simple method of OTU delimitation we will see. In this method, each ASV is examined one-by-one, starting from the most common ASV (based on the frequency of reads of that ASV stored in the <code class="docutils literal notranslate"><span class="pre">;size=</span></code> tag in the sequence headers). The first ASV always forms the centroid sequence of the first OTU; each successive ASV is compared against the current set of centroids, with the ASV joining an OTU cluster if it sufficiently similar to one of the existing centroids, or forming a new centroid if not sufficiently similar to any existing centroids. This method is called “greedy” clustering because in general it tends towards uneven clusters, with the addition of sequences to the largest cluster is preferred when there might be any ambiguity.</p>
<p>The similarity threshold determines how large the clusters are and how many OTUs form. The most common similarity threshold is 97% similarity, where new ASVs join a cluster if they are no more than 3% different from the centroid. This <em>difference</em> is often known as the cluster radius, and this value is often used as much as the similarity value to specify the OTUs. Note that any two ASVs within a cluster may be as much as 6% dissimilar: the key measurement is between the <em>centroid</em> and any ASV, not among ASVs, hence the use of the term “radius”.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a FASTA file comprising unique sequences (ASVs). If you’re following along step-by-step, this was produced at the end of <a class="reference internal" href="../../filtering/chimera_filtering.html#chimera"><span class="std std-ref">the previous section</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">9_mbc_final.fasta</span></code> within the <a class="reference internal" href="../../data.html#sectioncdata"><span class="std std-ref">sectionC archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../../gettingstarted/setup/installing_software.html#vsearch"><span class="std std-ref">VSEARCH</span></a> software.</p>
</div>
</div>
<div class="section" id="running-greedy-clustering">
<h2>Running greedy clustering<a class="headerlink" href="#running-greedy-clustering" title="Permalink to this headline">¶</a></h2>
<p>While the <strong>USEARCH</strong> <strong>UPARSE</strong> pipeline is probably the most commonly used tool for greedy clustering, we will instead once again use <strong>VSEARCH</strong>. <strong>VSEARCH</strong> implements exactly the same clustering algorithm as <strong>USEARCH</strong>, but also allows greater flexibility in outputs and in similarity algorithms available.</p>
<p>The following is the basic command for 97% clustering of our ASVs, using the <strong>USEARCH</strong> algorithm. <span class="guilabel">Run this command, using your ASVs as input and making sure to use a sensible name for the output file.</span> We would suggest including in your output file name the method and the threshold parameter (e.g. <code class="docutils literal notranslate"><span class="pre">otus_greedy_0.97.fasta</span></code>)</p>
<pre class="literal-block">vsearch --cluster_size <span class="var">input.fasta</span> --id 0.97 --centroids <span class="var">output.fasta</span> --sizein --relabel otu</pre>
<p>The <code class="docutils literal notranslate"><span class="pre">--cluster_size</span></code> function specifies that we want to cluster based on the <code class="docutils literal notranslate"><span class="pre">size=</span></code> tags in our input FASTA. The <code class="docutils literal notranslate"><span class="pre">--centroids</span></code> argument tells VSEARCH where to output the centroid sequences, <code class="docutils literal notranslate"><span class="pre">--relabel</span></code>led as <code class="docutils literal notranslate"><span class="pre">otu001</span></code> etc.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>What happens if you modify the <code class="docutils literal notranslate"><span class="pre">--id</span></code> threshold value? Try increasing it and decreasing it, changing the name of your output file as appropriate.</p></li>
<li><p>Can we know what is the most appropriate threshold, or is it somewhat arbitrary based loosely on our understanding of genetic variation in our taxon of interest and what seems to give the right values?</p></li>
</ul>
</div>
<p>If you’re familiar with <a class="reference external" href="http://weizhongli-lab.org/cd-hit/">CD-HIT-EST</a> , this is very similar. <strong>CD-HIT-EST</strong> uses a slightly different calculation of dissimilarity, which can be used by including the parameter <code class="docutils literal notranslate"><span class="pre">--iddef</span> <span class="pre">0</span></code></p>
</div>
<div class="section" id="tracking-our-asvs">
<h2>Tracking our ASVs<a class="headerlink" href="#tracking-our-asvs" title="Permalink to this headline">¶</a></h2>
<p>One common question after performing clustering like this is to ask “which ASVs formed which OTUs?”. <strong>VSEARCH</strong> allows you to find this out by outputting a table recording the matches of each ASV to each cluster.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Rerun the above command, adding <code class="docutils literal notranslate"><span class="pre">--uc</span> <span class="pre">output.uc</span></code> to the end (we suggest changing <code class="docutils literal notranslate"><span class="pre">output.uc</span></code> to a sensible name that also reflects the method and threshold.</p></li>
<li><p>Run <code class="docutils literal notranslate"><span class="pre">more</span></code> on the <code class="docutils literal notranslate"><span class="pre">.uc</span></code> file to view this table, checking the <a class="reference external" href="https://github.com/torognes/vsearch#getting-help">VSEARCH documentation</a> (page 11) to see what each column means.</p></li>
<li><p>Note that it labels the centroids (columnh 2) as 0, 1, 2, etc. rather than otu1, otu2, otu3, etc. We can correct this a little by doing the following:</p></li>
</ul>
<pre class="literal-block"><span class="comment"># Remove the cluster lines</span>
grep -v &quot;^C&quot; <span class="var">input.uc</span> &gt; asvgroups.temp

<span class="comment"># Convert to the correct names</span>
cut -f2 asvgroups.temp | while read l; do echo &quot;otu$(($l + 1))&quot;; done &gt; otus.temp

<span class="comment"># Extract the rest of the table</span>
cut -f3-9 asvgroups.temp &gt; rest.temp

<span class="comment"># Join the table parts together</span>
paste otus.temp rest.temp &gt; <span class="var">output.uc</span>

<span class="comment"># Clean up</span>
rm *.temp</pre>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We’ve generated a set of OTUs using one method. If you haven’t already, try out other types of OTU clustering: <a class="reference internal" href="linkage.html#linkage-delimitation"><span class="std std-ref">linkage delimitation</span></a> or <a class="reference internal" href="bayesian.html#bayesian-clustering"><span class="std std-ref">bayesian clustering</span></a>. You could optionally also look at the <a class="reference internal" href="bptp.html#bptp-delim"><span class="std std-ref">extension: phylogenetic delimitation</span></a>.</p>
<p>If you’ve now tried all of the methods, the <a class="reference internal" href="comparing_otus.html#comparing-otus"><span class="std std-ref">comparing OTU delimitation methods</span></a> tutorial outlines some ways to compare these methods and some hints for selecting a method and parameters.</p>
<p>Once you have a final set of OTUs and an associated record of ASVs within each OTU, you can proceed with these two files to the <a class="reference internal" href="../mapping_reads.html#mapping-reads"><span class="std std-ref">mapping reads</span></a> subsection to learn about mapping your reads to your OTUs. Later on, we’ll also use the OTU sequences to generate a phylogeny of our OTUs in the <a class="reference internal" href="../../building_otu_phylogeny.html#phylogeny"><span class="std std-ref">Building OTU phylogeny</span></a> section, and to retrieve taxonomic identifications and/or classifications for the OTU sequences in the <a class="reference internal" href="../../otuid.html#otuid"><span class="std std-ref">Identifying OTU sequences</span></a> section.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="linkage.html" class="btn btn-neutral float-right" title="Linkage-based clustering" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../otudelim.html" class="btn btn-neutral float-left" title="OTU Delimitation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>