

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>6. Chimera filtering &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="OTU Delimitation" href="../otudelim.html" />
    <link rel="prev" title="5. Point error filtering" href="point_error.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1. Quality Filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2. Not Filtering: Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">4. Indel Filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">6. Chimera filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">Filtering Amplicon Data</a> &raquo;</li>
        
      <li>6. Chimera filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/chimera_filtering.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="chimera-filtering">
<span id="chimera"></span><h1>6. Chimera filtering<a class="headerlink" href="#chimera-filtering" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Chimeras generally arise during PCR when two strands of DNA that are not in fact the correct mirrors of one another become annealed together. This can create ASVs where different parts of the sequence come from different true sources. We remove these with a clever algorithm, UCHIME denovo, that put simply, searches all of the ASVs against one another, looking for subregions that match, not just matches over the complete length of the ASVs. Where a sequence has two or more parts that match to two or more different ASVs <em>that are more common</em>, this sequence is thought to likely be a chimera. Like UNOISE, This algorithm, is implemented in both the USEARCH and VSEARCH software (see the :ref:&lt;denoising section <cite>denoise</cite>&gt; for a discussion about USEARCH and VSEARCH.</p>
</div>
<div class="section" id="performing-chimera-filtering">
<h2>Performing Chimera Filtering<a class="headerlink" href="#performing-chimera-filtering" title="Permalink to this headline">¶</a></h2>
<p>For a complex algorithm, chimera filtering is very easy to implement, as it doesn’t generally require much in the way of parametric tuning (i.e. fiddling with arbitrary thresholds).</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --uchime3_denovo ​in.fasta​ --nonchimeras ​out.fasta
</pre></div>
</div>
<p>It’s that simple, and in the vast majority of cases this will work really well. There is an alternate version of UCHIME that compares against a reference collection rather than doing <em>de novo</em> detection against other ASVs in the dataset. This version may be much more accurate, since rather than assuming that the more frequent ASV is the correct one, it has a set of known correct sequences. However, it requires a very comprehensive reference set, which is rarely possible in metabarcoding. Generally, the assumption that more frequent ASVs are more likely to be correct holds true, and the <em>de novo</em> approach is completely fine.</p>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We have completed our main filtering pipeline, and the ASVs that remain after chimera filtering are our best estimates of the true biological sequences in our dataset. We will be using this file for many of the other sections, so make sure you name it clearly so you can get to it later!</p>
<p>If you want to learn about clustering these sequences into OTUs, you can now go on to the <a class="reference internal" href="../otudelim.html#otu-delim"><span class="std std-ref">OTU delimitation</span></a> section.</p>
<p>If you want to use these ASVs as they are, that is a completely legitimate approach that lets us examine population-level patterns of community ecology. You may want read the discussion and explore the alternative filtering strategy available using metaMATE, examined in this extension: <a class="reference internal" href="metamate.html#metamate"><span class="std std-ref">Extension: metaMATE</span></a>. Whichever set of ASVs you use, you can use these in place of OTUs in the <span class="xref std std-ref">Generating Data</span> and <span class="xref std std-ref">Phylogenetic placement</span> sections.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../otudelim.html" class="btn btn-neutral float-right" title="OTU Delimitation" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="point_error.html" class="btn btn-neutral float-left" title="5. Point error filtering" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>