

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>6. Chimera filtering &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: Stringent ASV filtering" href="metamate.html" />
    <link rel="prev" title="5. Point error filtering" href="point_error.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1. Quality filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2. (not filtering) Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">4. Indel filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">B: Filtering amplicons</a> &raquo;</li>
        
      <li>6. Chimera filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/chimera_filtering.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="chimera-filtering">
<span id="chimera"></span><h1>6. Chimera filtering<a class="headerlink" href="#chimera-filtering" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Chimeras generally arise during PCR when two strands of DNA that are not in fact the correct mirrors of one another become annealed together. This can create ASVs where different parts of the sequence come from different true sources. We remove these with a clever algorithm, <strong>UCHIME denovo</strong>, that (simply) searches all of the ASVs against one another, looking for <em>subregions</em> that match, not just matches over the complete length of the ASVs. Where a sequence has two or more parts that match to two or more different ASVs <em>that are more common</em>, this sequence is thought to likely be a chimera. Like <strong>UNOISE</strong>, this algorithm is implemented in both the <strong>USEARCH</strong> and <strong>VSEARCH</strong> software (see the <a class="reference internal" href="freq_filtering_and_denoising.html#denoising"><span class="std std-ref">denoising tutorial</span></a> for a discussion about <strong>USEARCH</strong> and <strong>VSEARCH</strong>.</p>
<div class="green admonition">
<p class="admonition-title">Software and data</p>
<p>The input data for this tutorial is a FASTA file comprising unique sequences (ASVs). If you’re following along step-by-step, this was produced in <a class="reference internal" href="point_error.html#point-error"><span class="std std-ref">the previous tutorial</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">8_mbc_transpass.fasta</span></code> within the <a class="reference internal" href="../data.html#sectionbdata"><span class="std std-ref">sectionB archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#vsearch"><span class="std std-ref">VSEARCH</span></a> software.</p>
</div>
</div>
<div class="section" id="performing-chimera-filtering">
<h2>Performing chimera filtering<a class="headerlink" href="#performing-chimera-filtering" title="Permalink to this headline">¶</a></h2>
<p>For a complex algorithm, chimera filtering is very easy to implement, as it doesn’t generally require much in the way of parametric tuning (i.e. fiddling with arbitrary thresholds).</p>
<pre class="literal-block">vsearch --uchime3_denovo <span class="var">input.fasta</span> --nonchimeras <span class="var">output.fasta</span></pre>
<p>It’s that simple, and in the vast majority of cases this will work really well. There is an alternate version of <strong>UCHIME</strong> that compares against a reference collection rather than doing <em>de novo</em> detection against other ASVs in the dataset. This version may be much more accurate, since rather than assuming that the more frequent ASV is the correct one, it has a set of known correct sequences. However, it requires a very comprehensive reference set, which is rarely possible in metabarcoding. Generally, the assumption that more frequent ASVs are more likely to be correct holds true, and the <em>de novo</em> approach is completely fine.</p>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We have completed our main filtering pipeline, and the ASVs that remain after chimera filtering are our best estimates of the true biological sequences in our dataset. We will be using this file for many of the other sections, so make sure you name it clearly so you can get to it later.</p>
<p>In the next section, <a class="reference internal" href="../asv_otu_readmap.html#asv-otu-readmap"><span class="std std-ref">ASVs, OTUs and read mapping</span></a> we’ll discuss two of the most common next steps for a set of ASVs: delimiting the ASVs into OTUs, and mapping the reads onto the ASVs or OTUs to get counts of reads per ASV/OTU per sample.</p>
<p>The next section discusses using ASVs without OTU delimitation. This is a completely legitimate approach that lets us examine population-level patterns of community ecology. You may want to read the discussion and explore the alternative filtering strategy available using <strong>metaMATE</strong>, examined in this extension: <a class="reference internal" href="metamate.html#metamate"><span class="std std-ref">Extension: metaMATE</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="metamate.html" class="btn btn-neutral float-right" title="Extension: Stringent ASV filtering" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="point_error.html" class="btn btn-neutral float-left" title="5. Point error filtering" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>