

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>3. Frequency filtering / denoising &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: Denoising by sample" href="extensions/denoising_by_sample.html" />
    <link rel="prev" title="2. (not filtering) Dereplication" href="not_filtering_dereplication.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1. Quality filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2. (not filtering) Dereplication</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">3. Frequency filtering / denoising</a><ul>
<li class="toctree-l3"><a class="reference internal" href="extensions/denoising_by_sample.html">Extension: Denoising by sample</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">4. Indel filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">B: Filtering amplicons</a> &raquo;</li>
        
      <li>3. Frequency filtering / denoising</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/freq_filtering_and_denoising.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="frequency-filtering-denoising">
<span id="denoising"></span><h1>3. Frequency filtering / denoising<a class="headerlink" href="#frequency-filtering-denoising" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Due to the use of PCR to generate metabarcoding datasets, we should generate plenty of copies of each real variant of the amplicon of interest, and thus many reads - depending on sequencing depth, of course. We can therefore be relatively confident that ASVs that occur at very low frequencies in the dataset are more likely to be errors.</p>
<div class="green admonition">
<p class="admonition-title">Software and data</p>
<p>The input data for this tutorial is a FASTA file comprising all of the unique sequences (ASVs) in your dataset. If you’re following along step-by-step, this was produced in <a class="reference internal" href="not_filtering_dereplication.html#dereplication"><span class="std std-ref">the previous tutorial</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">5_mbc_derep.fasta</span></code> within the <a class="reference internal" href="../data.html#sectionbdata"><span class="std std-ref">sectionB archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#vsearch"><span class="std std-ref">VSEARCH</span></a> software.</p>
</div>
</div>
<div class="section" id="simple-frequency-filtering">
<h2>Simple frequency filtering<a class="headerlink" href="#simple-frequency-filtering" title="Permalink to this headline">¶</a></h2>
<p>We can apply a simple threshold, keeping only ASVs that are present at equal to or greater than a given number of reads in the entire dataset. A standard threshold here would be 2, i.e. any singletons will be removed. This is a very straightforward function to apply, once again this could be performed by any number of different software tools but we will use <strong>VSEARCH</strong> because we probably have it installed already. <span class="guilabel">Run the below command</span>, remembering to replace <code class="docutils literal notranslate"><span class="pre">input.fasta</span></code> with the input data (see the software and data box above) and <code class="docutils literal notranslate"><span class="pre">output.fasta</span></code> with a sensible name.</p>
<pre class="literal-block">vsearch --fastx_filter <span class="var">input.fasta</span> --minsize 2 --fastaout <span class="var">output.fasta</span></pre>
<p>This is a straightforward filter, but it doesn’t take into account all of the realities of amplification and sequencing. If an error arose in the later stages of PCR, or occured when sequencing a relatively rare ASV, then it’s reasonable that errors may be infrequent, but if an error arose in the early stages of PCR then it would have been amplified many times.</p>
</div>
<div class="section" id="id1">
<h2>Denoising<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>A much more sophisticated approach to filtering errors is denoising. Denoising algorithms use read frequency and sequence composition to infer likely sequencing errors. ASVs are compared against one another and frequency thresholds are applied relative to counts of similar (but more common) ASVs. Instead of doing the size filtering as part of dereplication, we will instead do it as part of a denoising command. We will use the <a class="reference external" href="https://drive5.com/usearch/manual/unoise_algo.html">**UNOISE3**</a> algorithm, implemented again in <strong>VSEARCH</strong>. <span class="guilabel">Run the following command</span>, remembering to replace <code class="docutils literal notranslate"><span class="pre">input.fasta</span></code> with the input data (see the software and data box above) and <code class="docutils literal notranslate"><span class="pre">output.fasta</span></code> with a sensible name.</p>
<pre class="literal-block">vsearch --cluster_unoise <span class="var">​input.fasta</span> --minsize 2 --unoise_alpha 2 --centroids <span class="var">output.fasta</span></pre>
<p>The key parameter here is the alpha parameter, which determines the threshold level of dissimilarity between frequent and infrequent reads for exclusion of infrequent reads. Note that we’re using a less conservative minsize threshold than the default of 8 because of the smaller size of this example dataset.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>How does the number of sequences filtered by this command compare with the output of the simple abundance threshold run before?</p></li>
<li><p>What is the effect on the number of sequences and size distribution of those sequences of varying the alpha parameter?</p></li>
<li><p>You can get the size distribution by running</p></li>
</ul>
<pre class="literal-block">grep -oP &quot;(?&lt;=size=).*$&quot; <span class="var">input.fasta</span> sort | uniq -c</pre>
<p>In the output of this command, the second column is the count of reads, and the first column is the count of these counts. So for example, if a row says <code class="docutils literal notranslate"><span class="pre">24</span> <span class="pre">12</span></code>, there are 24 ASVs that have 12 reads each.</p>
</div>
</div>
<div class="section" id="selecting-denoising-parameters">
<h2>Selecting denoising parameters<a class="headerlink" href="#selecting-denoising-parameters" title="Permalink to this headline">¶</a></h2>
<p>Denoising removed a large proportion of our ASVs, and this might seem like a large amount of data is being thrown away. However, while this removed a lot of unique sequences, the vast majority of them would have been rarer ASVs, and relative to the total number of reads in the dataset relatively few were discarded. Unfortunately, denoising is one of these steps where we cannot choose thresholds and parameter based on an <em>a priori</em> expectation, the settings must be tuned to the dataset and the research questions.</p>
<p>As a baseline, it is sensible to always discard singletons. Beyond this, if the dataset is large, the research questions are tolerant to the loss of rare real sequences and/or it is important that as high a proportion of ASVs be valid as possible, then stricter settings (higher <code class="docutils literal notranslate"><span class="pre">--minsize</span></code>, lower <code class="docutils literal notranslate"><span class="pre">--unoise_alpha</span></code>) would be sensible. If the dataset is small and the preservation of as many valid ASVs as possible is preferred despite the probably retention of errors, then less strict settings should be used. The important point is that the decision is justified by the research questions and the dataset is sufficiently sequenced as to be resilient to the suitable settings.</p>
</div>
<div class="section" id="alternative-pipelines-and-software">
<h2>Alternative pipelines and software<a class="headerlink" href="#alternative-pipelines-and-software" title="Permalink to this headline">¶</a></h2>
<p>Denoising is crucial to modern metabarcoding pipelines, especially as increased work with Metazoan communities shows the scale of erroneous ASVs present in datasets. This is just one way that denoising can be performed, albeit the most common algorithm used.</p>
<div class="section" id="usearch-vs-vsearch">
<h3>USEARCH vs VSEARCH<a class="headerlink" href="#usearch-vs-vsearch" title="Permalink to this headline">¶</a></h3>
<p>Here we have used the <strong>VSEARCH</strong> version of the <strong>UNOISE</strong> algorithm. The <strong>USEARCH</strong> version may on occasion be more up to date, and also includes chimera filtering (which we will do later). We are using the <strong>VSEARCH</strong> version because it only does denoising and because it’s completely free and open source, whereas <strong>USEARCH</strong> is closed source and the free version has memory limitations. Some versions of the <strong>USEARCH</strong> version also change the output sequence headers in annoying ways that can’t be changed.</p>
<p>We should recognise that Robert Edgar, the author of <strong>USEARCH</strong> and <strong>UNOISE</strong>, built this highly useful algorithm, but making software like this fully and freely available to the research community is an important principle and allows not only more rapid software development but greater validation of the algorithms and wider accessibility of these methods.</p>
</div>
<div class="section" id="denoising-by-sample">
<h3>Denoising by sample<a class="headerlink" href="#denoising-by-sample" title="Permalink to this headline">¶</a></h3>
<p>The standard usage of <strong>UNOISE</strong> works on the entire dataset, thus considers the frequency of ASVs relative to one another over the entire dataset. However, individual samples within a dataset are rarely evenly sequenced, therefore some ASVs may be rarer because they occur in undersequenced samples, not because they are errors. It may be more appropriate to denoise on the level of individual samples. If you’d like to try this out, you can try out this extension: <cite>Extension: Denoising by sample &lt;denoise_by_sample&gt;</cite>. Note it may take a little while so you may choose to come back to this later.</p>
</div>
<div class="section" id="other-software">
<h3>Other software<a class="headerlink" href="#other-software" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://github.com/lh3/bfc">BFC</a> is a fairly common denoiser, often used more in the genomic sequencing realm than in amplicon sequencing, but it is occasionally used in metabarcoding. <a class="reference external" href="https://benjjneb.github.io/dada2/index.html">dada2</a> has recently gained in popularity among metabarcoders. Both of these algorithms work on an earlier stage of the pipeline, on the reads including quality data. In our experience, these algorithms tend to be much more strict than <strong>UNOISE</strong>, which may well be appropriate for some questions. We’re not exploring them here because their approach wraps up many of the steps we’re learning one-by-one, thereby skipping over a lot of the points we try to illustrate in these resources. However, we certainly do not recommend against them; <strong>dada2</strong> has <a class="reference external" href="https://benjjneb.github.io/dada2/tutorial.html">an excellent tutorial</a> that in our experience is straightforward to follow and produces reasonable outputs, albeit conservative. We would note that if you try out <strong>dada2</strong> for a coding region such as CO1, length and translation filtering should be performed on the denoised outputs.</p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>You should have produced many denoised outputs while you were playing around with the thresholds. We suggest you clean these up and proceed using the output from the command given above, i.e. with <code class="docutils literal notranslate"><span class="pre">--minsize</span> <span class="pre">2</span> <span class="pre">--unoise_alpha</span> <span class="pre">2</span></code>. This file contains a more filtered set of ASVs than we started with, but there are still other sources of error we can address, starting with <a class="reference internal" href="indel_filtering.html#indel-filtering"><span class="std std-ref">4. Indel filtering</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="extensions/denoising_by_sample.html" class="btn btn-neutral float-right" title="Extension: Denoising by sample" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="not_filtering_dereplication.html" class="btn btn-neutral float-left" title="2. (not filtering) Dereplication" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>