

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>4) Frequency filtering / denoising &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="5) Point error filtering" href="point_error.html" />
    <link rel="prev" title="Extension: What if we have a more variable region?" href="extensions/more_variable_region.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1) Quality Filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2) Not Filtering: Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">3) Indel Filtering</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">4) Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5) Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6) Chimera filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">Filtering Amplicon Data</a> &raquo;</li>
        
      <li>4) Frequency filtering / denoising</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/freq_filtering_and_denoising.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="frequency-filtering-denoising">
<span id="denoising"></span><h1>4) Frequency filtering / denoising<a class="headerlink" href="#frequency-filtering-denoising" title="Permalink to this headline">¶</a></h1>
<p>Due to the use of PCR to generate metabarcoding datasets, we should generate plenty of copies of the amplicon of interest, and thus many reads - depending on sequencing depth, of course. We can therefore be relatively confident that sequences that occur at very low frequencies in the dataset are more likely to be errors.</p>
<p>These can be filtered out with a simple threshold, although the selection of this threshold is likely to be dataset-dependent.</p>
<p>We can actually do this as part of the dereplication command we just did. Try running the command from the last section again, with the same input and adding the argument <code class="docutils literal notranslate"><span class="pre">minuniquesize</span> <span class="pre">1</span></code> ​to get rid of all singleton sequences <a class="footnote-reference brackets" href="#f1" id="id1">1</a>. <strong>​Use a different output filename​</strong>, we’re not going to keep this output.</p>
<p>A much more sophisticated approach to filtering errors is denoising. Denoising algorithms use read frequency and sequence composition to infer likely sequencing errors. Instead of doing the size filtering as part of dereplication, we will instead do it as part of a denoising command. We will use the unoise3 algorithm, implemented again in VSEARCH. Your input file here should be the output from dereplicating in the last section.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --cluster_unoise ​in.fasta​ --minsize <span class="m">4</span> --unoise_alpha <span class="m">2</span> --centroids out.fasta
</pre></div>
</div>
<p>The key parameter here is the alpha parameter, which determines the threshold level of dissimilarity between frequent and infrequent reads for exclusion of infrequent reads. Note that we’re using a less conservative minsize threshold than the default of 8 because of the smaller size of our dataset.</p>
<ul class="simple">
<li><p>What is the effect on the number of sequences and size distribution of those sequences of varying the alpha parameter? You can get the size distribution by running</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep <span class="s2">&quot;^&gt;&quot;</span> ​in.fasta​ <span class="p">|</span> sed -e <span class="s2">&quot;s/size=\([^;]\)/\1/&quot;</span> <span class="p">|</span> sort <span class="p">|</span> uniq -c
</pre></div>
</div>
<p>Some researchers argue that denoising should be run at the level of the individual sample, not the dataset as a whole, because the frequency of reads is only meaningful relative to individual pools of amplicons. What do you think? If you’ve got the time, come back to this section later and do the following:</p>
<ul class="simple">
<li><p>Run <code class="docutils literal notranslate"><span class="pre">vsearch</span> <span class="pre">--fastx_filter</span></code> on each separate sample fastq file using a loop</p></li>
<li><p>Run dereplication on each sample fasta separately using a loop</p></li>
<li><p>Run denoising on each sample fasta separately using a loop</p></li>
<li><p>Concatenate the results using sed and re-run dereplication using the following command:</p></li>
</ul>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --derep_fulllength ​in.fasta --sizein --sizeout --relabel uniq --output out.fasta
</pre></div>
</div>
<ul class="simple">
<li><p>Compare the total unique read numbers and size distribution to the version produced earlier</p></li>
</ul>
<p>You’re now ready to move onto the <a class="reference internal" href="point_error.html#point-error"><span class="std std-ref">next step.</span></a></p>
<p class="rubric">Footnotes</p>
<dl class="footnote brackets">
<dt class="label" id="f1"><span class="brackets"><a class="fn-backref" href="#id1">1</a></span></dt>
<dd><p><code class="docutils literal notranslate"><span class="pre">$</span> <span class="pre">vsearch</span> <span class="pre">--derep_fulllength</span> <span class="pre">​in.fasta​</span> <span class="pre">--output</span> <span class="pre">​out.fasta​</span> <span class="pre">--sizeout</span> <span class="pre">--relabel</span> <span class="pre">uniq</span> <span class="pre">--minuniquesize</span> <span class="pre">1</span></code></p>
</dd>
</dl>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="point_error.html" class="btn btn-neutral float-right" title="5) Point error filtering" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extensions/more_variable_region.html" class="btn btn-neutral float-left" title="Extension: What if we have a more variable region?" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>