

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>3) Indel Filtering &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/customcss/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: What if we have a more variable region?" href="extensions/more_variable_region.html" />
    <link rel="prev" title="2) Not Filtering: Dereplication" href="not_filtering_dereplication.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1) Quality Filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2) Not Filtering: Dereplication</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">3) Indel Filtering</a><ul>
<li class="toctree-l3"><a class="reference internal" href="extensions/more_variable_region.html">Extension: What if we have a more variable region?</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">4) Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5) Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6) Chimera filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">Filtering Amplicon Data</a> &raquo;</li>
        
      <li>3) Indel Filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/indel_filtering.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="indel-filtering">
<span id="id1"></span><h1>3) Indel Filtering<a class="headerlink" href="#indel-filtering" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<p>In this case, I use the term indel to mean insertions or deletions from PCR or sequencing errors rather than natural mutations. The fundamental assumption here is that the sequenced region is sufficiently conserved that there will not be any naturally-occuring indels sequenced, because these would have been deleterious and the organism would not have survived to have been sampled.</p>
<p>Thus, you should think carefully about how to apply this type of filtering to your own data depending on the barcode region used. Insertions or deletions are easy to spot because they will change the length of the sequence from what is expected based on the primers. While filtering based on length primarily removes indels, it can also be used to remove other reads that are clearly erroneous for other reasons.</p>
<p>Before we start, let’s double-check the length distribution of our reads. We can do this using a command we used before, having adapted the command for fastas (where the sequences are every other line, rather than every 4 lines):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -n <span class="s1">&#39;2~2p&#39;</span> ​file​ <span class="p">|</span> <span class="k">while</span> <span class="nb">read</span> l<span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> <span class="si">${#</span><span class="nv">l</span><span class="si">}</span> <span class="p">;</span> <span class="k">done</span> <span class="p">|</span> sort <span class="p">|</span> uniq -c
</pre></div>
</div>
<p>Oh dear, what’s happened to our reads? Check the first 10 lines of the fasta:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ head -n <span class="m">10</span> ​file
</pre></div>
</div>
<p>VSEARCH, although it’s great in many respects, outputs files in wrapped format, which means it starts a new line after 80 sequence characters. While this is nicer to look at, this is a pain for using quick-and-easy tools to summarise data on the linux command line. So we must run a quick command first to unwrap this data:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ perl -pe <span class="s1">&#39;$. &gt; 1 and /^&gt;/ ? print &quot;\n&quot; : chomp&#39;</span> ​infile​ &gt; ​outfile
</pre></div>
</div>
<p>Use the output from this in the sed command above.</p>
<p>If we have a very variable region, we might not want to do any filtering at all, or we may know a reasonable range of lengths within which we expect our reads to fall. There are lots of tools for length filtering; we’ll use VSEARCH again - in fact, the <code class="docutils literal notranslate"><span class="pre">​--fastx_filter</span></code> command again. Let’s try filtering with quite a wide range:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --fastx_filter ​in.fasta --fastq_minlen <span class="m">400</span> --fastq_maxlen <span class="m">440</span> --fastaout out.fasta​
</pre></div>
</div>
<p>In this case, the region of CO1 we use is sufficiently conserved that, on balance of probabilities, any insertions or deletions are due to PCR and/or sequencing errors, and/or maybe errors with the pair merging we did, rather than natural mutations. So, if you have a strict length expectation for your reads, you can exclude any reads longer or shorter than this value.</p>
<ul class="simple">
<li><p>Run the filtering again, this time allowing no variants from our target length of 418bp</p></li>
</ul>
<p>If you want to learn about coding regions that expect no single-base indels but is overall more variable and could have established whole codon changes then go to this <a class="reference internal" href="extensions/more_variable_region.html#more-variable"><span class="std std-ref">extension task</span></a>.
Alternatively, go on to the <a class="reference internal" href="freq_filtering_and_denoising.html#denoising"><span class="std std-ref">Frequency filtering / denoising section</span></a>.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="extensions/more_variable_region.html" class="btn btn-neutral float-right" title="Extension: What if we have a more variable region?" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="not_filtering_dereplication.html" class="btn btn-neutral float-left" title="2) Not Filtering: Dereplication" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>