

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>4. Indel Filtering &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: What if we have a more variable region?" href="extensions/more_variable_region.html" />
    <link rel="prev" title="Extension: Denoising by sample" href="extensions/denoising_by_sample.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1. Quality Filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2. Not Filtering: Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">4. Indel Filtering</a><ul>
<li class="toctree-l3"><a class="reference internal" href="extensions/more_variable_region.html">Extension: What if we have a more variable region?</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otu_asv_read.html">OTUs, ASVs and Read Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">Building OTU Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">Filtering Amplicon Data</a> &raquo;</li>
        
      <li>4. Indel Filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/indel_filtering.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="indel-filtering">
<span id="id1"></span><h1>4. Indel Filtering<a class="headerlink" href="#indel-filtering" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>In metabarcoding, we amplify a known regoin of the genome and thus we have an expectation about the length of the resulting amplicons. Insertions or deletions (indels) are changes to the DNA sequence that will affect the number of bases between two known points in the genome. While this term is commonly used to refer to real genetic mutations, we use it here to refer to insertions or deletions caused by PCR or sequencing errors. It is important to note that we cannot necessarily distinguish between true indel mutations and erroneous mutations. However, as we are operating on a protein coding sequence, we can assume that the vast majority of true indels are likely to be deleterious and therefore we would not observe these in our dataset drawn from living organisms. We can then draw conclusions about the validity of ASVs based on their length relative to the expected amplicon length.</p>
<p>So, put simply, insertions or deletions are easy to spot because they will change the length of the sequence from what is expected based on the primers. While filtering based on length primarily removes indels, it can also be used to remove other reads that are clearly erroneous for other reasons.</p>
<div class="admonition-important-note admonition">
<p class="admonition-title">Important note</p>
<p>You should think carefully about how to apply this type of filtering to your own data depending on the barcode region used. Protein coding markers will have relatively little length variation, but some may still exist depending on the marker used and the taxonomic breadth of your study. On the other hand, non protein coding markers will be substantially more variable and this must be taken into account to avoid removing substantial portions of your dataset.</p>
</div>
</div>
<div class="section" id="length-distributions">
<h2>Length distributions<a class="headerlink" href="#length-distributions" title="Permalink to this headline">¶</a></h2>
<p>Before we start, let’s double-check the length distribution of our reads. Learn more about the <code class="docutils literal notranslate"><span class="pre">sed</span></code> command <a class="reference internal" href="../linux_subsections/sed.html#sed"><span class="std std-ref">here</span></a> - note we’ve adapted this command for FASTAs where the sequences are every other line, rather than every 4 lines):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -n <span class="s1">&#39;2~2p&#39;</span> ​file​ <span class="p">|</span> <span class="k">while</span> <span class="nb">read</span> l<span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> <span class="si">${#</span><span class="nv">l</span><span class="si">}</span> <span class="p">;</span> <span class="k">done</span> <span class="p">|</span> sort <span class="p">|</span> uniq -c
</pre></div>
</div>
<p>Oh dear, what’s happened to our reads? Check the first 10 lines of the fasta:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ head -n <span class="m">10</span> ​file
</pre></div>
</div>
<p>VSEARCH, although it’s great in many respects, outputs files in wrapped format, which means it starts a new line after 80 sequence characters. While this is nicer to look at, this is a pain for using quick-and-easy tools to summarise data on the linux command line. So we must run a quick command first to unwrap this data:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ perl -pe <span class="s1">&#39;$. &gt; 1 and /^&gt;/ ? print &quot;\n&quot; : chomp&#39;</span> ​infile​ &gt; ​outfile
</pre></div>
</div>
<p>Use the output from this in the sed command above.</p>
</div>
<div class="section" id="length-filtering">
<h2>Length filtering<a class="headerlink" href="#length-filtering" title="Permalink to this headline">¶</a></h2>
<p>If we have a very variable region, we might not want to do any filtering at all, or we may know a reasonable range of lengths within which we expect our reads to fall. There are lots of tools for length filtering; we’ll use VSEARCH again - in fact, the <code class="docutils literal notranslate"><span class="pre">​--fastx_filter</span></code> command again. Let’s try filtering with quite a wide range:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --fastx_filter ​in.fasta --fastq_minlen <span class="m">400</span> --fastq_maxlen <span class="m">440</span> --fastaout out.fasta​
</pre></div>
</div>
<p>In this case, the region of CO1 we use is sufficiently conserved that, on balance of probabilities, any insertions or deletions are due to PCR and/or sequencing errors, and/or maybe errors with the pair merging we did, rather than natural mutations. So, if you have a strict length expectation for your reads, you can exclude any reads longer or shorter than this value.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Run the filtering again, this time allowing no variants from our target length of 418bp</p>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --fastx_filter ​in.fasta --fastq_minlen <span class="m">418</span> --fastq_maxlen <span class="m">418</span> --fastaout out.fasta​
</pre></div>
</div>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>In our case, we have a very conserved region so we will use the output from the last command, i.e. only ASVs of 418bp long, for the next steps. Next, we will remove errors by checking the translation: <a class="reference internal" href="point_error.html#point-error"><span class="std std-ref">5. Point Error Filtering :ref:</span></a>.</p>
<p>In some cases, we may be metabarcoding a coding locus but our project includes a wide taxonomic breadth or the region is more variable. Where you might expect length variation, but only in terms of whole codons, we have to do something a little more complicated. If you want, you can try this out in this extension: <a class="reference internal" href="extensions/more_variable_region.html#more-variable"><span class="std std-ref">Extension: Variable length coding regions</span></a>. You can use either output for the next steps.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="extensions/more_variable_region.html" class="btn btn-neutral float-right" title="Extension: What if we have a more variable region?" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extensions/denoising_by_sample.html" class="btn btn-neutral float-left" title="Extension: Denoising by sample" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>