

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>2. (not filtering) Dereplication &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="3. Frequency filtering / denoising" href="freq_filtering_and_denoising.html" />
    <link rel="prev" title="Fundamentals: Phred Scores" href="fundamentals/phred_scores.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1. Quality filtering</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">2. (not filtering) Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">4. Indel filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">B: Filtering amplicons</a> &raquo;</li>
        
      <li>2. (not filtering) Dereplication</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/not_filtering_dereplication.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="not-filtering-dereplication">
<span id="dereplication"></span><h1>2. (not filtering) Dereplication<a class="headerlink" href="#not-filtering-dereplication" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Our concatenated FASTA file contains all putatively valid reads for our project. Now that we have used and discarded the quality data, we can now go through all of these reads and find all of the different unique sequences that are represented in our project. This stage is called dereplication, because we drop all the replicate reads, although we do keep a record of how many reads each unique sequence has in the dataset.</p>
<p>From now on, the main purpose is to find these unique sequences and filter out those that may be erroneous. For now, we ignore the source of these sequences; the dataset is considered as a whole with (largely) no reference to the individual samples. There is no need for this filtering process to be performed on many thousands of reads if lots of them are identical; better to dereplicate and reduce our data size to speed up what are generally quite computationally intensive tasks.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a FASTA file comprising all of the reads in your dataset. If you’re following along step-by-step, this was produced <a class="reference internal" href="quality_filtering.html#quality-filtering"><span class="std std-ref">the previous tutorial</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">4_mbc_concat.fasta</span></code> within the <a class="reference internal" href="../data.html#sectionbdata"><span class="std std-ref">sectionB archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#vsearch"><span class="std std-ref">VSEARCH</span></a> software.</p>
</div>
</div>
<div class="section" id="performing-dereplication">
<h2>Performing dereplication<a class="headerlink" href="#performing-dereplication" title="Permalink to this headline">¶</a></h2>
<p>Dereplication is a pretty straightforward task - there are very few parameters to worry about. Therefore choosing a tool for dereplication has few considerations, and is often a matter of convenience. We will use the dereplication function within <strong>VSEARCH</strong>, because we probably already <strong>VSEARCH</strong> installed for other steps.</p>
<pre class="literal-block">vsearch --derep_fulllength ​:var:<cite>input.fasta</cite> --output <span class="var">output.fasta</span> --sizeout --relabel uniq</pre>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Run the above command, remembering to replace <code class="docutils literal notranslate"><span class="pre">input.fasta</span></code> with the input file (see the Data and software box above), and <code class="docutils literal notranslate"><span class="pre">output.fasta</span></code> with a sensible name.</p></li>
<li><p>Use <code class="docutils literal notranslate"><span class="pre">head</span></code> on the output file to see how the sequence headers are formatted.</p></li>
<li><p>Use <code class="docutils literal notranslate"><span class="pre">grep</span></code> on the output file to count up how many unique sequences we have.</p></li>
</ul>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">--sizeout</span></code> option is very important here. This records in the header of each unique sequence how many copies it has across the entire dataset. This piece of information is used by several downstream processes. We also relabeled the sequences with <code class="docutils literal notranslate"><span class="pre">uniq</span></code> and an arbitrary serial number. This isn’t necessary, and the name we are using is arbitrary, but in our opinion this is a useful step to distinguish individual reads from the set of unique putatively biological sequences.</p>
</div>
<div class="section" id="what-are-these-sequences">
<h2>What are these sequences?<a class="headerlink" href="#what-are-these-sequences" title="Permalink to this headline">¶</a></h2>
<p>At this point, we have a set of unique amplicon sequences recovered from our metabarcoding project. It is not <em>all</em> of the unique sequences, we discarded many that lacked indices or primers, wouldn’t merge or were poor quality, but we are confident that this set contains all of our real amplicons, although we suspect it probably still contains many errors. This set of sequences is sometimes known as Amplicon Sequence Variants (ASVs) or Exact Sequence Variants (ESVs). These putatively represent all unique variants of the CO1 locus in our project metacommunity, so could be considered to be the set of haplotypes or unique biological variants in our study. We will refer to these as ASVs from now on.</p>
<p>For now, though, we know that there are highly likely to be errors present in our ASVs, so the next steps will try different methods to remove these errors and increase the proportion of our ASVs that are true biological sequences of our locus of interest.</p>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>Next we will use the “size” information (counts of occurence of each ASV in the whole dataset) as a basis to filter out possible errors through <a class="reference internal" href="freq_filtering_and_denoising.html#denoising"><span class="std std-ref">3. Frequency filtering</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="freq_filtering_and_denoising.html" class="btn btn-neutral float-right" title="3. Frequency filtering / denoising" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="fundamentals/phred_scores.html" class="btn btn-neutral float-left" title="Fundamentals: Phred Scores" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>