

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>5. Point error filtering &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="6. Chimera filtering" href="chimera_filtering.html" />
    <link rel="prev" title="Extension: What if we have a more variable region?" href="extensions/more_variable_region.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="quality_filtering.html">1. Quality filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2. (not filtering) Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">4. Indel filtering</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">5. Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">B: Filtering amplicons</a> &raquo;</li>
        
      <li>5. Point error filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/point_error.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="point-error-filtering">
<span id="point-error"></span><h1>5. Point error filtering<a class="headerlink" href="#point-error-filtering" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Filtering by length will remove sequences that have one or more PCR/sequencer-caused insertions or one or more deletions, however in some cases these errors may cancel one another out; or alternatively, PCR or sequencing may induce the equivalent of point mutations, where a single base is misread. Similarly, noncoding gene variants such as NUMTs or pseudogenes may actually have point mutations in comparison to the ‘true’ region.</p>
<p>We can identify some point errors because they will alter the translation of the genetic code in such a way that it becomes meaningless - if the barcode region is a coding region, of course. The obvious error is the introduction of stop codons into the translation. By translating all of our sequences and checking for stop codons, we can easily reject these errors or variants.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a FASTA file comprising unique sequences (ASVs). If you’re following along step-by-step, this was produced in <a class="reference internal" href="indel_filtering.html#indel-filtering"><span class="std std-ref">the previous tutorial</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">7_mbc_indelfil.fasta</span></code> within the <a class="reference internal" href="../data.html#sectionbdata"><span class="std std-ref">sectionB archive</span></a> can be used as example data.</p>
<p>This tutorial uses an accessory function, <strong>filtertranslate</strong>, from the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#metamate-install"><span class="std std-ref">metaMATE</span></a> software.</p>
</div>
</div>
<div class="section" id="filtering-by-translation">
<h2>Filtering by translation<a class="headerlink" href="#filtering-by-translation" title="Permalink to this headline">¶</a></h2>
<p>Check the helpfile for this script by running:</p>
<div class="codebg highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">filtertranslate</span> <span class="o">-</span><span class="n">help</span>
</pre></div>
</div>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Figure out what the command is to run <code class="docutils literal notranslate"><span class="pre">filtertranslate</span></code> with all of the following options:</p></li>
</ul>
<ol class="arabic simple">
<li><p>using automatic reading frame detection</p></li>
<li><p>outputting both succeeding and failing sequences in separate files</p></li>
</ol>
<ul class="simple">
<li><p>Hint: check the usage line to figure out where some of the arguments go. Don’t forget, our samples are arthropods.</p></li>
</ul>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<pre class="codebg literal-block">filtertranslate -i <span class="var">input.fasta</span> -t 5 -y separate -o <span class="var">output</span></pre>
</div>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Have a look at the failed file.</p></li>
<li><p>Go to an online amino acid translator (e.g. <a class="reference external" href="https://web.expasy.org/translate/">here</a>) and paste in a sequence. Make sure to set the correct genetic code.</p></li>
<li><p>See what the translation looks like. Frame 2 is the correct frame.</p></li>
<li><p>Can you see the stop(s)?</p></li>
</ul>
</div>
<p>Other ‘point errors’ that do not cause stops are harder to spot. Some will not affect coding at all, which is impossible to distinguish from natural variation. The majority will affect coding, but again distinguishing these from natural variation is very hard. One possibility is to look at the structure of the translated protein and see if it’s realistic, but there aren’t currently any tools that can do this…</p>
<div class="sidebar">
<p class="sidebar-title">Alternatives</p>
<p>The tool <strong>filtertranslate</strong> is part of the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#metamate-install"><span class="std std-ref">metaMATE</span></a> software, which you can find more about <a class="reference internal" href="metamate.html#metamate"><span class="std std-ref">in a later extension tutorial</span></a>. This is a tool we wrote, but we don’t know of any other software that performs simple translation filtering like this.</p>
<p>Most commonly, we see other metabarcoders either performing this step manually, by translating the sequences in some GUI sequence viewer like Geneious and manually checking for stops, or by using <a class="reference external" href="https://bioweb.supagro.inra.fr/macse/">MACSE</a>. MACSE is a very neat piece of software that performs alignment of nucleotide sequences against references using the translation, and is very good at finding frameshifts. However it’s somewhat complicated to implement and most seriously is not very efficient, scaling very poorly for larger amplicon datasets.</p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We now have a set of ASVs that are all of the correct length, with a lot of errors hopefully removed. The final thing to do is remove chimeras, which we will do in the next tutorial: <a class="reference internal" href="chimera_filtering.html#chimera"><span class="std std-ref">6. Chimera filtering</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="chimera_filtering.html" class="btn btn-neutral float-right" title="6. Chimera filtering" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extensions/more_variable_region.html" class="btn btn-neutral float-left" title="Extension: What if we have a more variable region?" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>