

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>1. Quality filtering &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Fundamentals: Phred Scores" href="fundamentals/phred_scores.html" />
    <link rel="prev" title="B: Filtering amplicons" href="../filtering.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">1. Quality filtering</a><ul>
<li class="toctree-l3"><a class="reference internal" href="fundamentals/phred_scores.html">Fundamentals: Phred Scores</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2. (not filtering) Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">4. Indel filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">B: Filtering amplicons</a> &raquo;</li>
        
      <li>1. Quality filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/quality_filtering.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="quality-filtering">
<span id="id1"></span><h1>1. Quality filtering<a class="headerlink" href="#quality-filtering" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Quality filtering aims to remove sequences that contain sequencing errors, as determined by the sequencer’s own quality scoring method. These scores are stored in the FASTQ file using phred scores. If you are unfamiliar with what phred scores are, have a look <a class="reference internal" href="fundamentals/phred_scores.html#phred-scores"><span class="std std-ref">here</span></a> first.</p>
<p>As you can imagine, there are many ways that phred scores can be used to filter sequences. The three common ones are:</p>
<ul class="simple">
<li><p>Excluding reads with any scores below a threshold</p></li>
<li><p>Trimming ends from reads based on a threshold</p></li>
<li><p>Excluding reads based on number or rate of expected errors</p></li>
</ul>
<p>All three are commonly used in amplicon filtering.</p>
<p>With paired amplicons, we know that the end of the read is likely to be lower quality; however, we have already merged our pairs with the result that if our reads have any low-quality portions, it’s the <em>middle</em> of the read, so selective trimming at this point is not a viable strategy. Excluding reads entirely based on quality is a more conservative approach and is generally suggested here.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a FASTQ file comprising all of the reads in your dataset, with the source sample for each read noted in the read’s header. If you worked through <a class="reference internal" href="../readprocessing.html#read-processing"><span class="std std-ref">the previous section</span></a>, this was produced during the <a class="reference internal" href="../processing/data_concat.html#data-concat"><span class="std std-ref">concatenation</span></a> tutorial. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">4_mbc_concat.fastq</span></code> file within the <a class="reference internal" href="../data.html#sectionbdata"><span class="std std-ref">sectionB archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#fastx-toolkit"><span class="std std-ref">fastx_toolkit</span></a> and <a class="reference internal" href="../gettingstarted/setup/installing_software.html#vsearch"><span class="std std-ref">VSEARCH</span></a> software.</p>
</div>
</div>
<div class="section" id="quality-threshold-filtering">
<h2>Quality threshold filtering<a class="headerlink" href="#quality-threshold-filtering" title="Permalink to this headline">¶</a></h2>
<p>For basic fastq filtering based on minimum score, we will use <strong>fastq_quality_filter</strong> from the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#fastx-toolkit"><span class="std std-ref">fastx_toolkit</span></a> package.</p>
<p>This simple tool alows us to remove any reads that contain any bases with a quality score lower than the threshold we set. For example, let’s keep only reads where 100% of the bases have a quality score greater than or equal to 13 (the equivalent of a 0.05 probability that that base is incorrect). The <code class="docutils literal notranslate"><span class="pre">-v</span></code> option reports the results to terminal for us. As always, replace <code class="docutils literal notranslate"><span class="pre">input.fastq</span></code> with the name of the file you’re filtering (see the Data and software box above) and <code class="docutils literal notranslate"><span class="pre">output.fasta</span></code> with a sensible name to call the output.</p>
<pre class="literal-block">fastq_quality_filter -v -q 13 -p 100 -i <span class="var">input.fastq</span> -o <span class="var">output.fastq</span></pre>
<p>We can additionally specify a proportion of the bases that must meet our threshold using <code class="docutils literal notranslate"><span class="pre">-p</span></code>. Following is another command that will keep only reads where 60% or more of the bases have a quality score equal to or greater than 30 (p = 0.001):</p>
<pre class="literal-block">fastq_quality_filter -v -q 30 -p 60 -i <span class="var">input.fastq</span> -o <span class="var">output.fastq</span></pre>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Make a note of your filtering thresholds and the number of reads they returned.</p></li>
<li><p>Try out some different thresholds, keeping notes on the numbers that were rejected or retained.</p></li>
<li><p>Is there a way to select these thresholds without just making an arbitrary choice?</p></li>
</ul>
</div>
</div>
<div class="section" id="quality-rate-filtering">
<h2>Quality rate filtering<a class="headerlink" href="#quality-rate-filtering" title="Permalink to this headline">¶</a></h2>
<p>Rather than setting per-base quality thresholds, another approach is to analyse all of the error probabilities for a read and calculate the overall probability of the read having errors. For this, we will use the <strong>fastx_filter</strong> function from the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#vsearch"><span class="std std-ref">VSEARCH</span></a> package for filtering by expected error and expected error rate.</p>
<p>Given that the quality scores are just a set of probabilities, we can calculate the expected number of errors in the read as a whole. The following command will exclude any reads that have one or more expected errors. Note the difference with the previous section: we don’t necessarily ascertain if any particular base is incorrect, we simply calculate how many bases would, on average, be incorrect on a read with this set of quality scores.</p>
<pre class="literal-block">vsearch --fastx_filter <span class="var">input.fastq</span> --fastq_maxee 1 --fastaout <span class="var">output.fasta</span></pre>
<p>As you can imagine, the longer the read, the higher the chance that there is at least one error, simply by chance alone. So instead of basing our threshold on the total number of expected errors, we could base our threshold on the rate of errors. The following command will exclude any reads with more than 0.1 expected errors per base:</p>
<pre class="literal-block">vsearch --fastx_filter <span class="var">input.fastq</span> --fastq_maxee_rate 0.1 --fastaout <span class="var">output.fasta</span></pre>
</div>
<div class="section" id="choosing-quality-filtering-thresholds">
<h2>Choosing quality filtering thresholds<a class="headerlink" href="#choosing-quality-filtering-thresholds" title="Permalink to this headline">¶</a></h2>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Compare the number of reads returned with different filters. If you use <code class="docutils literal notranslate"><span class="pre">​grep​</span></code> for this, note that <code class="docutils literal notranslate"><span class="pre">fastq_quality_filter</span></code> returns fastq files, but vsearch returns fastas. The regex for fastas should be <code class="docutils literal notranslate"><span class="pre">&quot;^&gt;&quot;</span></code></p></li>
<li><p>Can you adjust the filters to get roughly the same number of reads filtered out using the different methods? Is this meaningful?</p></li>
<li><p>Do you think these are the same reads that are being filtered each time?</p></li>
</ul>
</div>
<p>Which quality filtering parameter to pick? Well, it depends partly on the nature of the data, partly on the aim of your filtering, and partly on what feels right to you.</p>
<p>In our opinion, filtering based on the number of expected errors makes sense: there is a logical basis for the selection of the threshold, the removal of reads based on their overall likelihood of error, not some relatively arbitrary threshold of minimum quality score. While obviously this value increases with the length of the read, so could be argued isn’t a comparable value between different fragment lengths, my argument would be that it’s a reflection of the reality of sequencing, and that no matter how long my fragment is, we don’t want any errors. So we generally filter using <code class="docutils literal notranslate"><span class="pre">​--fastq_maxee</span> <span class="pre">1</span></code> . If we suspect from later examination that we still have a lot of sequencing errors, we’ll reduce this to <code class="docutils literal notranslate"><span class="pre">--fastq_maxee</span> <span class="pre">0.5</span></code>. If in the very rare case we’re simply not getting enough sequences returned, we might loosen this to <code class="docutils literal notranslate"><span class="pre">-​-fastq_maxee</span> <span class="pre">1.5</span></code> or even <code class="docutils literal notranslate"><span class="pre">2</span></code>,​ but generally this data isn’t really trustworthy.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Based on your comparison of different filtering outputs, select whichever method you feel is most appropriate. Delete the rest.</p></li>
<li><p>The selected output file will be the file used for the next step.</p></li>
</ul>
<p>If you pick the output of <code class="docutils literal notranslate"><span class="pre">fastq_quality_filter</span></code> , you will need to convert this output to fasta using:</p>
<pre class="literal-block">fastq_to_fasta -i <span class="var">input.fastq</span> -o <span class="var">output.fasta</span></pre>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We now have a FASTA file of all of our reads. This is an important file to retain, so make sure you keep a note of it. These reads will later be mapped to our final biological sequences to assess their occurence. Now that we have removed the quality data, we can now undertake dereplication in the next section: <a class="reference internal" href="not_filtering_dereplication.html#dereplication"><span class="std std-ref">2. Dereplication</span></a>.</p>
<div class="green admonition">
<p class="admonition-title">Note</p>
<p>Here we present quality filtering after <a class="reference internal" href="../processing/data_concat.html#data-concat"><span class="std std-ref">concatenation</span></a> because it allows us to separate two distinct themes: the processing of raw sequence reads into samples, in section A, and the filtering of processed sequence reads to remove errors, in section B.</p>
<p>However, when performing metabarcoding bioinformatics it may be more efficient to perform this quality filtering earlier, as it substantially reduces the data size. If you wanted to do this, you’d need to run quality filtering in a loop on all of your individual sample files. For example, if you wanted to quality filter your <a class="reference internal" href="../processing/primer_removal.html#primer-removal"><span class="std std-ref">primer-trimmed</span></a> using <strong>VSEARCH</strong>, you’d run the following, replacing <code class="docutils literal notranslate"><span class="pre">input</span></code> with the name of the directory containing the files you want to quality filter, and <code class="docutils literal notranslate"><span class="pre">output</span></code> with the sensible name of a directory to place the filtered files in.</p>
<pre class="literal-block">mkdir <span class="var">output</span>
samples=$(ls <span class="var">input</span>/ | cut -d_ -f1 | sort | uniq)
for s in $samples;
do
        vsearch --fastx_filter <span class="var">input</span>/$s.fastq --fastq_maxee 1 --fastaout <span class="var">output</span>/$s.fasta;
done</pre>
<p>You would then perform concatenation on the FASTA files as described in <a class="reference internal" href="../processing/data_concat.html#data-concat"><span class="std std-ref">that tutorial</span></a></p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="fundamentals/phred_scores.html" class="btn btn-neutral float-right" title="Fundamentals: Phred Scores" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../filtering.html" class="btn btn-neutral float-left" title="B: Filtering amplicons" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>