

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>1) Quality Filtering &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/customjs/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Fundamentals: Phred Scores" href="fundamentals/phred_scores.html" />
    <link rel="prev" title="Filtering Amplicon Data" href="../filtering.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">1) Quality Filtering</a><ul>
<li class="toctree-l3"><a class="reference internal" href="fundamentals/phred_scores.html">Fundamentals: Phred Scores</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="not_filtering_dereplication.html">2) Not Filtering: Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="indel_filtering.html">3) Indel Filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="freq_filtering_and_denoising.html">4) Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="point_error.html">5) Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="chimera_filtering.html">6) Chimera filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../filtering.html">Filtering Amplicon Data</a> &raquo;</li>
        
      <li>1) Quality Filtering</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/filtering/quality_filtering.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="quality-filtering">
<span id="id1"></span><h1>1) Quality Filtering<a class="headerlink" href="#quality-filtering" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<p>Quality filtering aims to remove sequences that contain sequencing errors. If you are unfamiliar with what phred scores are, have a look <a class="reference internal" href="fundamentals/phred_scores.html#phred-scores"><span class="std std-ref">here</span></a> first.</p>
<p>As you can imagine, there are many ways that Phred scores can be used to filter sequences. The three common ones are:</p>
<ul class="simple">
<li><p>Excluding reads with any scores below a threshold</p></li>
<li><p>Trimming ends from reads based on a threshold</p></li>
<li><p>Excluding reads based on number or rate of expected errors</p></li>
</ul>
<p>All three are commonly used in amplicon filtering.</p>
<p>With paired amplicons, we know that the end of the read is likely to be lower quality; but we have already merged our pairs. The quality scores in the middle of our reads have been adjusted to take account of the duplicate data, so in general it’s unlikely we have any specific poor regions of our reads. Excluding reads entirely based on quality is a more conservative approach and is generally suggested here.</p>
<p>For basic fastq filtering based on minimum score, we will use fastq_quality_filter from the fastx_toolkit package. This is a great little package of handy tools available <a class="reference external" href="http://hannonlab.cshl.edu/fastx_toolkit/index.html">here</a></p>
<p>We will use the fastx_filter function from the VSEARCH software for filtering by expected error and expected error rate. VSEARCH is a software package specifically designed for metabarcoding, based on the USEARCH package but completely free and open source. We’ll see other tools from this useful package later; you can read the documentation <a class="reference external" href="https://github.com/torognes/vsearch">here.</a></p>
<p>Let’s try doing some different sorts of quality filtering. Here are some commands.</p>
<p>Exclude any reads with quality score lower than 13 (probability of error ≈ 0.05):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ fastq_quality_filter -q <span class="m">13</span> -p <span class="m">100</span> -i ​in.fastq​ -o ​out.fastq
</pre></div>
</div>
<p>Exclude any reads with fewer than 60% of bases with a quality score equal to or greater than 30 (p =
0.001):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ fastq_quality_filter -q <span class="m">30</span> -p <span class="m">60</span> -i ​in.fastq​ -o ​out.fastq
</pre></div>
</div>
<p>Exclude any reads with more than 1 expected errors over the entire read:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --fastx_filter ​in.fastq​ --fastq_maxee <span class="m">1</span> --fastaout ​out.fasta
</pre></div>
</div>
<p>Exclude any reads with more than 0.1 expected errors per base:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ vsearch --fastx_filter ​in.fastq​ --fastq_maxee_rate <span class="m">0</span>.1 --fastaout ​out.fasta
</pre></div>
</div>
<ul class="simple">
<li><p>Compare the number of reads returned with different filters using <code class="docutils literal notranslate"><span class="pre">​grep​</span></code>. Note that <code class="docutils literal notranslate"><span class="pre">fastq_quality_filter</span></code> returns fastq files, but vsearch returns fastas. The regex for fastas should be “^&gt;”.</p></li>
<li><p>Can you adjust the filters to get roughly the same number of reads filtered out using the different methods?</p></li>
<li><p>Do you think these are the same reads that are being filtered each time?</p></li>
</ul>
<p>Which quality filtering parameter to pick? Well, it depends partly on the nature of the data, partly on the aim of your filtering, and partly on what feels right to you.</p>
<p><em>In my opinion, filtering based on the number of expected errors makes sense: there is a logical basis for the selection of the threshold, the removal of reads based on their overall likelihood of error, not some relatively arbitrary threshold of minimum quality score. While obviously this value increases with the length of the read, so could be argued isn’t a comparable value between different fragment lengths, my argument would be that it’s a reflection of the reality of sequencing, and that no matter how long my fragment is, I don’t want any errors. So I generally filter using</em> <code class="docutils literal notranslate"><span class="pre">​--fastq_maxee</span> <span class="pre">1</span></code> ​. <em>If I suspect from later examination that I still have a lot of sequencing errors, I’ll reduce this to</em> <code class="docutils literal notranslate"><span class="pre">--fastq_maxee</span> <span class="pre">0.5</span></code> . <em>If in the very rare case I’m simply not getting enough sequences returned, I might loosen this to</em> <code class="docutils literal notranslate"><span class="pre">-​-fastq_maxee</span> <span class="pre">1.5</span></code> <em>or even</em> <code class="docutils literal notranslate"><span class="pre">2</span></code> <em>,​ but generally this data isn’t really trustworthy.</em></p>
<ul class="simple">
<li><p>Keep whichever one of your filtered fastas you like best. Delete the rest. This file will be the file used for the next step.</p></li>
</ul>
<p>If you pick the output of <code class="docutils literal notranslate"><span class="pre">fastq_quality_filter</span></code> , you will need to convert this output to fasta using:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ fastq_to_fasta -i ​in.fastq​ -o ​out.fasta
</pre></div>
</div>
<p>You can now move onto the next section on <a class="reference internal" href="not_filtering_dereplication.html#dereplication"><span class="std std-ref">dereplication</span></a>.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="fundamentals/phred_scores.html" class="btn btn-neutral float-right" title="Fundamentals: Phred Scores" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../filtering.html" class="btn btn-neutral float-left" title="Filtering Amplicon Data" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>