

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Cheatsheet of useful commands &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/customcss/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Read processing" href="../readprocessing.html" />
    <link rel="prev" title="Loops in bash" href="loops.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="UNIX_cmd_line.html">UNIX command line</a></li>
<li class="toctree-l2"><a class="reference internal" href="basic_UNIX.html">Basic UNIX commands</a></li>
<li class="toctree-l2"><a class="reference internal" href="linux_servers.html">Accessing linux servers</a></li>
<li class="toctree-l2"><a class="reference internal" href="nano.html">Using Nano to edit documents</a></li>
<li class="toctree-l2"><a class="reference internal" href="grep.html">The grep command</a></li>
<li class="toctree-l2"><a class="reference internal" href="sed.html">The sed command</a></li>
<li class="toctree-l2"><a class="reference internal" href="loops.html">Loops in bash</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Cheatsheet of useful commands</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../linuxintro.html">Command line bioinformatics</a> &raquo;</li>
        
      <li>Cheatsheet of useful commands</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/linux_subsections/cheatsheet.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="cheatsheet-of-useful-commands">
<h1>Cheatsheet of useful commands<a class="headerlink" href="#cheatsheet-of-useful-commands" title="Permalink to this headline">¶</a></h1>
<p>Here are some useful bash commands for doing command-line bioinformatics. These draw upon some of the basics described in other pages in this section.</p>
<div class="section" id="unwrapping-fasta-files">
<h2>Unwrapping FASTA files<a class="headerlink" href="#unwrapping-fasta-files" title="Permalink to this headline">¶</a></h2>
<p>A lot of software outputs wrapped FASTA files, where sequence lines are truncated at a certain number of characters and continued on the following line, like this:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">sequence1</span>
<span class="n">ATACGACGATAAGCAGACTAGACATATT</span>
<span class="n">CGCCCGATTGACCG</span>
<span class="o">&gt;</span><span class="n">sequence2</span>
<span class="n">ACGCGCTGAAAAGCAGACTTTTTACACG</span>
<span class="n">CGCGTAAACAACGC</span>
</pre></div>
</div>
<p>This can be really annoying for quick and easy command line bash bioinformatics, because the number of lines for each sequence is unpredictable. The following perl one-liner, from <a class="reference external" href="https://www.biostars.org/p/9262/#118460">this biostars answer</a>, easily unwraps a fasta file:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ perl -pe <span class="s1">&#39;$. &gt; 1 and /^&gt;/ ? print &quot;\n&quot; : chomp&#39;</span> input.fasta &gt; output.fasta
</pre></div>
</div>
<p>This command is great because it can easily be used within a pipe.</p>
</div>
<div class="section" id="replacing-bases">
<h2>Replacing bases<a class="headerlink" href="#replacing-bases" title="Permalink to this headline">¶</a></h2>
<p>You may sometimes have sequence data that isn’t quite appropriate for a piece of software, for example you have an alignment but the software won’t accept it, or you have ambiguity codes but the software only accepts A, T, C, G or N. You can use <code class="docutils literal notranslate"><span class="pre">sed</span></code> to remove or replace all offending bases, but you must be careful only to do so on non-header lines - you don’t want to edit your sequence headers by accident! We do this by adding a line specifier, <code class="docutils literal notranslate"><span class="pre">/^[^&gt;]/,</span></code> before the substitution command. This specifies that we want to only operate on lines that do not start with <code class="docutils literal notranslate"><span class="pre">&gt;</span></code>.</p>
<p>Degap an alignment (i.e. de-align):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -e <span class="s2">&quot;/^[^&gt;]/,s/-//g&quot;</span> input.fasta &gt; output.fasta
</pre></div>
</div>
<p>Replace any ambiguities with <code class="docutils literal notranslate"><span class="pre">N</span></code>:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -e <span class="s2">&quot;/^[^&gt;]/,s/[YRWSKMDVHBX]/N/g&quot;</span> input.fasta &gt; output.fasta
</pre></div>
</div>
</div>
<div class="section" id="getting-sequence-lengths">
<h2>Getting sequence lengths<a class="headerlink" href="#getting-sequence-lengths" title="Permalink to this headline">¶</a></h2>
<p>Here we delete header lines, delete gaps then count how many characters each line has.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -e <span class="s2">&quot;/^&gt;/d&quot;</span> -e <span class="s2">&quot;s/-//g&quot;</span> input.fasta <span class="p">|</span> awk <span class="s1">&#39;{ print length }&#39;</span> &gt; lengths.txt
</pre></div>
</div>
<p>The FASTA file must be unwrapped - if it’s not, you could add the perl one-liner above to the beginning of the pipe.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ perl -pe <span class="s1">&#39;$. &gt; 1 and /^&gt;/ ? print &quot;\n&quot; : chomp&#39;</span> input.fasta <span class="p">|</span> sed -e <span class="s2">&quot;/^&gt;/d&quot;</span> -e <span class="s2">&quot;s/-//g&quot;</span> <span class="p">|</span> awk <span class="s1">&#39;{ print length }&#39;</span> &gt; lengths.txt
</pre></div>
</div>
<p>What to with these lengths? Well, we can get a rough idea of the distribution by counting the number of each unique length:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -e <span class="s2">&quot;/^&gt;/d&quot;</span> -e <span class="s2">&quot;s/-//g&quot;</span> input.fasta <span class="p">|</span> awk <span class="s1">&#39;{ print length }&#39;</span> <span class="p">|</span> sort <span class="p">|</span> uniq -c <span class="p">|</span> sort -hr
</pre></div>
</div>
<p>Or we could use an R one-liner to count the mean and standard deviation</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -e <span class="s2">&quot;/^&gt;/d&quot;</span> -e <span class="s2">&quot;s/-//g&quot;</span> input.fasta <span class="p">|</span> awk <span class="s1">&#39;{ print length }&#39;</span> <span class="p">|</span> Rscript -e <span class="s1">&#39;d&lt;-scan(&quot;stdin&quot;, quiet=TRUE);print(c(round(mean(d),0), round(100 * mean(c(mean(d)-min(d),max(d)-mean(d)))/mean(d), 1)));&#39;</span>
</pre></div>
</div>
</div>
<div class="section" id="generating-a-random-fasta">
<h2>Generating a random FASTA<a class="headerlink" href="#generating-a-random-fasta" title="Permalink to this headline">¶</a></h2>
<p>Sometimes we want some dummy data to test on. This command makes 30 sequences, each 300 bases long:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ paste -d <span class="s1">&#39;\n&#39;</span> &lt;<span class="o">(</span><span class="k">for</span> i in <span class="o">{</span><span class="m">01</span>..30<span class="o">}</span><span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> <span class="s2">&quot;&gt;seq</span><span class="nv">$i</span><span class="s2">&quot;</span><span class="p">;</span> <span class="k">done</span><span class="o">)</span> &lt;<span class="o">(</span> cat /dev/urandom <span class="p">|</span> tr -dc <span class="s1">&#39;ATCG&#39;</span> <span class="p">|</span> fold -w <span class="m">300</span> <span class="p">|</span> head -n <span class="m">30</span> <span class="o">)</span>
</pre></div>
</div>
</div>
<div class="section" id="get-the-headers-of-a-fasta">
<h2>Get the headers of a FASTA<a class="headerlink" href="#get-the-headers-of-a-fasta" title="Permalink to this headline">¶</a></h2>
<p>There are two ways to do this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep <span class="s2">&quot;^&gt;&quot;</span> input.fasta <span class="p">|</span> sed -e <span class="s2">&quot;s/&gt;//&quot;</span> &gt; headers.txt
$ grep -oP <span class="s2">&quot;(?&lt;=^&gt;).*</span>$<span class="s2">&quot;</span> &gt; headers.txt
</pre></div>
</div>
<p>The second one uses a fancy Regular Expression term called a “positive lookbehind”.</p>
</div>
<div class="section" id="extract-information-from-tables">
<h2>Extract information from tables<a class="headerlink" href="#extract-information-from-tables" title="Permalink to this headline">¶</a></h2>
<p>Say I have a table, data.csv, that looks like this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>seq1,Coleoptera,425
seq2,Diptera,256
seq3,Hemiptera,786
</pre></div>
</div>
<p>If I wanted to find the sequence names that are Coleoptera, I could do this:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep <span class="s2">&quot;Coleoptera&quot;</span> data.csv <span class="p">|</span> cut -d, -f1 &gt; headers.txt
</pre></div>
</div>
<p>Or the sequences where column three is greater than 500:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ awk -F<span class="s1">&#39;,&#39;</span> <span class="s1">&#39; $3 &gt; 500 &#39;</span> data.csv <span class="p">|</span> cut -d, -f1 &gt; headers.txt
</pre></div>
</div>
<p>The functions <code class="docutils literal notranslate"><span class="pre">awk</span></code> or <code class="docutils literal notranslate"><span class="pre">grep</span></code> do the searching, while <code class="docutils literal notranslate"><span class="pre">cut</span></code> extracts the first column from the result. If your table is tab-delimited, you can skip the <code class="docutils literal notranslate"><span class="pre">-F</span></code> argument in <code class="docutils literal notranslate"><span class="pre">awk</span></code> and the <code class="docutils literal notranslate"><span class="pre">-d</span></code> argument in <code class="docutils literal notranslate"><span class="pre">cut</span></code>, tab-delimited is the default for both.</p>
</div>
<div class="section" id="extract-sequence-by-header">
<h2>Extract sequence by header<a class="headerlink" href="#extract-sequence-by-header" title="Permalink to this headline">¶</a></h2>
<p>Say we have a list of sequence headers in headers.txt and we want only those sequences from a larger (unwrapped!) fasta called sequences.fasta.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep --no-group-separator -A1 -f -F headers.txt sequences.fasta &gt; output.fasta
</pre></div>
</div>
<p>By default grep prints only lines that match, the <code class="docutils literal notranslate"><span class="pre">-A1</span></code> adds one line after each match as well (this is why it must be unwrapped!). The <code class="docutils literal notranslate"><span class="pre">-f</span></code> means the search looks for fixed text strings, not RegEx, and the <code class="docutils literal notranslate"><span class="pre">-F</span></code> looks for a list of search terms supplied in a file.</p>
<p>We can of course pipe this with some of our previous examples. Note that the standard input to <code class="docutils literal notranslate"><span class="pre">grep</span></code> is the file to be searched, so we need to redirect standard input elsewhere:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep <span class="s2">&quot;Coleoptera&quot;</span> data.csv <span class="p">|</span> cut -d, -f1 <span class="p">|</span> grep --no-group-separator -A1 -f -F /dev/stdin sequences.fasta &gt; output.fasta
</pre></div>
</div>
</div>
<div class="section" id="aliasing-frequent-commands">
<h2>Aliasing frequent commands<a class="headerlink" href="#aliasing-frequent-commands" title="Permalink to this headline">¶</a></h2>
<p>Some commands I use a lot, but are a little fiddly to write, so I save them under an <a class="reference external" href="https://tldp.org/LDP/abs/html/aliases.html">alias</a> to my bash profile. This is like a shortcut to a command. A prime candidate for this is the FASTA unwrapping perl one-liner.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">echo</span> <span class="s2">&quot;alias unwrap=\&quot;perl -pe &#39;</span>$<span class="s2">. &gt; 1 and /^&gt;/ ? print \&quot;\n\&quot; : chomp&#39;\&quot;&quot;</span> &gt;&gt; ~/.bashrc
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">&gt;&gt;</span></code> is very important: don’t use a single <code class="docutils literal notranslate"><span class="pre">&gt;</span></code>. You might need to change <code class="docutils literal notranslate"><span class="pre">.bashrc</span></code> to <code class="docutils literal notranslate"><span class="pre">.bash_profile</span></code> or something else depending on your system.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../readprocessing.html" class="btn btn-neutral float-right" title="Read processing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="loops.html" class="btn btn-neutral float-left" title="Loops in bash" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>