.. _annotation_resources: 

====================================
Annotation resources
====================================

This is as far as we’ll take the assembly steps. The next stage after identifying these contigs is to find and annotate the mitochondrial genes, but this is beyond the scope of this workshop. If this is something you’re interested in learning about, some easy starting points for automated annotation are the `MITOS web server <http://mitos.bioinf.uni-leipzig.de/index.py>`_ and the `MitoZ annotate script <https://github.com/linzhi2013/MitoZ>`_. The latter works well but is still in development and can be tricky to set up and get to work properly.

We have generated complete, annotated versions of these novel mitogenomes, and these will be used in the next :ref:`session to build phylogenetic trees<trees>`.

