

<!DOCTYPE html>
<html class="writer-html4" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Software installation &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/css/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../../_static/jquery.js"></script>
        <script type="text/javascript" src="../../_static/underscore.js"></script>
        <script type="text/javascript" src="../../_static/doctools.js"></script>
        <script type="text/javascript" src="../../_static/language_data.js"></script>
        <script type="text/javascript" src="../../_static/js/custom.js"></script>
        <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="Accessing linux servers" href="../remote_access.html" />
    <link rel="prev" title="Using your own machine" href="using_your_own_machine.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="../../gettingstarted.html">Getting started</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="../setup.html">Setting up a bioinformatics machine</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="cloud_servers.html">Cloud servers</a></li>
<li class="toctree-l3"><a class="reference internal" href="using_your_own_machine.html">Using your own machine</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Software installation</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../remote_access.html">Accessing linux servers</a></li>
<li class="toctree-l2"><a class="reference internal" href="../how_to_use.html">How to use these resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="../toolbox_pipelines.html">The bioinformatics toolbox</a></li>
<li class="toctree-l2"><a class="reference internal" href="../cli_bioinformatics.html">Command line bioinformatics</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../gettingstarted.html">Getting started</a> &raquo;</li>
        
          <li><a href="../setup.html">Setting up a bioinformatics machine</a> &raquo;</li>
        
      <li>Software installation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../_sources/gettingstarted/setup/installing_software.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="software-installation">
<span id="installing-software"></span><h1>Software installation<a class="headerlink" href="#software-installation" title="Permalink to this headline">¶</a></h1>
<p>Here we detail how to install all Linux command-line software used in these resources. Note that if you are setting up a server from scratch, for example if you are using a <a class="reference internal" href="cloud_servers.html#cloud-servers"><span class="std std-ref">cloud server</span></a>, you should also ensure that you perform any necessary administration tasks and install and configure any necessary security software you need.</p>
<p>These commands are written for Ubuntu Linux. They will probably work fine on any Ubuntu or Debian machine, and on most Linux machines with the appropriate package manager changes.</p>
<p>We supply these instructions with absolutely no guarantee or warranty. If you break a machine following these instructions, it’s not our fault. If you fail to properly secure a cloud server and have it compromised and used to spam people or launch crippling viruses, it’s not our fault. If in doubt, seek expert advice.</p>
<p>These commands use the most recent software versions as of December 2020, where the code points to a specific version you should make sure to look up the most recent version and edit accordingly.</p>
<div class="section" id="setup">
<h2>Setup<a class="headerlink" href="#setup" title="Permalink to this headline">¶</a></h2>
<p>The following commands install some general purpose Linux tools and some libraries that are required for installing one or more software tools detailed below.</p>
<pre class="codebg literal-block">
<span class="comment"># Set up directories</span>
sudo mkdir /usr/local/deb/

<span class="comment"># Update installed packages</span>
sudo apt update &amp;&amp; sudo apt upgrade -y

<span class="comment"># Install some useful tools</span>
sudo apt install rename zip unzip -y

<span class="comment"># Install tools for building source packages</span>
sudo apt install build-essential pkg-config autoconf -y

<span class="comment"># Install useful language packages</span>
sudo apt install python3-pip perl-doc default-jre -y
</pre>
<div class="admonition warning">
<p class="first admonition-title">Warning</p>
<p class="last">You should make sure to run all of the above commands before installing any software in this section.</p>
</div>
<div class="section" id="alan">
<span id="id1"></span><h3>ALAN<a class="headerlink" href="#alan" title="Permalink to this headline">¶</a></h3>
<p>Commandline alignment viewer, github <a class="reference external" href="https://github.com/mpdunne/alan">here</a>.</p>
<pre class="codebg literal-block">
sudo wget -P /usr/local/bin/ https://raw.githubusercontent.com/mpdunne/alan/master/alan
sudo chmod a+x /usr/local/bin/alan
</pre>
</div>
<div class="section" id="blast">
<span id="id2"></span><h3>BLAST<a class="headerlink" href="#blast" title="Permalink to this headline">¶</a></h3>
<p>Widely used sequence searching software, documentation and installation instructions <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK279690/">here</a>.</p>
<pre class="codebg literal-block">
cd /usr/local/src/

BLAST_VERSION=&quot;2.11.0&quot; <span class="comment"># Change as appropriate</span>
sudo wget &quot;ftp://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/${BLAST_VERSION}/ncbi-blast-${BLAST_VERSION}+-x64-linux.tar.gz&quot;
sudo tar -zxf ncbi-blast-${BLAST_VERSION}+-x64-linux.tar.gz
sudo cp ncbi-blast-${BLAST_VERSION}+/bin/* /usr/local/bin/
</pre>
</div>
<div class="section" id="bptp">
<span id="id4"></span><h3>bPTP<a class="headerlink" href="#bptp" title="Permalink to this headline">¶</a></h3>
<p>Bayesian phylogeny-based species delimitation, github <a class="reference external" href="https://github.com/zhangjiajie/PTP">here</a>.</p>
<pre class="codebg literal-block">
cd /usr/local/src
sudo git clone https://github.com/zhangjiajie/PTP
cd PTP
sudo -H python3 -m pip install -r requirements.txt
cd /PTP/bin
<span class="comment"># Patch the python version</span>
for f in *.py; do sed -i &quot;1s/python$/python3/&quot; $f; done
<span class="comment"># Patch for running on headless server</span>
sudo sed -i -e &quot;14s/^$/nos.environ['QT_QPA_PLATFORM'] = 'offscreen'n/&quot; bPTP.py
sudo ln -s /usr/local/src/PTP/bin/bPTP.py /usr/local/bin/bPTP
sudo ln -s /usr/local/src/PTP/bin/PTP.py /usr/local/bin/PTP
</pre>
</div>
<div class="section" id="catfasta2phyml">
<span id="id6"></span><h3>catfasta2phyml<a class="headerlink" href="#catfasta2phyml" title="Permalink to this headline">¶</a></h3>
<p>Script for concatenating alignments into a supermatrix, github <a class="reference external" href="https://github.com/nylander/catfasta2phyml">here</a>.</p>
<pre class="codebg literal-block">
sudo wget -O /usr/local/bin/ https://raw.githubusercontent.com/nylander/catfasta2phyml/master/catfasta2phyml.pl
sudo chmod a+x /usr/local/bin/catfasta2phyml.pl
</pre>
<p>Installation not necessary: you could simply download it to your working directory and run it using <code class="docutils literal notranslate"><span class="pre">perl</span></code>:</p>
<pre class="codebg literal-block">
wget /usr/local/bin/ https://raw.githubusercontent.com/nylander/catfasta2phyml/master/catfasta2phyml.pl
perl catfasta2phyml.pl
</pre>
</div>
<div class="section" id="crop">
<span id="id8"></span><h3>CROP<a class="headerlink" href="#crop" title="Permalink to this headline">¶</a></h3>
<p>Bayesian OTU delimitation, github <a class="reference external" href="https://github.com/tingchenlab/CROP">here</a>.</p>
<pre class="codebg literal-block">
sudo apt install libgsl-dev

cd /usr/local/src
sudo git clone https://github.com/zhangjiajie/PTP
cd PTP
sudo -H python3 -m pip install -r requirements.txt
sudo -H python3 -m pip install python-nexus
sudo sed -i &quot;1s/python$/python3/&quot; bin/bPTP.py
sudo ln -s /usr/local/src/PTP/bin/bPTP.py /usr/local/bin/bPTP
</pre>
</div>
<div class="section" id="cutadapt">
<span id="id10"></span><h3>Cutadapt<a class="headerlink" href="#cutadapt" title="Permalink to this headline">¶</a></h3>
<p>Versatile adapter sequence removal, documentation <a class="reference external" href="https://cutadapt.readthedocs.io/en/stable/">here</a>.</p>
<div class="codebg highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">sudo</span> <span class="o">-</span><span class="n">H</span> <span class="n">python3</span> <span class="o">-</span><span class="n">m</span> <span class="n">pip</span> <span class="n">install</span> <span class="n">cutadapt</span>
</pre></div>
</div>
</div>
<div class="section" id="extract-genes">
<span id="id12"></span><h3>extract_genes<a class="headerlink" href="#extract-genes" title="Permalink to this headline">¶</a></h3>
<p>Script for extracting individual gene sequences from GenBank files, github <a class="reference external" href="https://github.com/tjcreedy/biotools">here</a>.</p>
<pre class="codebg literal-block">
sudo apt install bioperl libarray-utils-perl

sudo wget -O /usr/local/bin/ https://raw.githubusercontent.com/tjcreedy/biotools/master/extract_genes.pl
sudo chmod a+x /usr/local/bin/extract_genes.pl
</pre>
<p>Installation not necessary: you coul simply download it to your working directory and run it using <code class="docutils literal notranslate"><span class="pre">perl</span></code>:</p>
<pre class="codebg literal-block">
wget https://raw.githubusercontent.com/tjcreedy/biotools/master/extract_genes.pl
perl extract_genes.pl
</pre>
</div>
<div class="section" id="fasttree">
<span id="id14"></span><h3>FastTree<a class="headerlink" href="#fasttree" title="Permalink to this headline">¶</a></h3>
<p>Fast maximum likelihood phylogenetic inference, website <a class="reference external" href="http://www.microbesonline.org/fasttree/">here</a>.</p>
<pre class="codebg literal-block">
cd /usr/local/bin/
sudo wget http://www.microbesonline.org/fasttree/FastTree
sudo wget http://www.microbesonline.org/fasttree/FastTreeMP
sudo chmod a+x FastTree*
</pre>
</div>
<div class="section" id="fastqc">
<span id="id16"></span><h3>FastQC<a class="headerlink" href="#fastqc" title="Permalink to this headline">¶</a></h3>
<p>Raw sequence quality assessment, website <a class="reference external" href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">here</a>.</p>
<pre class="codebg literal-block">
cd /usr/local/src/
sudo wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip
sudo unzip fastqc_v0.11.9.zip
sudo chmod a+x FastQC/fastqc
sudo ln -s /usr/local/src/FastQC/fastqc /usr/local/bin/fastqc
</pre>
</div>
<div class="section" id="fastx-toolkit">
<span id="id18"></span><h3>FASTX Toolkit<a class="headerlink" href="#fastx-toolkit" title="Permalink to this headline">¶</a></h3>
<p>Various handy tools for dealing with FASTA and FASTQ files. A little out of date though, requires patching to work in modern Ubuntu (all done in the below commands). Website <a class="reference external" href="http://hannonlab.cshl.edu/fastx_toolkit/index.html">here</a>.</p>
<pre class="codebg literal-block">
cd /usr/local/src/

sudo wget https://github.com/agordon/libgtextutils/releases/download/0.7/libgtextutils-0.7.tar.gz
sudo tar -xzf libgtextutils-0.7.tar.gz
cd libgtextutils-0.7
sudo sed -i '47s/input_stream/static_cast&lt;bool&gt;(input_stream)/' src/gtextutils/text_line_reader.cpp
sudo ./configure
sudo make
sudo make install
cd ../

sudo wget https://github.com/agordon/fastx_toolkit/releases/download/0.0.14/fastx_toolkit-0.0.14.tar.bz2
sudo tar -xjf fastx_toolkit-0.0.14.tar.bz2
cd fastx_toolkit-0.0.14
sudo wget https://github.com/agordon/fastx_toolkit/files/1182724/fastx-toolkit-gcc7-patch.txt
sudo patch -p1 &lt; fastx-toolkit-gcc7-patch.txt
sudo ./configure
sudo make
sudo make install
cd ../
</pre>
</div>
<div class="section" id="mafft">
<span id="id20"></span><h3>MAFFT<a class="headerlink" href="#mafft" title="Permalink to this headline">¶</a></h3>
<p>Sequence alignment software, website <a class="reference external" href="https://mafft.cbrc.jp/alignment/software/">here</a>.</p>
<pre class="codebg literal-block">
sudo wget -P /usr/local/deb/ https://mafft.cbrc.jp/alignment/software/mafft_7.475-1_amd64.deb
sudo apt install /usr/local/deb/mafft_7.475-1_amd64.deb
</pre>
</div>
<div class="section" id="metamate">
<span id="metamate-install"></span><h3>metaMATE<a class="headerlink" href="#metamate" title="Permalink to this headline">¶</a></h3>
<p>Tool for exploring detailed read frequency filtering thresholds, github <a class="reference external" href="https://github.com/tjcreedy/metaMATE">here</a>. Requires <a class="reference internal" href="#blast"><span class="std std-ref">BLAST</span></a>, <a class="reference internal" href="#mafft"><span class="std std-ref">MAFFT</span></a> and <a class="reference internal" href="#r"><span class="std std-ref">R</span></a></p>
<pre class="codebg literal-block">
sudo -H python3 -m pip install metaMATE
sudo Rscript -e &quot;install.packages(c('getopt', 'ape', 'fastcluster'), repos = 'https://cloud.r-project.org')&quot;
</pre>
</div>
<div class="section" id="pairfq">
<span id="id23"></span><h3>Pairfq<a class="headerlink" href="#pairfq" title="Permalink to this headline">¶</a></h3>
<p>Ensures paired read files are syncronised, github <a class="reference external" href="https://github.com/sestaton/Pairfq">here</a>.</p>
<pre class="codebg literal-block">
sudo apt install libdbd-sqlite3-perl

sudo curl -sL cpanmin.us | sudo perl - git://github.com/sestaton/Pairfq.git
</pre>
</div>
<div class="section" id="pear">
<span id="id25"></span><h3>PEAR<a class="headerlink" href="#pear" title="Permalink to this headline">¶</a></h3>
<p>Paired end read merger, website <a class="reference external" href="http://www.exelixis-lab.org/pear">here</a>. You must sign up for a PEAR academic licence on that website and you will receive an email with a link to download the installation files. The command here assumes you’ve downloaded the file to your home directory.</p>
<pre class="codebg literal-block">
cd /usr/local/src/
sudo cp ~/pear-0.9.11-linux-x86_64.tar.gz ./   <span class="comment"># Change the source path if necessary</span>
sudo tar -xzf pear-0.9.11-linux-x86_64.tar.gz
cd pear-0.9.11-linux-x86_64
sudo cp bin/pear /usr/local/bin/
sudo gzip man/pear.1
sudo cp man/pear.1.gz /usr/share/man/man1/
</pre>
</div>
<div class="section" id="phylostuff">
<span id="id27"></span><h3>phylostuff<a class="headerlink" href="#phylostuff" title="Permalink to this headline">¶</a></h3>
<p>A set of scripts for doing stuff with phylogenies, including relabelling and inferring taxonomy. Github <a class="reference external" href="https://github.com/tjcreedy/phylostuff">here</a>. Requires <a class="reference internal" href="#r"><span class="std std-ref">R</span></a>.</p>
<pre class="codebg literal-block">
sudo apt install libxml2-dev libssl-dev libcurl4-openssl-dev

cd /usr/local/src
sudo git clone https://github.com/tjcreedy/phylostuff.git
cd phylostuff
sudo ./install.sh
</pre>
<p>Or just open them in Rstudio and use them as a guide!</p>
</div>
<div class="section" id="r">
<span id="id29"></span><h3>R<a class="headerlink" href="#r" title="Permalink to this headline">¶</a></h3>
<p>Versatile data handling language, website <a class="reference external" href="https://cran.r-project.org">here</a>.</p>
<pre class="codebg literal-block">
sudo apt install apt-transport-https
sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys E298A3A825C0D65DFD57CBB651716619E084DAB9
sudo add-apt-repository 'deb https://cloud.r-project.org/bin/linux/ubuntu focal-cran40/'
sudo apt install r-base
</pre>
</div>
<div class="section" id="swarm">
<span id="id31"></span><h3>swarm<a class="headerlink" href="#swarm" title="Permalink to this headline">¶</a></h3>
<p>Linkage-based OTU delimitation, github <a class="reference external" href="https://github.com/torognes/swarm">here</a>.</p>
<pre class="codebg literal-block">
cd /usr/local/src

sudo git clone https://github.com/torognes/swarm.git
cd swarm/
sudo make
sudo cp bin/swarm /usr/local/bin/

cd ~/
gzip -c /usr/local/src/man/swarm.1 &gt; swarm.1.gz
sudo mv swarm.1.gz /usr/share/man/man1/
</pre>
</div>
<div class="section" id="vsearch">
<span id="id33"></span><h3>VSEARCH<a class="headerlink" href="#vsearch" title="Permalink to this headline">¶</a></h3>
<p>Open-source implementation of USEARCH with more versatility. VSEARCH is a software package specifically designed for metabarcoding, based on the USEARCH package but completely free and open source. Github <a class="reference external" href="https://github.com/torognes/vsearch">here</a>.</p>
<pre class="codebg literal-block">
sudo apt install autoconf groff

cd /usr/local/src

sudo wget https://github.com/torognes/vsearch/archive/v2.15.1.tar.gz
sudo tar xzf v2.15.1.tar.gz
cd vsearch-2.15.1
sudo ./autogen.sh
sudo ./configure
sudo make
sudo make install
cd ../
</pre>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="../remote_access.html" class="btn btn-neutral float-right" title="Accessing linux servers" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="using_your_own_machine.html" class="btn btn-neutral float-left" title="Using your own machine" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020, The Vogler Lab.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>