

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Extension: Cutadapt parameters &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../../_static/css/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="../../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
        <script src="../../_static/jquery.js"></script>
        <script src="../../_static/underscore.js"></script>
        <script src="../../_static/doctools.js"></script>
        <script src="../../_static/language_data.js"></script>
        <script src="../../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link rel="next" title="2. Primer removal" href="../primer_removal.html" />
    <link rel="prev" title="1. Demultiplexing" href="../demultiplexing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../../index.html" class="icon icon-home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../data.html">Example data</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../../readprocessing.html">A: Read processing</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="../fastq.html">Fundamentals: FASTQ files</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="../demultiplexing.html">1. Demultiplexing</a><ul class="current">
<li class="toctree-l3 current"><a class="current reference internal" href="#">Extension: Cutadapt parameters</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="../primer_removal.html">2. Primer removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="../pair_merging.html">3. Pair merging</a></li>
<li class="toctree-l2"><a class="reference internal" href="../data_concat.html">4. Data concatenation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../../readprocessing.html">A: Read processing</a> &raquo;</li>
        
          <li><a href="../demultiplexing.html">1. Demultiplexing</a> &raquo;</li>
        
      <li>Extension: Cutadapt parameters</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="../../_sources/processing/extensions/cutadapt_extension.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="extension-cutadapt-parameters">
<span id="cutadapt-extension"></span><h1>Extension: Cutadapt parameters<a class="headerlink" href="#extension-cutadapt-parameters" title="Permalink to this headline">¶</a></h1>
<p><strong>Cutadapt</strong> has a lot more parameters for searching. Here are some of them that you might use to optimise returning as many reads as possible while still removing any errors.</p>
<div class="admonition-data-and-software admonition">
<p class="admonition-title">Data and software</p>
<p>This extension follows on from the tutorial on <a class="reference internal" href="../demultiplexing.html#demultiplexing"><span class="std std-ref">demultiplexing</span></a> using the same data and software.</p>
</div>
<div class="section" id="error-tolerance">
<h2>Error tolerance<a class="headerlink" href="#error-tolerance" title="Permalink to this headline">¶</a></h2>
<p><strong>Cutadapt</strong> can be error-tolerant, allowing you to permit some mismatches between your indices and the sequences. Our adapters are all at least 3 bases different, so in theory we could allow a one-base difference to try and get more reads for our samples. The relevant option is <code class="docutils literal notranslate"><span class="pre">-e​</span></code>, which is the maximum error rate (0-1), i.e. the total proportion of errors allowed in our indices. The default, 0.1, would allow 10% errors, but since our indices are only 6 bases this rounds to 0 errors allowed.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Create two more new directories and try <code class="docutils literal notranslate"><span class="pre">-e</span></code> values that allow one error or two errors, putting the outputs into those two directories. Explore the read numbers using <code class="docutils literal notranslate"><span class="pre">grep</span></code></p></li>
<li><p>How do the read numbers vary?</p></li>
<li><p>Is being more error-tolerant sensible?</p></li>
</ul>
</div>
</div>
<div class="section" id="variable-index-location">
<h2>Variable index location<a class="headerlink" href="#variable-index-location" title="Permalink to this headline">¶</a></h2>
<p>In the main section, we used the <code class="docutils literal notranslate"><span class="pre">^</span></code> symbol to denote that our index sequences must be exactly at the start of a read. However, sequencing errors may mean that this might not be the case. There may be extraneous bases before the index sequence, or some of the bases may be missing. Of course, if too many bases are missing it will no longer be identifiable, but as we discussed above, 5 out of 6 correct bases would still allow us to identify the correct index. Consider the following hypothetical reads (don’t run this):</p>
<pre class="literal-block">key: <strong>INDEX</strong><em>PRIMER</em>SEQUENCE
1  <strong>AACACC</strong><em>CCAGATATGGCCTTCCCACG</em>ATCC    <span class="comment"># This is perfect</span>
2  <strong>ACACC</strong><em>CCAGATATGGCCTTCCCACG</em>ATCCG    <span class="comment"># This is missing just one index base</span>
3  T<strong>AACACC</strong><em>CCAGATATGGCCTTCCCACG</em>ATC    <span class="comment"># This has an extra base before the index</span></pre>
<p><strong>Cutadapt</strong> is very flexible, and we can find these variants if we abandon the anchoring (i.e. we remove <code class="docutils literal notranslate"><span class="pre">^</span></code> before the indices).</p>
<p>To cope with the truncated index (example 2 above), we can set the minimum overlap parameter, <code class="docutils literal notranslate"><span class="pre">-O</span></code> to 5, i.e. we must find at least 5 bases (in order) of the index sequence. To allow the index sequence to move into the read, we can add <code class="docutils literal notranslate"><span class="pre">N</span></code> before the index, essentially searching for an index that is 7 bases long, but the first base can be anything. Paired with the <code class="docutils literal notranslate"><span class="pre">-O</span> <span class="pre">5</span></code> option, this will find all of the examples above.</p>
<p>But wait, we’ve removed index anchoring, so now it’ll <em>also</em> find any sequence motifs within the read that match the index - that’s not good! Never fear. We can add a special character, <code class="docutils literal notranslate"><span class="pre">X</span></code> to the beginning of our indices to make them “noninternal”. Within cutadapt, the <code class="docutils literal notranslate"><span class="pre">X</span></code> base is the opposite of <code class="docutils literal notranslate"><span class="pre">N</span></code>: where <code class="docutils literal notranslate"><span class="pre">N</span></code> is anything, <code class="docutils literal notranslate"><span class="pre">X</span></code> is nothing - <code class="docutils literal notranslate"><span class="pre">X</span></code> should never match a base, and therefore the index cannot match anything within the read.</p>
<p>To visualise this, we can consider a search using the index <code class="docutils literal notranslate"><span class="pre">g=XNAACACC</span></code> with <code class="docutils literal notranslate"><span class="pre">-O</span> <span class="pre">5</span></code> (don’t run this):</p>
<pre class="literal-block">1  <strong>AACACC</strong><em>CCAGATATGGCCTTCCCACG</em>ATCC
 <strong>XNAACACC</strong>  = match!

2  <strong>ACACC</strong><em>CCAGATATGGCCTTCCCACG</em>ATCCG
<strong>XNAACACC</strong>  = match!

3  T<strong>AACACC</strong><em>CCAGATATGGCCTTCCCACG</em>ATC
  <strong>XNAACACC</strong>  = match!

4  ATG<strong>AACACC</strong>CGUTTAGCTAAACGGATTTAGAC
    <strong>XNAACACC</strong> = no match, X may not match a base!</pre>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Create another new directory, then build a <strong>cutadapt</strong> command that uses these “noninternal” indices. If you get stuck, see the solution below.</p></li>
<li><p>Check the number of reads returned using <code class="docutils literal notranslate"><span class="pre">grep</span></code>.</p></li>
<li><p>Do we see a boost in read counts?</p></li>
</ul>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<pre class="codebg literal-block">cutadapt -g T4=XNAAGAGG -g T9=XNAATCGC -g T11=XNAGCTAC \
-G T4=XNAAGAGG -G T9=XNAATCGC -G T11=XNAGCTAC \
-O 5 \
-o 1_demux/{name1}-{name2}_R1.fastq -p 1_demux/{name1}-{name2}_R2.fastq \
0_rawsequences/Lib1_R1.fastq 0_rawsequences/Lib1_R2.fastq</pre>
</div>
</div>
<div class="section" id="tradeoffs">
<h2>Tradeoffs<a class="headerlink" href="#tradeoffs" title="Permalink to this headline">¶</a></h2>
<p>Changing the error rate, minimum overlap and/or using noninternal indices (rather than anchored indices) generally increases the number of reads we get, but these may allow more errors to creep in. As always with bioinformatics, it’s a trade-off between having highly accurate data or having as much data as possible. There is no right answer, you should consider what variation you consider to be reasonable within valid data, and which is more important: highly accurate data with possible missing information, or complete data with some inaccuracies.</p>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>If you’re following step-by-step with the example data, we suggest you delete or archive the files you created here as this was just to explore a bit, and use the outputs from the main <a class="reference internal" href="../demultiplexing.html#demultiplexing"><span class="std std-ref">demultiplexing</span></a> tutorial. But if you really want, you could run <strong>cutadapt</strong> using these modified parameters on all of the raw sequence files and continue with those outputs.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="../primer_removal.html" class="btn btn-neutral float-right" title="2. Primer removal" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="../demultiplexing.html" class="btn btn-neutral float-left" title="1. Demultiplexing" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020, The Vogler Lab.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>