

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Making a Barcode Tree &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Viewing alignments" href="otu_trees/alignment_viewing.html" />
    <link rel="prev" title="Building OTU Phylogeny" href="../building_otu_phylogeny.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu_asv_read.html">OTUs, ASVs and Read Mapping</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../building_otu_phylogeny.html">Building OTU Phylogeny</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">Making a Barcode Tree</a><ul>
<li class="toctree-l3"><a class="reference internal" href="otu_trees/alignment_viewing.html">Viewing alignments</a></li>
<li class="toctree-l3"><a class="reference internal" href="otu_trees/tree_viewing.html">Viewing Trees</a></li>
<li class="toctree-l3"><a class="reference internal" href="otu_trees/reference_otu_tree.html">Extension: Barcode Tree with References</a></li>
<li class="toctree-l3"><a class="reference internal" href="otu_trees/tree_comparison.html">Extension: Comparing OTU methods</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="placing_otus_in_trees.html">Phylogenetic Placement</a></li>
<li class="toctree-l2"><a class="reference internal" href="trees.html">Building a Mitogenome Reference Tree</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../building_otu_phylogeny.html">Building OTU Phylogeny</a> &raquo;</li>
        
      <li>Making a Barcode Tree</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otu_phylogeny/otu_trees.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="making-a-barcode-tree">
<span id="otu-tree"></span><h1>Making a Barcode Tree<a class="headerlink" href="#making-a-barcode-tree" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Our OTUs can directly be used to make a phylogenetic tree, as they are all sequences of the same locus. This tree may not be very accurate, because there is relatively little data: for example, our toy dataset only has 418 bases. On the other hand, building a phylogeny for short sequences is relatively fast for many many sequences, while building a phylogeny of much longer sequences could be much slower.</p>
</div>
<div class="section" id="alignment">
<h2>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h2>
<p>The first stage of any tree building is to align the sequences so that homologous bases are lined up with one another. There are many alignment programs, perhaps the three most commonly used are Clustal, MUSCLE and MAFFT. We will use <a class="reference external" href="https://mafft.cbrc.jp/alignment/software/">MAFFT</a> because it is straightforward to use, fast, and generally very accurate. We will perform a global alignment using the accuracy INS-i algorithm:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>mafft --thread <span class="m">1</span> --globalpair -- maxiterate <span class="m">1000</span> input.fasta &gt; output.fasta
</pre></div>
</div>
<p>Where we have a coding locus with a very strict length expectation, such as in the toy data where all sequences are 418bp, we can choose to skip this step, because the sequences are already in some form of alignment.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Download your input sequences and your alignment. Open each file in the software Aliview, or your preferred alignment viewer. If you’re completely new to viewing alignments, you may want to read the page on <span class="xref std std-ref">viewing alignments</span>. You should see no gaps in your input sequences, but some gaps in your aligned sequences. How accurate does the alignment appear to be - is it very gappy or only slightly. In general, are the alignmened sequences consistent?</p>
</div>
</div>
<div class="section" id="building-a-tree">
<h2>Building a tree<a class="headerlink" href="#building-a-tree" title="Permalink to this headline">¶</a></h2>
<p>We will build a tree of our OTUs using <a class="reference external" href="http://www.microbesonline.org/fasttree/index.html">FastTree</a>. We are using FastTree because it’s fast, not because it’s necessarily highly accurate, although it generally does pretty well. The command to use to build this tree is below. The <code class="docutils literal notranslate"><span class="pre">-gtr</span></code> option tells FastTree to use the GTR+CAT model, and the <code class="docutils literal notranslate"><span class="pre">-nt</span></code> option specifies that this is DNA data.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>FastTree -gtr -nt &lt; input.fasta &gt; output.tre
</pre></div>
</div>
<p>The algorithm generates a newick file, which is the most common format for phylogenetic trees. This is a plain text format, you can view the file using the <code class="docutils literal notranslate"><span class="pre">head</span></code> command. Each terminal in the tree has the ID of the sequence used to generate it.</p>
<p>You can download this tree to your computer and open it using a tree viewer such as Figtree or Dendroscope. If you’re completely new to viewing phylogenetic trees, you may want to read the page on <span class="xref std std-ref">viewing trees</span>.</p>
<p>It is hard to assess the accuracy of these trees with anonymous sequences, but we can get some idea by looking at the bootstrap values of the nodes. These tell us how many times this clade was recovered as a monophyletic clade in 100 random permutations of the tree building algorithm. A higher value suggests that a clade is more accurate, while a lower value suggest that a clade may have been grouped together inaccurately.</p>
<p>Another method to assess accuracy is to look at the distribution of branch lengths. If there are many long branches, this suggests that the phylogenetic reconstruction had a poor time trying to place the sequences at the end of these branches.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Build a tree using both your unaligned OTU sequences and the aligned OTU sequences. Download the trees to your computer and view them.</p>
<p>How do they compare? Select some monophyletic clades of OTUs from one tree and see if the same sets of OTUs each come out as monophyletic clades in the other tree.</p>
<p>How do the bootstrap values of the two trees compare? Overall, does either of the trees appear very accurate.</p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>One way to improve our ability to assess the accuracy of a tree built with anonymous sequences is to add in known sequences of the same barcode region. If you’d like to try this out, see the <a class="reference internal" href="otu_trees/reference_otu_tree.html#reference-otu-tree"><span class="std std-ref">Barcode Tree with References</span></a> extension. We can also use trees to compare the OTU sequences generated by different methods. If you like to see how, see the <a class="reference internal" href="otu_trees/tree_comparison.html#otu-comparison-tree"><span class="std std-ref">OTU Comparison Tree</span></a> extension.</p>
<p>Alternatively, you can proceed to the next subsection, <a class="reference internal" href="placing_otus_in_trees.html#phylogenetic-placement"><span class="std std-ref">Phylogenetic Placement</span></a>, where we will use reference trees rather than just reference sequences.</p>
<div class="green admonition">
<p class="admonition-title">Note</p>
<p>Remember: when performing these steps on a real dataset, we would strongly advise trying some different alignment algorithms and possibly even manually correcting your alignments to get the best dataset for phylogenetics. If using the MAFFT command above, you should probably increase the <code class="docutils literal notranslate"><span class="pre">--thread</span></code> argument to better suit your data, otherwise it’ll take a long time to run! We would also recommend using alternative phylogenetics software, such as RAxML, IQtree or MrBayes which are generally more accurate, but much slower.</p>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="otu_trees/alignment_viewing.html" class="btn btn-neutral float-right" title="Viewing alignments" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../building_otu_phylogeny.html" class="btn btn-neutral float-left" title="Building OTU Phylogeny" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>The content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>