

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Phylogenetic Placement &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Building a Mitogenome Reference Tree" href="trees.html" />
    <link rel="prev" title="Extension: Comparing OTU methods" href="otu_trees/tree_comparison.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="otu_trees.html">Making a Barcode Tree</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Phylogenetic Placement</a></li>
<li class="toctree-l2"><a class="reference internal" href="trees.html">Building a Mitogenome Reference Tree</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a> &raquo;</li>
        
      <li>Phylogenetic Placement</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otu_phylogeny/placing_otus_in_trees.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="phylogenetic-placement">
<span id="id1"></span><h1>Phylogenetic Placement<a class="headerlink" href="#phylogenetic-placement" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>In the previous section we built a <em>de novo</em> phylogenetic tree from our OTU sequences, but this is a short fragment of DNA and not highly accurate. Lots of researchers put a lot of effort into building increasingly accurate phylogenies with much more comprehensive datasets. Wouldn’t it be much easier if we could just use those existing phylogenies, rather than building a phylogeny from scratch, and reduce our effort down to simply figuring out where our new sequences fit into those trees? Well, we can. In this subsection, we will use a reference phylogeny of the Coleoptera, built with complete mitochondrial genomes, and place our OTUs onto this tree. The more comprehensive reference phylogeny allows us better resolution for figuring out the deep-level relationships between our OTUs than with the marker region alone. The reference phylogeny forms a ‘backbone’, to which we add the OTUs. The backbone tree is fixed, and new branches are added for the OTUs. This is often called a “constraint tree”.</p>
<p>The most important point to remember here is that we must have genetic data for all of the terminals of the existing tree, and at least some of that genetic data should cover the same region of the genome as our metabarcoding marker. Ideally, all of the terminals of the existing tree would have sequence data for our metabarcoding region, but often we have incomplete datasets. This is not the end of the world: the phylogenetic algorithm is able to take account of this missing data and estimate the placement of a new sequence based on its similarity to the sequences with data, and given that incomplete sequences will correspond to other parts of other reference sequences, pairwise distances can be inferred anyway.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this tutorial is a FASTA file comprising the sequences you want to place on a phylogeny, and data for the phylogeny itself. The phylogeny data should comprise of 1. the final alignment/supermatrix in FASTA format, 2. the tree in newick format and 3. a table of taxonomy for the sequences on the tree.</p>
<p>If you’re following along step-by-step, for the sequences to place you can use one of your OTU outputs from <a class="reference internal" href="../asv_otu_readmap/otudelim.html#otu-delim"><span class="std std-ref">the OTU delimitation tutorials</span></a>, or your ASVs produced in <a class="reference internal" href="../filtering.html#filtering"><span class="std std-ref">the filtering section</span></a>. Alternatively, the file <code class="docutils literal notranslate"><span class="pre">otus_greedy_0.97.fasta</span></code> within the <a class="reference internal" href="../data.html#sectionddata"><span class="std std-ref">sectionD archive</span></a> can be used as example data. For the phylogeny, you should use the files TODO XXXX XXXX, and XXXXX within the <a class="reference internal" href="../data.html#sectionddata"><span class="std std-ref">sectionD archive</span></a></p>
<p>If you’re using your own data, you would need to acquire a published phylogeny, or build one yourself. You can review how to do the latter in the <a class="reference internal" href="trees.html#trees"><span class="std std-ref">Building a mitogenome reference tree</span></a>, although we’d suggest you try this tutorial with the example data first to understand the context.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#mafft"><span class="std std-ref">**MAFFT**</span></a> and <span class="xref std std-ref">**FastTree**</span>, as well as the <span class="xref std std-ref">XXXX</span> script TODO. You should also have an alignment viewer, such as <strong>Aliview</strong>, and a tree viewer, such as <strong>FigTree</strong> or <strong>Dendroscope</strong>, installed on your personal computer.</p>
</div>
</div>
<div class="section" id="getting-started">
<h2>Getting started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h2>
<p>Make sure you have the four files you need for this tutorial (see the Data and software box above) copied into your working directory.</p>
</div>
<div class="section" id="alignment">
<h2>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h2>
<p>The first step is to align our OTUs to our reference mitogenome dataset. Run the following command, using your OTU FASTA and the supermatrix FASTA.</p>
<pre class="literal-block">mafft --thread 1 --addfragments ​:var:<cite>otus.fasta`​ --6merpair :var:`​supermatrix.fasta​</cite> &gt; ​:var:<cite>output.fasta</cite></pre>
<p>Here we use <a href="#id2"><span class="problematic" id="id3">**</span></a>MAFFT**`s ​``–addfragments`` argument, which you can read about <a class="reference external" href="https://mafft.cbrc.jp/alignment/software/addsequences.html">here</a>. Again, we’re using some options which make this alignment very very fast, but the accuracy might not be great.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Download this alignment and have a look at it using your alignment viewer.</p></li>
<li><p>How well have the OTUs aligned?</p></li>
</ul>
</div>
</div>
<div class="section" id="setting-up-the-constraint">
<h2>Setting up the constraint<a class="headerlink" href="#setting-up-the-constraint" title="Permalink to this headline">¶</a></h2>
<p>We want to build a tree with this combined alignment, but instead of building it from scratch we want to use our reference mitogenome tree as a constraint tree.</p>
<p>For <strong>FastTree</strong>, we must convert our existing tree into a constraint alignment, as detailed in the <a class="reference external" href="http://www.microbesonline.org/fasttree/constrained.html">documentation</a> ​. You’ll notice that they supply a handy script for this conversion.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Download this script to your working directory</p></li>
<li><p>See if you can figure out how to run this script from the documentation, then use it convert the reference tree.</p></li>
</ul>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<pre class="literal-block">wget <a class="reference external" href="http://www.microbesonline.org/fasttree/TreeToConstraints.pl">http://www.microbesonline.org/fasttree/TreeToConstraints.pl</a>
perl TreeToConstraints.pl &lt; ​:var:<cite>reference.tre</cite> ​&gt; <a href="#id4"><span class="problematic" id="id5">:var:`​constraints.txt`</span></a></pre>
</div>
</div>
<div class="section" id="building-the-tree">
<h2>Building the tree<a class="headerlink" href="#building-the-tree" title="Permalink to this headline">¶</a></h2>
<p>Now we can run the new tree building to place the OTUs within the reference. We add the <code class="docutils literal notranslate"><span class="pre">​-constraints</span></code> option to <strong>FastTree</strong> to do this. Remember, we’re running this using the combined supermatrix you just made with <strong>MAFFT</strong>.</p>
<pre class="literal-block">FastTree -nt -gtr -constraints <a href="#id6"><span class="problematic" id="id7">:var:`​constraints.txt`</span></a> &lt; ​:var:<cite>combinedsupermatrix.fasta</cite> &gt;​ <a href="#id8"><span class="problematic" id="id9">:var:`output.tre`</span></a></pre>
<p>Finally, we just need to add the taxonomy onto the tree for our reference sequences using the <strong>XX</strong> script, as follows:</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<ul class="simple">
<li><p>Download this tree to your computer and view it.</p></li>
<li><p>Use your tree viewer’s search function to highlight the OTUs. Has their topology changed compared with the OTU-only tree? Is it improved?</p></li>
<li><p>Are OTUs distributed across taxonomic clades, or are they clustered within clades? What might be the reasons for these patterns?</p></li>
<li><p>Have any OTUs been placed very close to any of our novel references? What might this mean?</p></li>
</ul>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>You now have a phylogeny that you can use for downstream analyses. Of course, there are a lot of sequences on this tree that aren’t your OTUs, but it would be completely valid to remove the non-OTU sequences and retain only the OTU phylogeny for analysis. You could do this in <strong>R</strong> using commands from the <a class="reference external" href="https://cran.r-project.org/web/packages/ape/">ape</a> package, for example.</p>
<p>If you want to learn how to build a reference tree like we used here from scratch, check out the <a class="reference internal" href="trees.html#trees"><span class="std std-ref">Building a Mitogenome Tree</span></a> subsection.</p>
<p>If you want to learn how to identify your OTUs, you can see the <a class="reference internal" href="../otuid.html#otuid"><span class="std std-ref">Identifying OTU sequences</span></a> section, where we will look at several methods for taxonomic identification and/or classification of OTUs. One of these methods draws directly from the tree you’ve produced here, using the taxonomy of the reference sequences in the backbone tree to infer taxonomy of the OTUs: see the <a class="reference internal" href="../otuid/phylogeny_based.html#phylogenetic-classification"><span class="std std-ref">Phylogenetic classification</span></a> tutorial.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="trees.html" class="btn btn-neutral float-right" title="Building a Mitogenome Reference Tree" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="otu_trees/tree_comparison.html" class="btn btn-neutral float-left" title="Extension: Comparing OTU methods" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>