

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Phylogenetic Placement &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Building a Mitogenome Reference Tree" href="trees.html" />
    <link rel="prev" title="Viewing Trees" href="otu_trees/tree_viewing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">Introduction</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Command line bioinformatics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../readprocessing.html">Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu_asv_read.html">OTUs, ASVs and Read Mapping</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../building_otu_phylogeny.html">Building OTU Phylogeny</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="otu_trees.html">Making a Barcode Tree</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">Phylogenetic Placement</a></li>
<li class="toctree-l2"><a class="reference internal" href="trees.html">Building a Mitogenome Reference Tree</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../building_otu_phylogeny.html">Building OTU Phylogeny</a> &raquo;</li>
        
      <li>Phylogenetic Placement</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/otu_phylogeny/placing_otus_in_trees.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="phylogenetic-placement">
<span id="id1"></span><h1>Phylogenetic Placement<a class="headerlink" href="#phylogenetic-placement" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>In the previous section we built a <em>de novo</em> phylogenetic tree from our OTU sequences, but this is a short fragment of DNA and not highly accurate. Lots of researchers put a lot of effort into building increasingly accurate phylogenies with much more comprehensive datasets. Wouldn’t it be much easier if we could just use those existing phylogenies, rather than building a phylogeny from scratch, and reduce our effort down to simply figuring out where our new sequences fit into those trees? Well, we can. In this subsection, we will use a reference phylogeny of the Coleoptera, built with complete mitochondrial genomes, and place our OTUs onto this tree. The more comprehensive reference phylogeny allows us better resolution for figuring out the deep-level relationships between our OTUs than with the marker region alone. The reference phylogeny forms a ‘backbone’, to which we add the OTUs. The backbone tree is fixed, and new branches are added for the OTUs. This is often called a “constraint tree”.</p>
<p>The most important point to remember here is that we must have genetic data for all of the terminals of the existing tree, and at least some of that genetic data should cover the same region of the genome as our metabarcoding marker. Ideally, all of the terminals of the existing tree would have sequence data for our metabarcoding region, but often we have incomplete datasets. This is not the end of the world: the phylogenetic algorithm is able to take account of this missing data and estimate the placement of a new sequence based on its similarity to the sequences with data, and given that incomplete sequences will correspond to other parts of other reference sequences, pairwise distances can be inferred anyway.</p>
</div>
<div class="section" id="getting-started">
<h2>Getting Started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h2>
<p>To run this section with the toy dataset, you will need to copy over the reference tree and reference supermatrix FASTA to your working directory. You will also need a FASTA file of your OTU sequences. If you’re doing this with your own data, you will need to have a reference tree and set of sequences for those references. If you’re not sure how to get these, try this out with our toy data first, then see the <a class="reference internal" href="trees.html#trees"><span class="std std-ref">Building a Mitogenome Tree</span></a> subsection on how to go about building a reference mitogenome tree to use with your own data.</p>
</div>
<div class="section" id="alignment">
<h2>Alignment<a class="headerlink" href="#alignment" title="Permalink to this headline">¶</a></h2>
<p>The first step is to align our OTUs to our reference mitogenome dataset. Run the following command, using your OTU FASTA and the supermatrix FASTA.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ mafft --thread <span class="m">1</span> --addfragments ​otus.fasta​ --6merpair ​supermatrix.fasta​ &gt; ​out.fasta
</pre></div>
</div>
<p>Here we use mafft’s ​–addfragments argument, which you can read about <a class="reference external" href="https://mafft.cbrc.jp/alignment/software/addsequences.html">here</a>. Again, we’re using some options which make this alignment very very fast, but the accuracy might not be great.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Download this alignment and have a look at it using your alignment viewer.</p>
<p>How well have the OTUs aligned?</p>
</div>
</div>
<div class="section" id="setting-up-the-constraint">
<h2>Setting up the Constraint<a class="headerlink" href="#setting-up-the-constraint" title="Permalink to this headline">¶</a></h2>
<p>We want to build a tree with this combined alignment, but instead of building it from scratch we want to use our reference mitogenome tree as a constraint tree.</p>
<p>For FastTree, we must convert our existing tree into a constraint alignment, as detailed in the <a class="reference external" href="http://www.microbesonline.org/fasttree/constrained.html">documentation</a> ​. You’ll notice that they supply a handy script for this conversion - we’ve already downloaded this for you, so go ahead and copy it from <code class="docutils literal notranslate"><span class="pre">/AMM/resources/phylogenetics/TreeToConstraints.pl</span></code> to your directory.</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>See if you can figure out how to run this script from the documentation, then use it convert the reference tree.</p>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>perl TreeToConstraints.pl &lt; ​reference.tre ​&gt; ​constraints.txt
</pre></div>
</div>
</div>
</div>
<div class="section" id="building-the-tree">
<h2>Building the Tree<a class="headerlink" href="#building-the-tree" title="Permalink to this headline">¶</a></h2>
<p>Now we can run the new tree building to place the OTUs within the reference. We add the <code class="docutils literal notranslate"><span class="pre">​-constraints</span></code> option to FastTree to do this. Remember, we’re running this using the combined supermatrix you just made with MAFFT.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ FastTree -nt -gtr -constraints ​constraints.txt &lt; ​combinedsupermatrix.fasta &gt;​ tree.tre
</pre></div>
</div>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Download this tree to your computer and view it.</p>
<p>Use your tree viewer’s search function to highlight the OTUs. Has their topology changed compared with the OTU-only tree? Is it improved?</p>
<p>Are OTUs distributed across taxonomic clades, or are they clustered within clades? What might be the reasons for these patterns?</p>
<p>Have any OTUs been placed very close to any of our novel references? What might this mean?</p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>You now have a phylogeny that you can use for downstream analyses. Of course, there are a lot of sequences on this tree that aren’t your OTUs, but it would be completely valid to remove the non-OTU sequences and retain only the OTU phylogeny for analysis. You could do this in R using commands from the <a class="reference external" href="https://cran.r-project.org/web/packages/ape/">ape</a> package, for example.</p>
<p>If you want to learn how to build a reference tree like we used here from scratch, check out the <a class="reference internal" href="trees.html#trees"><span class="std std-ref">Building a Mitogenome Tree</span></a> subsection.</p>
<p>If you want to learn how to identify your OTUs, you can see the <a class="reference internal" href="../otuid.html#otuid"><span class="std std-ref">Identifying OTU Sequences</span></a> section, where we will look at several methods for taxonomic identification and/or classification of OTUs. One of these methods draws directly from the tree you’ve produced here, using the taxonomy of the reference sequences in the backbone tree to infer taxonomy of the OTUs: see the <span class="xref std std-ref">Phylogenetic Classification</span> subsection.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="trees.html" class="btn btn-neutral float-right" title="Building a Mitogenome Reference Tree" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="otu_trees/tree_viewing.html" class="btn btn-neutral float-left" title="Viewing Trees" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>