

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>4. Data concatenation &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="B: Filtering amplicons" href="../filtering.html" />
    <link rel="prev" title="Extension: Pair concatenation" href="extensions/pair_concat.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../readprocessing.html">A: Read processing</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="fastq.html">Fundamentals: FASTQ files</a></li>
<li class="toctree-l2"><a class="reference internal" href="demultiplexing.html">1. Demultiplexing</a></li>
<li class="toctree-l2"><a class="reference internal" href="primer_removal.html">2. Primer removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="pair_merging.html">3. Pair merging</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">4. Data concatenation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="../contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../readprocessing.html">A: Read processing</a> &raquo;</li>
        
      <li>4. Data concatenation</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/processing/data_concat.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="data-concatenation">
<span id="data-concat"></span><h1>4. Data concatenation<a class="headerlink" href="#data-concatenation" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>At this point we have a set of sequence reads for each sample with extraneous sequence removed and read pairs brought together. Metabarcoding works on all sequence reads from across the dataset to find OTUs, and it’s more efficient if we do the following steps with all reads compiled into one file.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this section is a directory of FASTQ (or FASTA) format files, one per sample, with primer and index sequences removed and mate pairs merged or concatenated. If you’re following along step-by-step, this was produced in <a class="reference internal" href="pair_merging.html#pair-merging"><span class="std std-ref">the previous tutorial</span></a>. Alternatively, the <code class="docutils literal notranslate"><span class="pre">3_merged</span></code> directory within the <a class="reference internal" href="../data.html#sectionadata"><span class="std std-ref">sectionA archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#cutadapt"><span class="std std-ref">cutadapt</span></a> software.</p>
</div>
<div class="green admonition">
<p class="admonition-title">Note</p>
<p>If you’re following along step-by-step, we present this step at this stage because it allows us to separate two distinct themes: the processing of raw sequence reads into samples, in section A, and the filtering of processed sequence reads to remove errors, in the section B. In this case, your input files would be FASTQs, so you’d use the first part of this tutorial.</p>
<p>However, when performing metabarcoding bioinformatics it may be more efficient to perform this task after quality filtering, since quality filtering substantially reduces the data size. If you wanted to do this, the input files would be FASTAs, so you’d use the second part of this tutorial.</p>
</div>
</div>
<div class="section" id="performing-concatenation">
<h2>Performing concatenation<a class="headerlink" href="#performing-concatenation" title="Permalink to this headline">¶</a></h2>
<p>We could simply use the <code class="docutils literal notranslate"><span class="pre">cat</span></code> command to concatenate all the individual sample files into one big file, but then we’d lose information about what reads come from what sample and this is important later. So, as part of the concatenation we will also modify the headers of each of our files to contain the file name.</p>
<div class="section" id="fastq-files">
<h3>FASTQ files<a class="headerlink" href="#fastq-files" title="Permalink to this headline">¶</a></h3>
<p>If you’re following along step-by-step, we’ll be working on a set of FASTQ files. As we’ve seen previously using <code class="docutils literal notranslate"><span class="pre">grep</span></code>, it’s not necessarily straightforward to identify the lines corresponding to headers in FASTQ files, because quality lines can also start with <code class="docutils literal notranslate"><span class="pre">&#64;</span></code>. However, in this case we know that our headers start with <code class="docutils literal notranslate"><span class="pre">&#64;D00</span></code> and we can use this to our advantage. If you’re using your own data, you should use <code class="docutils literal notranslate"><span class="pre">head</span></code> to check what your headers start with.</p>
<p>To run the following command, your working directory should be the directory containing the sequences you want to concatenate, namely the directory of FASTQ files that have indices and primers removed, and pairs merged.</p>
<p>Run the following command, making sure to replace <code class="docutils literal notranslate"><span class="pre">output.fastq</span></code> with a sensible name:</p>
<pre class="literal-block">for f in *;
do                         <span class="comment"># ​↓​ the space here should be included!</span>
        sed -e &quot;s/\(^&#64;D00.*\) .*$/\1;sample=${f%.*};/&quot; $f \
        &gt;&gt; ../<span class="var">output.fastq</span>;
done</pre>
<div class="togglegreen admonition">
<p class="admonition-title">How this command works</p>
<p>This command loops through all the files in the directory. For each file, the <code class="docutils literal notranslate"><span class="pre">sed</span></code> command uses regular expression substitution to add a string to the end of each header line containing the sample name, and adds the entire file to the end of a single file in the parent directory called <code class="docutils literal notranslate"><span class="pre">output.fastq​</span></code>. See <a class="reference internal" href="../gettingstarted/cli_bioinformatics/sed.html#sed"><span class="std std-ref">here</span></a> to learn more about <code class="docutils literal notranslate"><span class="pre">sed</span></code>.</p>
<p>Note the <code class="docutils literal notranslate"><span class="pre">&gt;&gt;</span></code> symbol. We use this to <em>append</em> to a file, rather than overwriting it. This allows to add to it every iteration of the loop. Be careful though: the first iteration of the loop will create this file if it doesn’t exist, but <em>if it already exists it will just add to it</em>. So if you make a mistake and need to re-run this loop, you should delete the output file (<code class="docutils literal notranslate"><span class="pre">rm</span> <span class="pre">../output.fastq</span></code>) so you don’t just keep adding to one huge file.</p>
</div>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>Return to the parent directory, use <code class="docutils literal notranslate"><span class="pre">grep</span></code> to count the number of sequences in output file ​and view the <code class="docutils literal notranslate"><span class="pre">​head​</span></code> of the file to check that we’ve correctly added the sample names to the headers.</p>
</div>
</div>
<div class="section" id="fasta-files">
<h3>FASTA files<a class="headerlink" href="#fasta-files" title="Permalink to this headline">¶</a></h3>
<p>If you instead have a directory of FASTAs, the command is pretty much the same, but we don’t have to worry about FASTQ header format. Again, to run this command your working directory should be the directory containing the sequences you want to concatenate.</p>
<pre class="literal-block">for f in *;
do                     # ​↓​ the space here should be included!
        sed -e &quot;s/\(^&gt;.*\) .*$/\1;sample=${f%.*};/&quot; $f \
        &gt;&gt; ../<span class="var">output.fasta</span>;
done</pre>
<p><span class="guilabel">Can you spot the difference?</span></p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next Steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We have generated a master file containing all sequences from across all of our samples. These sequences still contain quality data, which will be useful for filtering in the next section.</p>
<p>If you’re following along step-by-step, we recommend heading to the next section: <a class="reference internal" href="../filtering.html#filtering"><span class="std std-ref">B. Filtering amplicons</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="../filtering.html" class="btn btn-neutral float-right" title="B: Filtering amplicons" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extensions/pair_concat.html" class="btn btn-neutral float-left" title="Extension: Pair concatenation" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>