

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>FASTQ files &mdash; Analysis of Metagenetic data for Macroecology  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Demultiplexing" href="demultiplexing.html" />
    <link rel="prev" title="Read processing" href="../readprocessing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Analysis of Metagenetic data for Macroecology
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../notes.html">The Practical Notes</a></li>
<li class="toctree-l1"><a class="reference internal" href="../linuxintro.html">Introduction to the linux command line</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../readprocessing.html">Read processing</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">FASTQ files</a></li>
<li class="toctree-l2"><a class="reference internal" href="demultiplexing.html">Demultiplexing</a></li>
<li class="toctree-l2"><a class="reference internal" href="primer_removal.html">Primer removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="pair_merging.html">Pair merging</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_concat.html">Data concatenation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otudelim.html">OTU Delimitation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTUs and Generating Outputs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../trees.html">Building Phylogenetic Trees for Metagenetics</a></li>
<li class="toctree-l1"><a class="reference internal" href="../placing_otus_in_trees.html">Placing OTUs in reference trees</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Analysis of Metagenetic data for Macroecology</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../readprocessing.html">Read processing</a> &raquo;</li>
        
      <li>FASTQ files</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/processing/fastq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="fastq-files">
<h1>FASTQ files<a class="headerlink" href="#fastq-files" title="Permalink to this headline">¶</a></h1>
<p>Let’s review a few basic points about sequence file structure and exploring these files using the command line. If you have experience working with fastq files you can move onto the <a class="reference internal" href="demultiplexing.html#demultiplexing"><span class="std std-ref">demultiplexing section</span></a>.</p>
<p>Change into the directory and list its contents, showing sizes:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> 0_rawsequences/

$ ls -lh
</pre></div>
</div>
<p>We can see how many lines in each file using the word count function, specifying we want the number of lines:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ wc -l *.fastq
</pre></div>
</div>
<p>The ​:samp:<cite>*.fastq</cite> here means we want all of the fasta files in the directory. We could replace this with a single
file name.
We would like to review the structure of these files. We could print the whole file to the terminal using the cat command, but these files are over 35000 lines. We expect them to be the same structure all through the file, so the first 10 lines should be fine. Run the following command, replacing ​file with one of the <code class="docutils literal notranslate"><span class="pre">file</span></code> names.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ head -n <span class="m">10</span> ​file
</pre></div>
</div>
<p>You will see the fastq format comprising header, sequence and quality scores. A useful point to note is the structure of the file header, specifically that it starts with “&#64;D00”. If the structure of this file is completely new to you, take a few minutes to read the first section on the wikipedia page on fastq format: <a class="reference external" href="https://en.wikipedia.org/wiki/FASTQ_format">https://en.wikipedia.org/wiki/FASTQ_format</a></p>
<p>To get specific lines from a file, use the sed function:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -n ‘4,8p’ ​file <span class="c1"># prints lines 4-8</span>
</pre></div>
</div>
<ul class="simple">
<li><p>Use this to have a look at some files. If it doesn’t work - well you might have copied and pasted the command, which has formatted quotation marks. You need to type this out.</p></li>
</ul>
<p>Note that the R1 and R2 files from the same library have the same read headers, apart from a 1 or 2 in the second part of the name. Reads with the same header were read from the same location on the sequencer, so they are assumed to be the forward and reverse read of the same fragment - these are called <strong>​mate pairs​.</strong> It’s important to ensure that both the forward and reverse read for each fragment are always kept present (“in sync”) for some future processes.</p>
<p>While we can divide the results of our wordcount function above by four to get the number of sequences, it is quicker to use <code class="docutils literal notranslate"><span class="pre">​grep</span></code> ​. This is a powerful tool that searches for text in a file using regular expressions, and we can use it to search for all the headers and return a count of them, as follows:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep -c <span class="s2">&quot;^@D00&quot;</span> ​file
</pre></div>
</div>
<p>The ​``^&#64;D00`` is a regular expression - the ​^ character is a special symbol in regex meaning “the beginning of a line”. So this regex searches for lines beginning with <code class="docutils literal notranslate"><span class="pre">​&#64;D00</span></code> We use more than just <code class="docutils literal notranslate"><span class="pre">​&#64;</span></code> because it is also the symbol for a quality score of 31. Any sequences where the first base has a score of 31 will be counted twice - once for the header, once for the quality string, as they are two different lines starting with <code class="docutils literal notranslate"><span class="pre">​&#64;</span></code> ​. Hence using multiple characters.</p>
<p>Like the ​wc -l function above, we can run grep on all of our files at once to get the total read numbers for each of our libraries:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep -c <span class="s2">&quot;^@D00&quot;</span> *.fastq
</pre></div>
</div>
<p>We can see that we’re dealing with about 9000 reads per library.</p>
<ul class="simple">
<li><p>Do all the libraries have the same R1 and R2 read numbers?</p></li>
</ul>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="demultiplexing.html" class="btn btn-neutral float-right" title="Demultiplexing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../readprocessing.html" class="btn btn-neutral float-left" title="Read processing" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>