

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Fundamentals: FASTQ files &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="1. Demultiplexing" href="demultiplexing.html" />
    <link rel="prev" title="Read processing" href="../readprocessing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../readprocessing.html">Read processing</a><ul class="current">
<li class="toctree-l2 current"><a class="current reference internal" href="#">Fundamentals: FASTQ files</a></li>
<li class="toctree-l2"><a class="reference internal" href="demultiplexing.html">1. Demultiplexing</a></li>
<li class="toctree-l2"><a class="reference internal" href="primer_removal.html">2. Primer removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="pair_merging.html">3. Pair merging</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_concat.html">4. Data concatenation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">Filtering Amplicon Data</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otu_asv_read.html">OTUs, ASVs and Read Mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">Building OTU Phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../readprocessing.html">Read processing</a> &raquo;</li>
        
      <li>Fundamentals: FASTQ files</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/processing/fastq.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="fundamentals-fastq-files">
<span id="fastq"></span><h1>Fundamentals: FASTQ files<a class="headerlink" href="#fundamentals-fastq-files" title="Permalink to this headline">¶</a></h1>
<p>This page reviews a few basic points about sequence file structure and how to explore these files using the command line. If you have experience working with FASTQ files you can move onto the <a class="reference internal" href="demultiplexing.html#demultiplexing"><span class="std std-ref">demultiplexing section</span></a>. The commands here assume that your current working directory is the directory into which you copied the starting data.</p>
<p>Change into the directory containing the starting data and list its contents, showing sizes:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ <span class="nb">cd</span> 0_rawsequences/

$ ls -lh
</pre></div>
</div>
<p>We can see how many lines in each file using the word count function, specifying we want the number of lines:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ wc -l *.fastq
</pre></div>
</div>
<p>The ​``*.fastq`` here means we want all of the files ending in <code class="docutils literal notranslate"><span class="pre">.fastq</span></code> in the directory. We could replace this with a single file name if we just wanted to count the lines of a single file.</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ head -n <span class="m">10</span> ​file
</pre></div>
</div>
<p>You will see the FASTQ format comprising header, sequence and quality scores. A useful point to note is the structure of the file header, specifically that it starts with <code class="docutils literal notranslate"><span class="pre">&#64;D00</span></code>. If the structure of this file is completely new to you, take a few minutes to read the first section on the <a class="reference external" href="https://en.wikipedia.org/wiki/FASTQ_format">FASTQ wikipedia page</a>.</p>
<p>To get specific lines from a file, use the sed function:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ sed -n <span class="s1">&#39;4,8p&#39;</span> ​file     <span class="c1"># prints lines 4-8</span>
</pre></div>
</div>
<p>Use this to have a look at some files.</p>
<p>Note that the R1 and R2 files from the same library have the same read headers, apart from a 1 or 2 in the second part of the name. Reads with the same header were read from the same location on the sequencer, so they are assumed to be the forward and reverse read of the same fragment - these are called <strong>​mate pairs</strong>. It’s important to ensure that both the forward and reverse read for each fragment are always kept present and in the same relative location in the paired files (“in sync”) for some future processes.</p>
<p>We can use the <code class="docutils literal notranslate"><span class="pre">grep</span> <span class="pre">-c</span></code> function to count the number of sequences in a file (again, replace file with the name of one of the files):</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep -c <span class="s2">&quot;^@D00&quot;</span> ​file
</pre></div>
</div>
<p>If you want to learn more about grep, see <a class="reference internal" href="../gettingstarted/cli_bioinformatics/grep.html#grep"><span class="std std-ref">here</span></a>.</p>
<p>Like the ​wc -l function above, we can run grep on all of our files at once to get the total read numbers for each of our libraries:</p>
<div class="highlight-bash notranslate"><div class="highlight"><pre><span></span>$ grep -c <span class="s2">&quot;^@D00&quot;</span> *.fastq
</pre></div>
</div>
<p>We can see that we’re dealing with about 9000 reads per library.</p>
<div class="topic">
<p class="topic-title">Questions</p>
<p>Do all the libraries have the same R1 and R2 read numbers?</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="demultiplexing.html" class="btn btn-neutral float-right" title="1. Demultiplexing" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="../readprocessing.html" class="btn btn-neutral float-left" title="Read processing" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>