

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>2. Primer removal &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/css/custom.css" type="text/css" />

  
  
  
  

  
  <!--[if lt IE 9]>
    <script src="../_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script src="../_static/jquery.js"></script>
        <script src="../_static/underscore.js"></script>
        <script src="../_static/doctools.js"></script>
        <script src="../_static/language_data.js"></script>
        <script src="../_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="next" title="Extension: Quality Settings" href="extensions/quality_settings_extension.html" />
    <link rel="prev" title="Extension: Cutadapt parameters" href="extensions/cutadapt_extension.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home" alt="Documentation Home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="../gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="../data.html">Example data</a></li>
<li class="toctree-l1 current"><a class="reference internal" href="../readprocessing.html">A: Read processing</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="fastq.html">Fundamentals: FASTQ files</a></li>
<li class="toctree-l2"><a class="reference internal" href="demultiplexing.html">1. Demultiplexing</a></li>
<li class="toctree-l2 current"><a class="current reference internal" href="#">2. Primer removal</a><ul>
<li class="toctree-l3"><a class="reference internal" href="extensions/quality_settings_extension.html">Extension: Quality Settings</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="pair_merging.html">3. Pair merging</a></li>
<li class="toctree-l2"><a class="reference internal" href="data_concat.html">4. Data concatenation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="../filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="../asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="../building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="../otuid.html">Identifying OTU Sequences</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="../readprocessing.html">A: Read processing</a> &raquo;</li>
        
      <li>2. Primer removal</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/processing/primer_removal.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="primer-removal">
<span id="id1"></span><h1>2. Primer removal<a class="headerlink" href="#primer-removal" title="Permalink to this headline">¶</a></h1>
<div class="toctree-wrapper compound">
</div>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>When performing demultiplexing, <strong>cutadapt</strong> removed the indices from our samples. You can check this using the <code class="docutils literal notranslate"><span class="pre">grep</span></code> command from the previous tutorial. Alternatively, you can use the following command to explore the length distribution of the sequences. Run this on a pre- and post- demultiplexed file.</p>
<pre class="literal-block">sed -n '2~4p'​ <span class="var">input.fastq​</span> | while read l; do echo ${#l} ; done | sort | uniq -c</pre>
<div class="togglegreen admonition">
<p class="admonition-title">How this command works</p>
<p>The <code class="docutils literal notranslate"><span class="pre">sed</span></code> command ​p​rints only every 4th line, starting at the 2nd. This is sent to a <code class="docutils literal notranslate"><span class="pre">while</span></code> loop, which reads each line and stores it in <code class="docutils literal notranslate"><span class="pre">l</span></code> ​. The loop outputs the number of characters in the <code class="docutils literal notranslate"><span class="pre">l</span></code> variable, one per line. These are then sorted into alphanumeric order, and then each unique number is counted to get the distribution.</p>
</div>
<p>You should see that the average sequence length has reduced by 6.</p>
<p>We can also use <strong>cutadapt</strong> to remove primers. Unlike indices, which are not valid DNA sequences but artificial sequences we added, primer regions reflect real DNA sequence data. However, because we used degenerate primers, we cannot be certain that the amplified region of the primer sequence is exactly identical to that region on our source DNA, because primers do not always bind perfectly. So this region must be discarded.</p>
<div class="green admonition">
<p class="admonition-title">Data and software</p>
<p>The input data for this section is a directory of FASTQ format file pairs, one pair per sample, with primer sequences at (or near) the beginning of each read. If you’re following along step-by-step, this was produced in <a class="reference internal" href="demultiplexing.html#demultiplexing"><span class="std std-ref">the previous tutorial</span></a>. Alternatively, the <code class="docutils literal notranslate"><span class="pre">1_demux</span></code> directory within the <a class="reference internal" href="../data.html#sectionadata"><span class="std std-ref">sectionA archive</span></a> can be used as example data.</p>
<p>This tutorial uses the <a class="reference internal" href="../gettingstarted/setup/installing_software.html#cutadapt"><span class="std std-ref">cutadapt software</span></a>.</p>
</div>
</div>
<div class="section" id="performing-primer-removal">
<h2>Performing primer removal<a class="headerlink" href="#performing-primer-removal" title="Permalink to this headline">¶</a></h2>
<p>This process is very similar to demultiplexing, except we only have one sequence to remove, rather than three, and we only want one output file for each input file. The power of <strong>cutadapt</strong>’s paired-file-aware approach is that again, we can filter out any mate pairs that don’t have both primers - this is definitely a mark of a sequencing error!</p>
<div class="admonition-exercise admonition">
<p class="admonition-title">Exercise</p>
<p>First, create another new directory in our parent directory for the trimming output.</p>
<p>Making sure you’re in the parent directory, try and adapt our demultiplexing command to trim the primers (the sequences are given again below) from one of the demultiplexed file pairs. You will want to add the parameter <code class="docutils literal notranslate"><span class="pre">--discard-untrimmed</span></code> ​. We could have added this to demultiplexing to remove all “unknown” files as well.</p>
<p><strong>Forward(R1)</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">CCNGAYATRGCNTTYCCNCG</span>
</pre></div>
</div>
<p><strong>Reverse (R2)</strong></p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">TANACYTCNGGRTGNCCRAARAAYCA</span>
</pre></div>
</div>
<p><strong>Cutadapt</strong> is aware of ambiguous bases so it’s fine to use the primer sequences as-is. The primers should have been consecutive with the indices, so now must be at the start of the reads: thus you can use ​^ to anchor the sequence as before. You don’t need to name the primer sequences (<code class="docutils literal notranslate"><span class="pre">XX=</span></code>), and you don’t need to use <code class="docutils literal notranslate"><span class="pre">{name}</span></code>  in the output - the file name will do. Try running  your command, if it doesn’t work, check the solution below.</p>
</div>
<div class="toggle admonition">
<p class="admonition-title">Solution</p>
<p>Make sure you change <code class="docutils literal notranslate"><span class="pre">output</span></code> in the command below to the name of the new directory you created to store these files in. We’re assuming the directory with your files is named <code class="docutils literal notranslate"><span class="pre">1_demux</span></code>, if it isn’t, make sure to change that below as well.</p>
<pre class="literal-block">cutadapt -g ^CCNGAYATRGCNTTYCCNCG -G ^TANACYTCNGGRTGNCCRAARAAYCA \
-o <span class="var">output</span>/T11_R1.fastq -p <span class="var">output</span>/T11_R2.fastq --discard-untrimmed \
1_demux/T11_R1.fastq 1_demux/T11_R2.fastq</pre>
</div>
<p>Make sure to look over the output from <strong>cutadapt</strong> because this is very informative. You’ll notice now that some errors are being allowed, since these sequences are longer and so the default 10% allows 2 errors in these primers. Additionally you’ll notice that unlike with the indices, the length of sequence removed has varied slightly. We’ll come back to this.</p>
<p>We want to run this on all of our files, ideally without writing the command over and over. We can put this in a loop using bash. Since we need all of the unique samples, we first need to design a command for listing these:</p>
<pre class="literal-block">ls 1_demux/ | cut -d_ -f1 | sort | uniq</pre>
<p>This extracts the part of each name before the first ​``_`` and finds the unique ones. We can then store this list of library names in a bash variable, and check it’s contents with <code class="docutils literal notranslate"><span class="pre">echo</span></code>:</p>
<pre class="literal-block">samples=$(ls 1_demux/ | cut -d_ -f1 | sort | uniq)
echo $samples</pre>
<p>We can then loop on the contents of this variable using <code class="docutils literal notranslate"><span class="pre">for</span></code></p>
<pre class="literal-block">for s in $samples;
do
        cutadapt -g ^CCNGAYATRGCNTTYCCNCG -G ^TANACYTCNGGRTGNCCRAARAAYCA \
        -o <span class="var">output</span>/${s}_R1.fastq -p <span class="var">output</span>/${s}_R2.fastq --discard-untrimmed \
        1_demux/${s}_R1.fastq 1_demux/${s}_R2.fastq \
done</pre>
<p>Remember that you can write this as a single line without the <code class="docutils literal notranslate"><span class="pre">\</span></code> if you want:</p>
<pre class="literal-block">for s in $samples; do cutadapt -g ^CCNGAYATRGCNTTYCCNCG -G ^TANACYTCNGGRTGNCCRAARAAYCA -o <span class="var">output</span>/${s}_R1.fastq -p <span class="var">output</span>/${s}_R2.fastq --discard-untrimmed 1_demux/${s}_R1.fastq 1_demux/${s}_R2.fastq done</pre>
<p>The read command reads from the piped list command, and the while command works through this bit-by-bit. The <code class="docutils literal notranslate"><span class="pre">​$s</span></code> therefore refers to each sample name in turn - note that <code class="docutils literal notranslate"><span class="pre">​${s}</span></code> is used where we want to add a ​_ immediately after, otherwise bash will look for a variable called <code class="docutils literal notranslate"><span class="pre">$s_R1</span></code>.</p>
<p>Check your trimmed directory to make sure you have all of your files, and check back through the terminal output to make sure that you didn’t miss any errors. As always, review your read numbers.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>To run these loops, we generated a list of our unique sample names by listing our files and then editing the text to remove file extensions and direction information, as reprinted below.</p>
<pre class="literal-block">samples=$(ls 1_demux/ | cut -d_ -f1 | sort | uniq)</pre>
<p>If you have your own data, your file names are likely to be different and the exact code used above may not work! You will need to come up with your own version. If you have multiple underscores in your file names, but the number of underscores is consistent, you could simply tweak the number given to <code class="docutils literal notranslate"><span class="pre">cut</span> <span class="pre">-f</span></code>. Otherwise, you might need to use <code class="docutils literal notranslate"><span class="pre">sed</span></code> to remove parts of the names, or even use <code class="docutils literal notranslate"><span class="pre">rename</span></code> to rename the files to something that will work.</p>
</div>
</div>
<div class="section" id="next-steps">
<h2>Next steps<a class="headerlink" href="#next-steps" title="Permalink to this headline">¶</a></h2>
<p>We have generated another set of files, with the primers trimmed and some mate pairs rejected for not having the primer sequence present.</p>
<p>If you want to explore more about the quality settings of primer removal, there are some further exercises <a class="reference internal" href="extensions/quality_settings_extension.html#quality-settings"><span class="std std-ref">here</span></a>, but this isn’t necessary for the next step.</p>
<p>Next we join these mate pairs together into single reads representing individual amplicon sequences: <a class="reference internal" href="pair_merging.html#pair-merging"><span class="std std-ref">3. Pair merging</span></a>.</p>
</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="extensions/quality_settings_extension.html" class="btn btn-neutral float-right" title="Extension: Quality Settings" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="extensions/cutadapt_extension.html" class="btn btn-neutral float-left" title="Extension: Cutadapt parameters" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        
        &copy; Copyright 2020, The Vogler Lab

    </p>
  </div>
    
    
    
    Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>

        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>