

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Bioinformatic Methods for Biodiversity Metabarcoding &mdash; Bioinformatic Methods for Biodiversity Metabarcoding  documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/custom.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/language_data.js"></script>
        <script src="_static/js/custom.js"></script>
        <script async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/latest.js?config=TeX-AMS-MML_HTMLorMML"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Getting started" href="gettingstarted.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="#" class="icon icon-home"> Bioinformatic Methods for Biodiversity Metabarcoding
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a></li>
<li class="toctree-l1"><a class="reference internal" href="data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="readprocessing.html">A: Read processing</a></li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">B: Filtering amplicons</a></li>
<li class="toctree-l1"><a class="reference internal" href="asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a></li>
<li class="toctree-l1"><a class="reference internal" href="building_otu_phylogeny.html">D: Building OTU phylogeny</a></li>
<li class="toctree-l1"><a class="reference internal" href="otuid.html">E: Identifying OTU sequences</a></li>
<li class="toctree-l1"><a class="reference internal" href="contactus.html">Contact us</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="#">Bioinformatic Methods for Biodiversity Metabarcoding</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="#" class="icon icon-home"></a> &raquo;</li>
        
      <li>Bioinformatic Methods for Biodiversity Metabarcoding</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            <a href="_sources/index.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bioinformatic-methods-for-biodiversity-metabarcoding">
<span id="index"></span><h1>Bioinformatic Methods for Biodiversity Metabarcoding<a class="headerlink" href="#bioinformatic-methods-for-biodiversity-metabarcoding" title="Permalink to this headline">¶</a></h1>
<div class="section" id="introduction">
<h2>Introduction<a class="headerlink" href="#introduction" title="Permalink to this headline">¶</a></h2>
<p>Welcome to the online resources for Bioinformatic Methods for Biodiversity Metabarcoding. These resources aim to introduce a beginner to the bioinformatics required for metabarcoding through a self-guided practical course, and act as a reference for those more experienced in bioinformatics or metabarcoding. To this end, these resources are structured as a set of independent tutorials. Using a set of example data, or your own dataset, you can follow along with these tutorials, step-by-step, to learn how to go from raw sequence reads to the generation of biodiversity data for use in a wide range of applications. Alternatively, you can seek out only those areas where you want to learn more, as most tutorials can stand alone, with clear guidance on what input data and software might bee needed.</p>
<p>Each tutorial is generally centred around a particular step in metabarcoding, introducing the important concepts of this step, a software tool to perform this step, and key parameters to consider when using this software. Some tutorials also have expansion exercises, which explore more advanced topics around a bioinformatic step for those looking for an added challenge. There is also a substantial reference section covering basic topics in bioinformatics to assist those who are just starting out.</p>
<p>While we have made every effort to cover a wide range of metabarcoding bioinfirmatic topics, metabarcoding is a continually evolving field and it’s impossible to be comprehensive. This course explicitly focuses on metabarcoding for bulk community samples of arthropods using protein-coding markers, as distinct from many other forms of metabarcoding such as environmental DNA sequencing, dietary-derived metabarcoding, or microbiome sequencing using ribosomal RNA markers. As such, we do not tackle some areas, such as adaptation of the bioinformatic approaches to degraded DNA, and some areas we do cover may not be universal, such as translation-based ASV filtering. Nonetheless, the majority of the core concepts are identical across all metabarcoding so we hope this resource will be useful to any metabarcoder, no matter their background.</p>
</div>
<div class="section" id="feedback">
<h2>Feedback<a class="headerlink" href="#feedback" title="Permalink to this headline">¶</a></h2>
<p>We’re really keen to hear about your experience using these resources, so whether you’ve just read one or two pages or gone through the entire site, please fill in our <a class="reference external" href="https://forms.gle/46FSR9husitW6LDR7">feedback form</a> - it’ll only take a minute!</p>
</div>
<div class="section" id="resource-structure">
<h2>Resource structure<a class="headerlink" href="#resource-structure" title="Permalink to this headline">¶</a></h2>
<p>These resources are structured into sections, accessible through the links on the left.</p>
<p>We suggest you start with the <a class="reference internal" href="gettingstarted.html#gettingstarted"><span class="std std-ref">Getting started</span></a> section, which will cover the computational requirements of the course and outline in more detail how best to use the tutorials. This section also has a <a class="reference internal" href="gettingstarted/cli_bioinformatics.html#cli-bioinformatics"><span class="std std-ref">reference section</span></a> for hints and tips for those new to the Linux commandline environment. Even if you’re an experienced bioinformatician and just want to get going, we suggest you read at least the <a class="reference internal" href="gettingstarted/how_to_use.html#how-to-use"><span class="std std-ref">How to use these resources</span></a> to familiarise yourself with the format of the tutorials.</p>
<p>If you want to use our example dataset, you should then have a look at the <a class="reference internal" href="data.html#data"><span class="std std-ref">Example data</span></a> section, which provides instructions on how to download this data.</p>
<p>The main course consists of five sections, A to E. If you want to follow through this step-by-step, these sections can be tackled in a linear fashion. So, once you have access to a suitable computer and have read about the example data, head to the first section, on <a class="reference internal" href="readprocessing.html#read-processing"><span class="std std-ref">Read processing</span></a> where you will learn about the steps needed to process raw sequence reads into complete amplicon sequences. Once you’ve worked through the four sequential tutorials in that section, you can then move on to the <a class="reference internal" href="filtering.html#filtering"><span class="std std-ref">Filtering amplicon data</span></a> section, where you will learn about sources of errors in metabarcoding datasets and how to remove them in six tutorials. Once you have error-free sequences, head to the section on <a class="reference internal" href="asv_otu_readmap/mapping_reads.html#mapping-reads"><span class="std std-ref">OTUs, ASVs and read mapping</span></a> to learn how to generate biologically meaningful sequence units and community data. Then, the next section on <a class="reference internal" href="building_otu_phylogeny.html#phylogeny"><span class="std std-ref">Building OTU phylogeny</span></a> will take you through how to construct phylogenetic trees using your sequences. Finally, learn how to generate taxonomic information for your sequences in the final section on <a class="reference internal" href="otuid.html#otuid"><span class="std std-ref">Identifying OTU sequences</span></a>.</p>
<p>If you already have some experience with metabarcoding bioinformatics and want to learn a specific topic, you can check out the complete index of the tutorials available below, or use the search bar in the top left hand corner to find the topic you’re interested in. While we try to make each tutorial stand alone, to avoid unecessary redundancy some relevant information may be found in the parent page of a specific tutorial, so we recommend if you’re intested in, for example, <a class="reference internal" href="asv_otu_readmap/otudelim/greedy.html#greedy-clustering"><span class="std std-ref">greedy clustering</span></a>, you make sure to read the pages on <a class="reference internal" href="asv_otu_readmap/mapping_reads.html#mapping-reads"><span class="std std-ref">OTUs, ASVs and read mapping</span></a> and <a class="reference internal" href="asv_otu_readmap/otudelim.html#otu-delim"><span class="std std-ref">OTU delimitation</span></a> first.</p>
</div>
<div class="section" id="page-tree">
<h2>Page tree<a class="headerlink" href="#page-tree" title="Permalink to this headline">¶</a></h2>
<div class="toctree-wrapper compound">
<ul>
<li class="toctree-l1"><a class="reference internal" href="gettingstarted.html">Getting started</a><ul>
<li class="toctree-l2"><a class="reference internal" href="gettingstarted/setup.html">Setting up a bioinformatics machine</a></li>
<li class="toctree-l2"><a class="reference internal" href="gettingstarted/remote_access.html">Accessing linux servers</a></li>
<li class="toctree-l2"><a class="reference internal" href="gettingstarted/how_to_use.html">How to use these resources</a></li>
<li class="toctree-l2"><a class="reference internal" href="gettingstarted/toolbox_pipelines.html">The bioinformatics toolbox</a></li>
<li class="toctree-l2"><a class="reference internal" href="gettingstarted/cli_bioinformatics.html">Command line bioinformatics</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="data.html">Example data</a></li>
<li class="toctree-l1"><a class="reference internal" href="readprocessing.html">A: Read processing</a><ul>
<li class="toctree-l2"><a class="reference internal" href="processing/fastq.html">Fundamentals: FASTQ files</a></li>
<li class="toctree-l2"><a class="reference internal" href="processing/demultiplexing.html">1. Demultiplexing</a></li>
<li class="toctree-l2"><a class="reference internal" href="processing/primer_removal.html">2. Primer removal</a></li>
<li class="toctree-l2"><a class="reference internal" href="processing/pair_merging.html">3. Pair merging</a></li>
<li class="toctree-l2"><a class="reference internal" href="processing/data_concat.html">4. Data concatenation</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="filtering.html">B: Filtering amplicons</a><ul>
<li class="toctree-l2"><a class="reference internal" href="filtering/quality_filtering.html">1. Quality filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtering/not_filtering_dereplication.html">2. (not filtering) Dereplication</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtering/freq_filtering_and_denoising.html">3. Frequency filtering / denoising</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtering/indel_filtering.html">4. Indel filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtering/point_error.html">5. Point error filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtering/chimera_filtering.html">6. Chimera filtering</a></li>
<li class="toctree-l2"><a class="reference internal" href="filtering/metamate.html">Extension: Stringent ASV filtering</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="asv_otu_readmap.html">C: ASVs, OTUs and read mapping</a><ul>
<li class="toctree-l2"><a class="reference internal" href="asv_otu_readmap/otudelim.html">OTU delimitation</a></li>
<li class="toctree-l2"><a class="reference internal" href="asv_otu_readmap/mapping_reads.html">Mapping reads</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="building_otu_phylogeny.html">D: Building OTU phylogeny</a><ul>
<li class="toctree-l2"><a class="reference internal" href="otu_phylogeny/otu_trees.html">Making a barcode tree</a></li>
<li class="toctree-l2"><a class="reference internal" href="otu_phylogeny/placing_otus_in_trees.html">Phylogenetic placement</a></li>
<li class="toctree-l2"><a class="reference internal" href="otu_phylogeny/trees.html">Building a mitogenome reference tree</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="otuid.html">E: Identifying OTU sequences</a><ul>
<li class="toctree-l2"><a class="reference internal" href="otuid/database.html">Database classification</a></li>
<li class="toctree-l2"><a class="reference internal" href="otuid/id_local.html">Identifying OTUs with references</a></li>
<li class="toctree-l2"><a class="reference internal" href="otuid/phylogeny_based.html">Phylogenetic classification</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="contactus.html">Contact us</a></li>
</ul>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="gettingstarted.html" class="btn btn-neutral float-right" title="Getting started" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 2020, The Vogler Lab.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
    <p><br>Creedy, Vogler & Penlington, Natural History Museum London, Bioinformatic Methods for Biodiversity Metabarcoding. This site is a a deliverable of the iBioGen consortium funded by EU grant agreement No 810729, the content of this site is licensed under the <a href="https://creativecommons.org/licenses/by-sa/4.0/" target="blank">Creative Commons Attribution ShareAlike 4.0 License</a></p>


</footer>
        </div>
      </div>

    </section>

  </div>
  

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>